System.register(["./vendor/svelte-CkBJVEsY.js","./vendor/carbon-components-svelte-DfRG9uuU.js","./vendor/carbon-icons-svelte-Boxs1M4A.js","./vendor/temml-BTciUaue.js"],function(e){"use strict";var n,t,s,r,a,i,o,l,c,p,d,_,u,m,f,g,$,h,b,v,x,y,C,S,w,M,R,N,G,L,P,T,A,E,I,U,k,V,B,D,O,z,F,j,H,q,K,W,X,Y,Z,J,Q,ee,ne,te,se,re,ae,ie,oe,le,ce,pe,de,_e,ue,me,fe,ge,$e,he,be,ve,xe,ye,Ce,Se,we,Me,Re,Ne,Ge,Le,Pe,Te;return{setters:[function(e){n=e.S,t=e.i,s=e.s,r=e.a3,a=e.d,i=e.a8,o=e.t,l=e.a,c=e.g,p=e.c,d=e.b,_=e.p,u=e.Q,m=e.B,f=e.e,g=e.K,$=e.M,h=e.L,b=e.z,v=e.I,x=e.T,y=e.U,C=e.D,S=e.q,w=e.F,M=e.G,R=e.N,N=e.y,G=e.k,L=e.u,P=e.l,T=e.m,A=e.h,E=e.o,I=e.a2,U=e.f,k=e.j,V=e.A,B=e.ac,D=e.n,O=e.W,z=e.ad,F=e.x,j=e.ae,H=e.R},function(e){q=e.S,K=e.a,W=e.b,X=e.H,Y=e.T,Z=e.c,J=e.d,Q=e.e,ee=e.f,ne=e.g,te=e.M,se=e.I,re=e.h,ae=e.i,ie=e.j,oe=e.C,le=e.k,ce=e.l,pe=e.G,de=e.R,_e=e.m,ue=e.B,me=e.n,fe=e.o,ge=e.A},function(e){$e=e.C,he=e.a,be=e.D,ve=e.A,xe=e.S,ye=e.I,Ce=e.b,Se=e.c,we=e.V,Me=e.d,Re=e.T,Ne=e.G,Ge=e.J,Le=e.P,Pe=e.e,Te=e.f},null],execute:function(){function Ae(e,n,t){const s=e.slice();return s[4]=n[t],s}function Ee(e,n,t){const s=e.slice();return s[7]=n[t],s}function Ie(e){let n,t;return n=new K({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text,expanded:!0,$$slots:{default:[Ve]},$$scope:{ctx:e}}}),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},p(e,t){const s={};1&t&&(s.href=`${e[1](e[4].page)}#${e[4].slug}`),1&t&&(s.class=e[2](e[4].page)),1&t&&(s.title=e[4].text),1&t&&(s.text=e[4].text),1025&t&&(s.$$scope={dirty:t,ctx:e}),n.$set(s)},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function Ue(e){let n,t;return n=new q({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text}}),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},p(e,t){const s={};1&t&&(s.href=`${e[1](e[4].page)}#${e[4].slug}`),1&t&&(s.class=e[2](e[4].page)),1&t&&(s.title=e[4].text),1&t&&(s.text=e[4].text),n.$set(s)},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function ke(e){let n,t;return n=new W({props:{href:`${e[1](e[7].page)}#${e[7].slug}`,class:e[2](e[7].page),title:e[7].text,text:e[7].text}}),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},p(e,t){const s={};1&t&&(s.href=`${e[1](e[7].page)}#${e[7].slug}`),1&t&&(s.class=e[2](e[7].page)),1&t&&(s.title=e[7].text),1&t&&(s.text=e[7].text),n.$set(s)},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function Ve(e){let n,t,s=r(e[4].children),_=[];for(let n=0;n<s.length;n+=1)_[n]=ke(Ee(e,s,n));const u=e=>o(_[e],1,1,()=>{_[e]=null});return{c(){for(let e=0;e<_.length;e+=1)_[e].c();n=b()},m(e,s){for(let n=0;n<_.length;n+=1)_[n]&&_[n].m(e,s);d(e,n,s),t=!0},p(e,t){if(7&t){let a;for(s=r(e[4].children),a=0;a<s.length;a+=1){const r=Ee(e,s,a);_[a]?(_[a].p(r,t),l(_[a],1)):(_[a]=ke(r),_[a].c(),l(_[a],1),_[a].m(n.parentNode,n))}for(c(),a=s.length;a<_.length;a+=1)u(a);p()}},i(e){if(!t){for(let e=0;e<s.length;e+=1)l(_[e]);t=!0}},o(e){_=_.filter(Boolean);for(let e=0;e<_.length;e+=1)o(_[e]);t=!1},d(e){e&&a(n),i(_,e)}}}function Be(e){let n,t,s,r;const i=[Ue,Ie],_=[];function u(e,n){return 0==e[4].children.length?0:1}return n=u(e),t=_[n]=i[n](e),{c(){t.c(),s=f()},m(e,t){_[n].m(e,t),d(e,s,t),r=!0},p(e,r){let a=n;n=u(e),n===a?_[n].p(e,r):(c(),o(_[a],1,1,()=>{_[a]=null}),p(),t=_[n],t?t.p(e,r):(t=_[n]=i[n](e),t.c()),l(t,1),t.m(s.parentNode,s))},i(e){r||(l(t),r=!0)},o(e){o(t),r=!1},d(e){e&&a(s),_[n].d(e)}}}function De(e){let n,t,s=r(e[0]),m=[];for(let n=0;n<s.length;n+=1)m[n]=Be(Ae(e,s,n));const f=e=>o(m[e],1,1,()=>{m[e]=null});return{c(){n=_("ul");for(let e=0;e<m.length;e+=1)m[e].c();u(n,"class","bx--side-nav__items pipen-report-sidenav svelte-12gpq3")},m(e,s){d(e,n,s);for(let e=0;e<m.length;e+=1)m[e]&&m[e].m(n,null);t=!0},p(e,[t]){if(7&t){let a;for(s=r(e[0]),a=0;a<s.length;a+=1){const r=Ae(e,s,a);m[a]?(m[a].p(r,t),l(m[a],1)):(m[a]=Be(r),m[a].c(),l(m[a],1),m[a].m(n,null))}for(c(),a=s.length;a<m.length;a+=1)f(a);p()}},i(e){if(!t){for(let e=0;e<s.length;e+=1)l(m[e]);t=!0}},o(e){m=m.filter(Boolean);for(let e=0;e<m.length;e+=1)o(m[e]);t=!1},d(e){e&&a(n),i(m,e)}}}function Oe(e,n,t){let{toc:s}=n,{page:r}=n;return m(()=>{let e,n=new URL(window.location.href);e=n.hash.length>0?document.querySelector(`[href="${n.hash}"]`):document.querySelector(".toc-current-page"),e.scrollIntoView({behavior:"smooth"})}),e.$$set=e=>{"toc"in e&&t(0,s=e.toc),"page"in e&&t(3,r=e.page)},[s,e=>{if(r===e)return"";let n=new URL(window.location.href);n.hash="";const t=(n.searchParams.get("proc")||"_index").replace(/-\d+$/,"");return n.searchParams.set("proc",0===e?t:`${t}-${e}`),n.toString()},e=>e!=r?"toc-noncurrent-page":"toc-current-page",r]}class ze extends n{constructor(e){super(),t(this,e,Oe,De,s,{toc:0,page:3})}}function Fe(e,n,t){const s=e.slice();return s[7]=n[t],s}function je(e,n,t){const s=e.slice();return s[10]=n[t],s}function He(e){let n,t;return n=new J({props:{isOpen:pn,transition:e[5],$$slots:{default:[sn]},$$scope:{ctx:e}}}),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},p(e,t){const s={};8197&t&&(s.$$scope={dirty:t,ctx:e}),n.$set(s)},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function qe(e){let n;return{c(){n=M("Go to home:")},m(e,t){d(e,n,t)},d(e){e&&a(n)}}}function Ke(e){let n;return{c(){n=M(e[0])},m(e,t){d(e,n,t)},p(e,t){1&t&&w(n,e[0])},d(e){e&&a(n)}}}function We(e){let n;return{c(){n=M("Switch to process:")},m(e,t){d(e,n,t)},d(e){e&&a(n)}}}function Xe(e){let n,t;return n=new ne({props:{title:e[7].name,href:`?proc=${e[7].name}`,$$slots:{default:[Ze]},$$scope:{ctx:e}}}),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},p(e,t){const s={};4&t&&(s.title=e[7].name),4&t&&(s.href=`?proc=${e[7].name}`),8196&t&&(s.$$scope={dirty:t,ctx:e}),n.$set(s)},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function Ye(e){let n,t,s,_;n=new ne({props:{class:"procgroup",$$slots:{default:[Je]},$$scope:{ctx:e}}});let u=r(e[7].procs),m=[];for(let n=0;n<u.length;n+=1)m[n]=en(je(e,u,n));const v=e=>o(m[e],1,1,()=>{m[e]=null});return{c(){h(n.$$.fragment),t=b();for(let e=0;e<m.length;e+=1)m[e].c();s=f()},m(e,r){$(n,e,r),d(e,t,r);for(let n=0;n<m.length;n+=1)m[n]&&m[n].m(e,r);d(e,s,r),_=!0},p(e,t){const a={};if(8196&t&&(a.$$scope={dirty:t,ctx:e}),n.$set(a),4&t){let n;for(u=r(e[7].procs),n=0;n<u.length;n+=1){const r=je(e,u,n);m[n]?(m[n].p(r,t),l(m[n],1)):(m[n]=en(r),m[n].c(),l(m[n],1),m[n].m(s.parentNode,s))}for(c(),n=u.length;n<m.length;n+=1)v(n);p()}},i(e){if(!_){l(n.$$.fragment,e);for(let e=0;e<u.length;e+=1)l(m[e]);_=!0}},o(e){o(n.$$.fragment,e),m=m.filter(Boolean);for(let e=0;e<m.length;e+=1)o(m[e]);_=!1},d(e){e&&(a(t),a(s)),g(n,e),i(m,e)}}}function Ze(e){let n,t,s=e[7].name+"";return{c(){n=M(s),t=b()},m(e,s){d(e,n,s),d(e,t,s)},p(e,t){4&t&&s!==(s=e[7].name+"")&&w(n,s)},d(e){e&&(a(n),a(t))}}}function Je(e){let n,t=e[7].name+"";return{c(){n=M(t)},m(e,t){d(e,n,t)},p(e,s){4&s&&t!==(t=e[7].name+"")&&w(n,t)},d(e){e&&a(n)}}}function Qe(e){let n,t,s,r=e[10].name+"";return{c(){n=M("- "),t=M(r),s=b()},m(e,r){d(e,n,r),d(e,t,r),d(e,s,r)},p(e,n){4&n&&r!==(r=e[10].name+"")&&w(t,r)},d(e){e&&(a(n),a(t),a(s))}}}function en(e){let n,t;return n=new ne({props:{title:e[10].name,class:"procgroup-proc",href:`?proc=${e[10].name}`,$$slots:{default:[Qe]},$$scope:{ctx:e}}}),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},p(e,t){const s={};4&t&&(s.title=e[10].name),4&t&&(s.href=`?proc=${e[10].name}`),8196&t&&(s.$$scope={dirty:t,ctx:e}),n.$set(s)},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function nn(e){let n,t,s,r;const i=[Ye,Xe],_=[];function u(e,n){return e[7].procs?0:1}return n=u(e),t=_[n]=i[n](e),{c(){t.c(),s=f()},m(e,t){_[n].m(e,t),d(e,s,t),r=!0},p(e,r){let a=n;n=u(e),n===a?_[n].p(e,r):(c(),o(_[a],1,1,()=>{_[a]=null}),p(),t=_[n],t?t.p(e,r):(t=_[n]=i[n](e),t.c()),l(t,1),t.m(s.parentNode,s))},i(e){r||(l(t),r=!0)},o(e){o(t),r=!1},d(e){e&&a(s),_[n].d(e)}}}function tn(e){let n,t,s,_,u,m,v,x;n=new ee({props:{$$slots:{default:[qe]},$$scope:{ctx:e}}}),s=new ne({props:{href:"index.html",$$slots:{default:[Ke]},$$scope:{ctx:e}}}),u=new ee({props:{$$slots:{default:[We]},$$scope:{ctx:e}}});let y=r(e[2]),C=[];for(let n=0;n<y.length;n+=1)C[n]=nn(Fe(e,y,n));const S=e=>o(C[e],1,1,()=>{C[e]=null});return{c(){h(n.$$.fragment),t=b(),h(s.$$.fragment),_=b(),h(u.$$.fragment),m=b();for(let e=0;e<C.length;e+=1)C[e].c();v=f()},m(e,r){$(n,e,r),d(e,t,r),$(s,e,r),d(e,_,r),$(u,e,r),d(e,m,r);for(let n=0;n<C.length;n+=1)C[n]&&C[n].m(e,r);d(e,v,r),x=!0},p(e,t){const a={};8192&t&&(a.$$scope={dirty:t,ctx:e}),n.$set(a);const i={};8193&t&&(i.$$scope={dirty:t,ctx:e}),s.$set(i);const o={};if(8192&t&&(o.$$scope={dirty:t,ctx:e}),u.$set(o),4&t){let n;for(y=r(e[2]),n=0;n<y.length;n+=1){const s=Fe(e,y,n);C[n]?(C[n].p(s,t),l(C[n],1)):(C[n]=nn(s),C[n].c(),l(C[n],1),C[n].m(v.parentNode,v))}for(c(),n=y.length;n<C.length;n+=1)S(n);p()}},i(e){if(!x){l(n.$$.fragment,e),l(s.$$.fragment,e),l(u.$$.fragment,e);for(let e=0;e<y.length;e+=1)l(C[e]);x=!0}},o(e){o(n.$$.fragment,e),o(s.$$.fragment,e),o(u.$$.fragment,e),C=C.filter(Boolean);for(let e=0;e<C.length;e+=1)o(C[e]);x=!1},d(e){e&&(a(t),a(_),a(m),a(v)),g(n,e),g(s,e),g(u,e),i(C,e)}}}function sn(e){let n,t;return n=new Q({props:{$$slots:{default:[tn]},$$scope:{ctx:e}}}),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},p(e,t){const s={};8197&t&&(s.$$scope={dirty:t,ctx:e}),n.$set(s)},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function rn(e){let n,t,s,r;t=new Y({props:{render:"select",theme:"g10",persist:!0,persistKey:"__pipen-report-theme",select:{themes:["g10","white","g80","g90","g100"],labelText:"Theme",inline:!0}}});let i=e[2].length>0&&He(e);return{c(){n=_("div"),h(t.$$.fragment),s=b(),i&&i.c(),u(n,"class","header-util"),S(n,"bx--header__global",!0)},m(e,a){d(e,n,a),$(t,n,null),C(n,s),i&&i.m(n,null),r=!0},p(e,t){e[2].length>0?i?(i.p(e,t),4&t&&l(i,1)):(i=He(e),i.c(),l(i,1),i.m(n,null)):i&&(c(),o(i,1,1,()=>{i=null}),p())},i(e){r||(l(t.$$.fragment,e),l(i),r=!0)},o(e){o(t.$$.fragment,e),o(i),r=!1},d(e){e&&a(n),g(t),i&&i.d()}}}function an(e){let n;return{c(){n=M(e[4])},m(e,t){d(e,n,t)},p(e,t){16&t&&w(n,e[4])},d(e){e&&a(n)}}}function on(e){let n,t,s,r,i=null!==e[4]&&an(e);return{c(){n=_("h1"),t=M(e[3]),s=b(),i&&i.c(),r=f()},m(e,a){d(e,n,a),C(n,t),d(e,s,a),i&&i.m(e,a),d(e,r,a)},p(e,n){8&n&&w(t,e[3]),null!==e[4]?i?i.p(e,n):(i=an(e),i.c(),i.m(r.parentNode,r)):i&&(i.d(1),i=null)},d(e){e&&(a(n),a(s),a(r)),i&&i.d(e)}}}function ln(e){let n,t;return n=new Z({props:{class:"header-tile",slot:"platform",$$slots:{default:[on]},$$scope:{ctx:e}}}),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},p(e,t){const s={};8216&t&&(s.$$scope={dirty:t,ctx:e}),n.$set(s)},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function cn(e){let n,t,s;function r(n){e[6](n)}let a={$$slots:{platform:[ln],default:[rn]},$$scope:{ctx:e}};return void 0!==e[1]&&(a.isSideNavOpen=e[1]),n=new X({props:a}),v.push(()=>x(n,"isSideNavOpen",r)),{c(){h(n.$$.fragment)},m(e,t){$(n,e,t),s=!0},p(e,[s]){const r={};8221&s&&(r.$$scope={dirty:s,ctx:e}),!t&&2&s&&(t=!0,r.isSideNavOpen=e[1],y(()=>t=!1)),n.$set(r)},i(e){s||(l(n.$$.fragment,e),s=!0)},o(e){o(n.$$.fragment,e),s=!1},d(e){g(n,e)}}}let pn=!1;function dn(e,n,t){let{entries:s}=n,{logo:r}=n,{logotext:a=null}=n,{pipeline_name:i}=n,{isSideNavOpen:o}=n;return i=i||r,e.$$set=e=>{"entries"in e&&t(2,s=e.entries),"logo"in e&&t(3,r=e.logo),"logotext"in e&&t(4,a=e.logotext),"pipeline_name"in e&&t(0,i=e.pipeline_name),"isSideNavOpen"in e&&t(1,o=e.isSideNavOpen)},[i,o,s,r,a,{duration:200},function(e){o=e,t(1,o)}]}class _n extends n{constructor(e){super(),t(this,e,dn,cn,s,{entries:2,logo:3,logotext:4,pipeline_name:0,isSideNavOpen:1})}}function un(e){let n,t,s;function r(n){e[3](n)}let a={modalHeading:"Running Information",passiveModal:!0,hasScrollingContent:!0,class:"runinfo",$$slots:{default:[bn]},$$scope:{ctx:e}};return void 0!==e[2]&&(a.open=e[2]),n=new te({props:a}),v.push(()=>x(n,"open",r)),{c(){h(n.$$.fragment)},m(e,t){$(n,e,t),s=!0},p(e,s){const r={};34&s&&(r.$$scope={dirty:s,ctx:e}),!t&&4&s&&(t=!0,r.open=e[2],y(()=>t=!1)),n.$set(r)},i(e){s||(l(n.$$.fragment,e),s=!0)},o(e){o(n.$$.fragment,e),s=!1},d(e){g(n,e)}}}function mn(e){let n,t,s,r,i,c;return n=new ae({props:{label:"Session Information"}}),s=new ae({props:{label:"Time Spent"}}),i=new ae({props:{label:"Device Information"}}),{c(){h(n.$$.fragment),t=b(),h(s.$$.fragment),r=b(),h(i.$$.fragment)},m(e,a){$(n,e,a),d(e,t,a),$(s,e,a),d(e,r,a),$(i,e,a),c=!0},p:R,i(e){c||(l(n.$$.fragment,e),l(s.$$.fragment,e),l(i.$$.fragment,e),c=!0)},o(e){o(n.$$.fragment,e),o(s.$$.fragment,e),o(i.$$.fragment,e),c=!1},d(e){e&&(a(t),a(r)),g(n,e),g(s,e),g(i,e)}}}function fn(e){let n,t;return n=new oe({props:{type:"multi",code:e[1].session}}),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},p(e,t){const s={};2&t&&(s.code=e[1].session),n.$set(s)},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function gn(e){let n,t;return n=new oe({props:{type:"multi",code:e[1].time}}),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},p(e,t){const s={};2&t&&(s.code=e[1].time),n.$set(s)},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function $n(e){let n,t;return n=new oe({props:{type:"multi",code:e[1].device}}),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},p(e,t){const s={};2&t&&(s.code=e[1].device),n.$set(s)},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function hn(e){let n,t,s,r,i,c;return n=new ie({props:{$$slots:{default:[fn]},$$scope:{ctx:e}}}),s=new ie({props:{$$slots:{default:[gn]},$$scope:{ctx:e}}}),i=new ie({props:{$$slots:{default:[$n]},$$scope:{ctx:e}}}),{c(){h(n.$$.fragment),t=b(),h(s.$$.fragment),r=b(),h(i.$$.fragment)},m(e,a){$(n,e,a),d(e,t,a),$(s,e,a),d(e,r,a),$(i,e,a),c=!0},p(e,t){const r={};34&t&&(r.$$scope={dirty:t,ctx:e}),n.$set(r);const a={};34&t&&(a.$$scope={dirty:t,ctx:e}),s.$set(a);const o={};34&t&&(o.$$scope={dirty:t,ctx:e}),i.$set(o)},i(e){c||(l(n.$$.fragment,e),l(s.$$.fragment,e),l(i.$$.fragment,e),c=!0)},o(e){o(n.$$.fragment,e),o(s.$$.fragment,e),o(i.$$.fragment,e),c=!1},d(e){e&&(a(t),a(r)),g(n,e),g(s,e),g(i,e)}}}function bn(e){let n,t,s,r;return n=new se({props:{kind:"warning",lowContrast:!0,hideCloseButton:!0,title:"Showing the first job only. Check the workdir for information of other jobs if any."}}),s=new re({props:{$$slots:{content:[hn],default:[mn]},$$scope:{ctx:e}}}),{c(){h(n.$$.fragment),t=b(),h(s.$$.fragment)},m(e,a){$(n,e,a),d(e,t,a),$(s,e,a),r=!0},p(e,n){const t={};34&n&&(t.$$scope={dirty:n,ctx:e}),s.$set(t)},i(e){r||(l(n.$$.fragment,e),l(s.$$.fragment,e),r=!0)},o(e){o(n.$$.fragment,e),o(s.$$.fragment,e),r=!1},d(e){e&&a(t),g(n,e),g(s,e)}}}function vn(e){let n,t,s;return{c(){n=_("a"),n.textContent="> Running Information",u(n,"href","javascript:void(0);")},m(r,a){d(r,n,a),t||(s=N(n,"click",e[4]),t=!0)},p:R,d(e){e&&a(n),t=!1,s()}}}function xn(e){let n,t,s,r,i,m,f,g,$=e[1]&&un(e),h=e[1]&&vn(e);return{c(){$&&$.c(),n=b(),t=_("div"),s=_("div"),h&&h.c(),r=b(),i=_("div"),m=M("Powered by "),f=_("span"),u(s,"class","runinfo"),u(f,"class","pipen-report-footer-versions"),u(i,"class","versions"),u(t,"class","pipen-report-footer bx--grid")},m(a,o){$&&$.m(a,o),d(a,n,o),d(a,t,o),C(t,s),h&&h.m(s,null),C(t,r),C(t,i),C(i,m),C(i,f),f.innerHTML=e[0],g=!0},p(e,[t]){e[1]?$?($.p(e,t),2&t&&l($,1)):($=un(e),$.c(),l($,1),$.m(n.parentNode,n)):$&&(c(),o($,1,1,()=>{$=null}),p()),e[1]?h?h.p(e,t):(h=vn(e),h.c(),h.m(s,null)):h&&(h.d(1),h=null),(!g||1&t)&&(f.innerHTML=e[0])},i(e){g||(l($),g=!0)},o(e){o($),g=!1},d(e){e&&(a(n),a(t)),$&&$.d(e),h&&h.d()}}}function yn(e,n,t){let{versions:s}=n,{runinfo:r}=n,a=!1;return e.$$set=e=>{"versions"in e&&t(0,s=e.versions),"runinfo"in e&&t(1,r=e.runinfo)},[s,r,a,function(e){a=e,t(2,a)},()=>{t(2,a=!0)}]}class Cn extends n{constructor(e){super(),t(this,e,yn,xn,s,{versions:0,runinfo:1})}}function Sn(e){let n,t,s;function r(n){e[12](n)}let a={$$slots:{default:[wn]},$$scope:{ctx:e}};return void 0!==e[0]&&(a.isOpen=e[0]),n=new ce({props:a}),v.push(()=>x(n,"isOpen",r)),{c(){h(n.$$.fragment)},m(e,t){$(n,e,t),s=!0},p(e,s){const r={};8576&s&&(r.$$scope={dirty:s,ctx:e}),!t&&1&s&&(t=!0,r.isOpen=e[0],y(()=>t=!1)),n.$set(r)},i(e){s||(l(n.$$.fragment,e),s=!0)},o(e){o(n.$$.fragment,e),s=!1},d(e){g(n,e)}}}function wn(e){let n,t;return n=new ze({props:{toc:e[7],page:e[8]}}),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},p(e,t){const s={};128&t&&(s.toc=e[7]),256&t&&(s.page=e[8]),n.$set(s)},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function Mn(e){let n;const t=e[10].default,s=G(t,e,e[13],null);return{c(){s&&s.c()},m(e,t){s&&s.m(e,t),n=!0},p(e,r){s&&s.p&&(!n||8192&r)&&L(s,t,e,e[13],n?T(t,e[13],r,null):P(e[13]),null)},i(e){n||(l(s,e),n=!0)},o(e){o(s,e),n=!1},d(e){s&&s.d(e)}}}function Rn(e){let n,t;return n=new _e({props:{$$slots:{default:[Mn]},$$scope:{ctx:e}}}),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},p(e,t){const s={};8192&t&&(s.$$scope={dirty:t,ctx:e}),n.$set(s)},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function Nn(e){let n,t;return n=new de({props:{$$slots:{default:[Rn]},$$scope:{ctx:e}}}),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},p(e,t){const s={};8192&t&&(s.$$scope={dirty:t,ctx:e}),n.$set(s)},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function Gn(e){let n,t,s,r;return n=new pe({props:{$$slots:{default:[Nn]},$$scope:{ctx:e}}}),s=new Cn({props:{versions:e[3],runinfo:e[9]}}),{c(){h(n.$$.fragment),t=b(),h(s.$$.fragment)},m(e,a){$(n,e,a),d(e,t,a),$(s,e,a),r=!0},p(e,t){const r={};8192&t&&(r.$$scope={dirty:t,ctx:e}),n.$set(r);const a={};8&t&&(a.versions=e[3]),512&t&&(a.runinfo=e[9]),s.$set(a)},i(e){r||(l(n.$$.fragment,e),l(s.$$.fragment,e),r=!0)},o(e){o(n.$$.fragment,e),o(s.$$.fragment,e),r=!1},d(e){e&&a(t),g(n,e),g(s,e)}}}function Ln(e){let n,t,s,r,i,_;function u(n){e[11](n)}let m={logo:e[1],logotext:e[2],entries:e[4],pipeline_name:e[5]};void 0!==e[0]&&(m.isSideNavOpen=e[0]),n=new _n({props:m}),v.push(()=>x(n,"isSideNavOpen",u));let f=e[6]&&e[7].length>0&&Sn(e);return i=new le({props:{$$slots:{default:[Gn]},$$scope:{ctx:e}}}),{c(){h(n.$$.fragment),s=b(),f&&f.c(),r=b(),h(i.$$.fragment)},m(e,t){$(n,e,t),d(e,s,t),f&&f.m(e,t),d(e,r,t),$(i,e,t),_=!0},p(e,[s]){const a={};2&s&&(a.logo=e[1]),4&s&&(a.logotext=e[2]),16&s&&(a.entries=e[4]),32&s&&(a.pipeline_name=e[5]),!t&&1&s&&(t=!0,a.isSideNavOpen=e[0],y(()=>t=!1)),n.$set(a),e[6]&&e[7].length>0?f?(f.p(e,s),192&s&&l(f,1)):(f=Sn(e),f.c(),l(f,1),f.m(r.parentNode,r)):f&&(c(),o(f,1,1,()=>{f=null}),p());const d={};8712&s&&(d.$$scope={dirty:s,ctx:e}),i.$set(d)},i(e){_||(l(n.$$.fragment,e),l(f),l(i.$$.fragment,e),_=!0)},o(e){o(n.$$.fragment,e),o(f),o(i.$$.fragment,e),_=!1},d(e){e&&(a(s),a(r)),g(n,e),f&&f.d(e),g(i,e)}}}function Pn(e,n,t){let s,{$$slots:r={},$$scope:a}=n,{logo:i}=n,{logotext:o=null}=n,{versions:l}=n,{entries:c}=n,{pipeline_name:p}=n,{report_toc:d}=n,{toc:_}=n,{page:u}=n,{isSideNavOpen:f}=n,{runinfo:g}=n;return m(()=>{const e=new URL(window.location.href);let n;n=e.hash.length>0?document.querySelector(`li:has(>[href="${e.hash}"])`):document.querySelector("li:has(>.toc-current-page)"),n&&(n.classList.add("toc-active"),n.scrollIntoView());const t=document.querySelectorAll(".pipen-report-toc-anchor");s=new IntersectionObserver(e=>{e.forEach(e=>{const n=e.target.getAttribute("id"),t=document.querySelector(`li:has(>[href="#${n}"])`);t&&(e.isIntersecting?t.classList.add("toc-active"):t.classList.remove("toc-active"))})},{root:null,rootMargin:"0px",threshold:.5}),t.forEach(e=>{s.observe(e)})}),e.$$set=e=>{"logo"in e&&t(1,i=e.logo),"logotext"in e&&t(2,o=e.logotext),"versions"in e&&t(3,l=e.versions),"entries"in e&&t(4,c=e.entries),"pipeline_name"in e&&t(5,p=e.pipeline_name),"report_toc"in e&&t(6,d=e.report_toc),"toc"in e&&t(7,_=e.toc),"page"in e&&t(8,u=e.page),"isSideNavOpen"in e&&t(0,f=e.isSideNavOpen),"runinfo"in e&&t(9,g=e.runinfo),"$$scope"in e&&t(13,a=e.$$scope)},[f,i,o,l,c,p,d,_,u,g,r,function(e){f=e,t(0,f)},function(e){f=e,t(0,f)},a]}class Tn extends n{constructor(e){super(),t(this,e,Pn,Ln,s,{logo:1,logotext:2,versions:3,entries:4,pipeline_name:5,report_toc:6,toc:7,page:8,isSideNavOpen:0,runinfo:9})}}function An(e){let n;return{c(){n=M(e[2])},m(e,t){d(e,n,t)},p:R,d(e){e&&a(n)}}}function En(e){let n,t,s,r;return t=new ue({props:{icon:e[1],kind:"tertiary",href:e[3](),$$slots:{default:[An]},$$scope:{ctx:e}}}),{c(){n=_("div"),h(t.$$.fragment),u(n,"class",s=`pagenav ${e[0]}`)},m(e,s){d(e,n,s),$(t,n,null),r=!0},p(e,[a]){const i={};16&a&&(i.$$scope={dirty:a,ctx:e}),t.$set(i),(!r||1&a&&s!==(s=`pagenav ${e[0]}`))&&u(n,"class",s)},i(e){r||(l(t.$$.fragment,e),r=!0)},o(e){o(t.$$.fragment,e),r=!1},d(e){e&&a(n),g(t)}}}function In(e,n,t){let{dir:s}=n,r="up"===s?$e:he,a="up"===s?"Go to previous page":"Go to next page";return e.$$set=e=>{"dir"in e&&t(0,s=e.dir)},[s,r,a,()=>{let e=new URL(window.location.href);e.hash="";const n=e.searchParams.get("proc")||"_index",t=parseInt((n.match(/-(\d+)$/)||["-0","0"])[1]),r="up"===s?t-1:t+1,a=n.replace(/-\d+$/,""),i=0===r?a:`${a}-${r}`;return e.searchParams.set("proc",i),e.toString()}]}class Un extends n{constructor(e){super(),t(this,e,In,En,s,{dir:0})}}function kn(e){let n,t;const s=[{tooltipText:e[0]},{icon:e[1]},{direction:"bottom"},{align:"start"},e[3]];let r={};for(let e=0;e<s.length;e+=1)r=A(r,s[e]);return n=new me({props:r}),n.$on("click",e[2]),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},p(e,[t]){const r=11&t?E(s,[1&t&&{tooltipText:e[0]},2&t&&{icon:e[1]},s[2],s[3],8&t&&I(e[3])]):{};n.$set(r)},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function Vn(e,n,t){const s=["src","tip","icon"];let r,a=U(n,s),{src:i=""}=n,{tip:o=""}=n,{icon:l="auto"}=n,c=i.split(".").at(-1).toLocaleLowerCase();if("string"==typeof l)switch("auto"===l&&(l=c),l){case"pdf":case"DocumentPdf":r=Te;break;case"html":case"htm":case"ChartLineData":r=Pe;break;case"png":case"Png":r=Le;break;case"jpg":case"jpeg":case"Jpg":r=Ge;break;case"gif":case"Gif":r=Ne;break;case"tif":case"tiff":case"Tif":r=Re;break;case"eps":case"DatabaseElastic":r=Me;break;case"mp4":case"avi":case"mov":case"mkv":case"webm":case"VideoPlayer":r=we;break;case"Image":r=Se;break;case"Data":case"DataVolume":r=Ce;break;case"Code":case"IbmWatsonxCodeAssistantForZRefactor":r=ye;break;case"svg":case"Svg":r=xe;break;case"zip":case"tar":case"gz":case"bz2":case"7z":case"rar":case"xz":case"Archive":r=ve;break;default:r=be}const p={...a,class:[a.class,"pipen-report-image-download"].filter(Boolean).join(" ")};return e.$$set=e=>{n=A(A({},n),k(e)),t(7,a=U(n,s)),"src"in e&&t(5,i=e.src),"tip"in e&&t(0,o=e.tip),"icon"in e&&t(4,l=e.icon)},[o,r,()=>{const e=document.createElement("a");e.href=i,e.download=i.split("/").pop(),e.click()},p,l,i]}class Bn extends n{constructor(e){super(),t(this,e,Vn,kn,s,{src:5,tip:0,icon:4})}}function Dn(e){let n,t,s;return{c(){n=M(e[1]),t=M("x"),s=M(e[2])},m(e,r){d(e,n,r),d(e,t,r),d(e,s,r)},p(e,t){2&t&&w(n,e[1]),4&t&&w(s,e[2])},d(e){e&&(a(n),a(t),a(s))}}}function On(e){let n,t;return n=new fe({props:{tooltipText:e[0],class:"pipen-report-image-dimension",align:"start",direction:"bottom",$$slots:{default:[Dn]},$$scope:{ctx:e}}}),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},p(e,[t]){const s={};1&t&&(s.tooltipText=e[0]),38&t&&(s.$$scope={dirty:t,ctx:e}),n.$set(s)},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function zn(e,n,t){let{src:s=""}=n,{tip:r=""}=n,a=0,i=0,o=new Image;return o.src=s,o.onload=()=>{t(1,a=o.width),t(2,i=o.height)},e.$$set=e=>{"src"in e&&t(3,s=e.src),"tip"in e&&t(0,r=e.tip)},[r,a,i,s]}class Fn extends n{constructor(e){super(),t(this,e,zn,On,s,{src:3,tip:0})}}const{Boolean:jn}=B;function Hn(e,n,t){const s=e.slice();return s[6]=n[t].src,s[7]=n[t].tip,s[8]=n[t].icon,s[10]=t,s}function qn(e){let n,t,s,u,m=r(e[0]),f=[];for(let n=0;n<m.length;n+=1)f[n]=Xn(Hn(e,m,n));const g=e=>o(f[e],1,1,()=>{f[e]=null});let $=[e[2],{role:"button"},{tabindex:"0"}],h={};for(let e=0;e<$.length;e+=1)h=A(h,$[e]);return{c(){n=_("div");for(let e=0;e<f.length;e+=1)f[e].c();D(n,h),S(n,"svelte-14lcnln",!0)},m(r,a){d(r,n,a);for(let e=0;e<f.length;e+=1)f[e]&&f[e].m(n,null);t=!0,s||(u=N(n,"mouseenter",e[4]),s=!0)},p(e,t){if(3&t){let s;for(m=r(e[0]),s=0;s<m.length;s+=1){const r=Hn(e,m,s);f[s]?(f[s].p(r,t),l(f[s],1)):(f[s]=Xn(r),f[s].c(),l(f[s],1),f[s].m(n,null))}for(c(),s=m.length;s<f.length;s+=1)g(s);p()}S(n,"svelte-14lcnln",!0)},i(e){if(!t){for(let e=0;e<m.length;e+=1)l(f[e]);t=!0}},o(e){f=f.filter(jn);for(let e=0;e<f.length;e+=1)o(f[e]);t=!1},d(e){e&&a(n),i(f,e),s=!1,u()}}}function Kn(e){let n,t;return n=new Bn({props:{src:e[6],tip:e[7],icon:e[8]}}),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},p(e,t){const s={};1&t&&(s.src=e[6]),1&t&&(s.tip=e[7]),1&t&&(s.icon=e[8]),n.$set(s)},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function Wn(e){let n,t;return n=new Fn({props:{src:e[1],tip:e[7]}}),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},p(e,t){const s={};2&t&&(s.src=e[1]),1&t&&(s.tip=e[7]),n.$set(s)},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function Xn(e){let n,t,s,r;const i=[Wn,Kn],_=[];function u(e,n){return"x"===e[6]?0:1}return n=u(e),t=_[n]=i[n](e),{c(){t.c(),s=f()},m(e,t){_[n].m(e,t),d(e,s,t),r=!0},p(e,r){let a=n;n=u(e),n===a?_[n].p(e,r):(c(),o(_[a],1,1,()=>{_[a]=null}),p(),t=_[n],t?t.p(e,r):(t=_[n]=i[n](e),t.c()),l(t,1),t.m(s.parentNode,s))},i(e){r||(l(t),r=!0)},o(e){o(t),r=!1},d(e){e&&a(s),_[n].d(e)}}}function Yn(e){let n,t,s=e[0].length>0&&qn(e);return{c(){s&&s.c(),n=f()},m(e,r){s&&s.m(e,r),d(e,n,r),t=!0},p(e,[t]){e[0].length>0?s?(s.p(e,t),1&t&&l(s,1)):(s=qn(e),s.c(),l(s,1),s.m(n.parentNode,n)):s&&(c(),o(s,1,1,()=>{s=null}),p())},i(e){t||(l(s),t=!0)},o(e){o(s),t=!1},d(e){e&&a(n),s&&s.d(e)}}}function Zn(e,n,t){const s=["data","dsrc"];let r=U(n,s),{data:a=[]}=n,{dsrc:i=""}=n;Array.isArray(a)||(a=[a]),a=[{src:"x",tip:"widthxheight"},...a],a=a.map(e=>"string"==typeof e?{src:e,tip:`Download the ${e.split(".").at(-1)} format.`,icon:"auto"}:(e.tip=e.tip||`Download the ${e.src.split(".").at(-1)} format.`,e.icon=e.icon||"auto",e));const o={...r,class:[r.class,"pipen-report-image-download-list"].filter(Boolean).join(" ")},l=V();return e.$$set=e=>{n=A(A({},n),k(e)),t(5,r=U(n,s)),"data"in e&&t(0,a=e.data),"dsrc"in e&&t(1,i=e.dsrc)},[a,i,o,l,()=>l("mouseenter")]}class Jn extends n{constructor(e){super(),t(this,e,Zn,Yn,s,{data:0,dsrc:1})}}const Qn=e=>({}),et=e=>({}),nt=e=>({}),tt=e=>({}),st=e=>({}),rt=e=>({}),at=e=>({}),it=e=>({});function ot(e){let n,t;return n=new ge({props:{ratio:e[6],$$slots:{default:[ut]},$$scope:{ctx:e}}}),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},p(e,t){const s={};64&t&&(s.ratio=e[6]),33556159&t&&(s.$$scope={dirty:t,ctx:e}),n.$set(s)},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function lt(e){let n,t,s,r,i=e[2]&&mt(e),_=e[0]&&ft(e),u=e[1]&&$t(e);return{c(){i&&i.c(),n=b(),_&&_.c(),t=b(),u&&u.c(),s=f()},m(e,a){i&&i.m(e,a),d(e,n,a),_&&_.m(e,a),d(e,t,a),u&&u.m(e,a),d(e,s,a),r=!0},p(e,r){e[2]?i?(i.p(e,r),4&r&&l(i,1)):(i=mt(e),i.c(),l(i,1),i.m(n.parentNode,n)):i&&(c(),o(i,1,1,()=>{i=null}),p()),e[0]?_?(_.p(e,r),1&r&&l(_,1)):(_=ft(e),_.c(),l(_,1),_.m(t.parentNode,t)):_&&(c(),o(_,1,1,()=>{_=null}),p()),e[1]?u?(u.p(e,r),2&r&&l(u,1)):(u=$t(e),u.c(),l(u,1),u.m(s.parentNode,s)):u&&(c(),o(u,1,1,()=>{u=null}),p())},i(e){r||(l(i),l(_),l(u),r=!0)},o(e){o(i),o(_),o(u),r=!1},d(e){e&&(a(n),a(t),a(s)),i&&i.d(e),_&&_.d(e),u&&u.d(e)}}}function ct(e){let n;const t=e[15].loading,s=G(t,e,e[25],tt);return{c(){s&&s.c()},m(e,t){s&&s.m(e,t),n=!0},p(e,r){s&&s.p&&(!n||33554432&r)&&L(s,t,e,e[25],n?T(t,e[25],r,nt):P(e[25]),tt)},i(e){n||(l(s,e),n=!0)},o(e){o(s,e),n=!1},d(e){s&&s.d(e)}}}function pt(e){let n,t,s,r,i,u,m,g,$=[e[12],{src:t=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],h={};for(let e=0;e<$.length;e+=1)h=A(h,$[e]);let v=e[9]&&dt(e);return{c(){n=_("img"),r=b(),v&&v.c(),i=f(),D(n,h)},m(t,s){d(t,n,s),d(t,r,s),v&&v.m(t,s),d(t,i,s),u=!0,m||(g=[N(n,"click",e[11]),N(n,"mouseenter",e[19]),N(n,"mouseleave",e[20])],m=!0)},p(s,r){D(n,h=E($,[(e=s)[12],(!u||8&r&&!j(n.src,t=e[3]))&&{src:t},(!u||16&r)&&{alt:e[4]},{title:"Click to zoom in ..."}])),e[9]?v?(v.p(e,r),512&r&&l(v,1)):(v=dt(e),v.c(),l(v,1),v.m(i.parentNode,i)):v&&(c(),o(v,1,1,()=>{v=null}),p())},i(t){u||(t&&F(()=>{u&&(s||(s=O(n,z,{duration:e[7]?vt:0},!0)),s.run(1))}),l(v),u=!0)},o(t){t&&(s||(s=O(n,z,{duration:e[7]?vt:0},!1)),s.run(0)),o(v),u=!1},d(e){e&&(a(n),a(r),a(i)),e&&s&&s.end(),v&&v.d(e),m=!1,H(g)}}}function dt(e){let n,t;return n=new Jn({props:{data:e[5]}}),n.$on("mouseenter",e[21]),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},p(e,t){const s={};32&t&&(s.data=e[5]),n.$set(s)},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function _t(e){let n;const t=e[15].error,s=G(t,e,e[25],et);return{c(){s&&s.c()},m(e,t){s&&s.m(e,t),n=!0},p(e,r){s&&s.p&&(!n||33554432&r)&&L(s,t,e,e[25],n?T(t,e[25],r,Qn):P(e[25]),et)},i(e){n||(l(s,e),n=!0)},o(e){o(s,e),n=!1},d(e){s&&s.d(e)}}}function ut(e){let n,t,s,r,i=e[2]&&ct(e),_=e[0]&&pt(e),u=e[1]&&_t(e);return{c(){i&&i.c(),n=b(),_&&_.c(),t=b(),u&&u.c(),s=f()},m(e,a){i&&i.m(e,a),d(e,n,a),_&&_.m(e,a),d(e,t,a),u&&u.m(e,a),d(e,s,a),r=!0},p(e,r){e[2]?i?(i.p(e,r),4&r&&l(i,1)):(i=ct(e),i.c(),l(i,1),i.m(n.parentNode,n)):i&&(c(),o(i,1,1,()=>{i=null}),p()),e[0]?_?(_.p(e,r),1&r&&l(_,1)):(_=pt(e),_.c(),l(_,1),_.m(t.parentNode,t)):_&&(c(),o(_,1,1,()=>{_=null}),p()),e[1]?u?(u.p(e,r),2&r&&l(u,1)):(u=_t(e),u.c(),l(u,1),u.m(s.parentNode,s)):u&&(c(),o(u,1,1,()=>{u=null}),p())},i(e){r||(l(i),l(_),l(u),r=!0)},o(e){o(i),o(_),o(u),r=!1},d(e){e&&(a(n),a(t),a(s)),i&&i.d(e),_&&_.d(e),u&&u.d(e)}}}function mt(e){let n;const t=e[15].loading,s=G(t,e,e[25],it);return{c(){s&&s.c()},m(e,t){s&&s.m(e,t),n=!0},p(e,r){s&&s.p&&(!n||33554432&r)&&L(s,t,e,e[25],n?T(t,e[25],r,at):P(e[25]),it)},i(e){n||(l(s,e),n=!0)},o(e){o(s,e),n=!1},d(e){s&&s.d(e)}}}function ft(e){let n,t,s,r,i,u,m,g,$=[e[12],{src:t=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],h={};for(let e=0;e<$.length;e+=1)h=A(h,$[e]);let v=e[9]&&gt(e);return{c(){n=_("img"),r=b(),v&&v.c(),i=f(),D(n,h)},m(t,s){d(t,n,s),d(t,r,s),v&&v.m(t,s),d(t,i,s),u=!0,m||(g=[N(n,"click",e[11]),N(n,"mouseenter",e[16]),N(n,"mouseleave",e[17])],m=!0)},p(s,r){D(n,h=E($,[(e=s)[12],(!u||8&r&&!j(n.src,t=e[3]))&&{src:t},(!u||16&r)&&{alt:e[4]},{title:"Click to zoom in ..."}])),e[9]?v?(v.p(e,r),512&r&&l(v,1)):(v=gt(e),v.c(),l(v,1),v.m(i.parentNode,i)):v&&(c(),o(v,1,1,()=>{v=null}),p())},i(t){u||(t&&F(()=>{u&&(s||(s=O(n,z,{duration:e[7]?vt:0},!0)),s.run(1))}),l(v),u=!0)},o(t){t&&(s||(s=O(n,z,{duration:e[7]?vt:0},!1)),s.run(0)),o(v),u=!1},d(e){e&&(a(n),a(r),a(i)),e&&s&&s.end(),v&&v.d(e),m=!1,H(g)}}}function gt(e){let n,t;return n=new Jn({props:{dsrc:e[3],data:e[5]}}),n.$on("mouseenter",e[18]),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},p(e,t){const s={};8&t&&(s.dsrc=e[3]),32&t&&(s.data=e[5]),n.$set(s)},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function $t(e){let n;const t=e[15].error,s=G(t,e,e[25],rt);return{c(){s&&s.c()},m(e,t){s&&s.m(e,t),n=!0},p(e,r){s&&s.p&&(!n||33554432&r)&&L(s,t,e,e[25],n?T(t,e[25],r,st):P(e[25]),rt)},i(e){n||(l(s,e),n=!0)},o(e){o(s,e),n=!1},d(e){s&&s.d(e)}}}function ht(e){let n,t,s,r;return{c(){n=_("img"),u(n,"class","pipen-report-image-lg"),j(n.src,t=e[3])||u(n,"src",t),u(n,"alt",e[4])},m(e,t){d(e,n,t),r=!0},p(s,a){e=s,(!r||8&a&&!j(n.src,t=e[3]))&&u(n,"src",t),(!r||16&a)&&u(n,"alt",e[4])},i(t){r||(t&&F(()=>{r&&(s||(s=O(n,z,{duration:e[7]?vt:0},!0)),s.run(1))}),r=!0)},o(t){t&&(s||(s=O(n,z,{duration:e[7]?vt:0},!1)),s.run(0)),r=!1},d(e){e&&a(n),e&&s&&s.end()}}}function bt(e){let n,t,s,r,i,_;const u=[lt,ot],m=[];function f(e,n){return void 0===e[6]?0:1}function C(n){e[22](n)}n=f(e),t=m[n]=u[n](e);let S={modalHeading:e[4],primaryButtonText:"Open in new window ...",secondaryButtonText:"Close",$$slots:{default:[ht]},$$scope:{ctx:e}};return void 0!==e[8]&&(S.open=e[8]),r=new te({props:S}),v.push(()=>x(r,"open",C)),r.$on("click:button--secondary",e[23]),r.$on("click:button--primary",e[24]),{c(){t.c(),s=b(),h(r.$$.fragment)},m(e,t){m[n].m(e,t),d(e,s,t),$(r,e,t),_=!0},p(e,[a]){let d=n;n=f(e),n===d?m[n].p(e,a):(c(),o(m[d],1,1,()=>{m[d]=null}),p(),t=m[n],t?t.p(e,a):(t=m[n]=u[n](e),t.c()),l(t,1),t.m(s.parentNode,s));const _={};16&a&&(_.modalHeading=e[4]),33554584&a&&(_.$$scope={dirty:a,ctx:e}),!i&&256&a&&(i=!0,_.open=e[8],y(()=>i=!1)),r.$set(_)},i(e){_||(l(t),l(r.$$.fragment,e),_=!0)},o(e){o(t),o(r.$$.fragment,e),_=!1},d(e){e&&a(s),m[n].d(e),g(r,e)}}}const vt=110;function xt(e,n,t){const s=["src","alt","download","ratio","loading","loaded","error","fadeIn","border","loadImage"];let r=U(n,s),{$$slots:a={},$$scope:i}=n,{src:o=""}=n,{alt:l=""}=n,{download:c=[]}=n,{ratio:p}=n,{loading:d=!1}=n,{loaded:_=!1}=n,{error:u=!1}=n,{fadeIn:f=!1}=n,{border:g=!0}=n,$=!1,h=!1;const b=e=>{null!=y&&(y=null),t(0,_=!1),t(1,u=!1),y=new Image,y.src=e||o,y.onload=()=>t(0,_=!0),y.onerror=()=>t(1,u=!0)},v=V();let x,y=null;m(()=>()=>y=null);const C={...r,class:[r.class,"pipen-report-image",g?"":"pipen-report-image-noborder"].filter(Boolean).join(" ")};return e.$$set=e=>{n=A(A({},n),k(e)),t(28,r=U(n,s)),"src"in e&&t(3,o=e.src),"alt"in e&&t(4,l=e.alt),"download"in e&&t(5,c=e.download),"ratio"in e&&t(6,p=e.ratio),"loading"in e&&t(2,d=e.loading),"loaded"in e&&t(0,_=e.loaded),"error"in e&&t(1,u=e.error),"fadeIn"in e&&t(7,f=e.fadeIn),"border"in e&&t(13,g=e.border),"$$scope"in e&&t(25,i=e.$$scope)},e.$$.update=()=>{3&e.$$.dirty&&t(2,d=!_&&!u),8&e.$$.dirty&&o&&"undefined"!=typeof window&&b(),1&e.$$.dirty&&_&&v("load"),2&e.$$.dirty&&u&&v("error")},[_,u,d,o,l,c,p,f,$,h,x,()=>{t(8,$=!0)},C,g,b,a,()=>{t(9,h=!0)},()=>{t(10,x=setTimeout(()=>{t(9,h=!1)},100))},()=>{clearTimeout(x),t(9,h=!0)},()=>{t(9,h=!0)},()=>{t(10,x=setTimeout(()=>{t(9,h=!1)},100))},()=>{clearTimeout(x),t(9,h=!0)},function(e){$=e,t(8,$)},()=>{t(8,$=!1)},()=>{t(8,$=!1),window.open(o)},i]}class yt extends n{constructor(e){super(),t(this,e,xt,bt,s,{src:3,alt:4,download:5,ratio:6,loading:2,loaded:0,error:1,fadeIn:7,border:13,loadImage:14})}get loadImage(){return this.$$.ctx[14]}}function Ct(e){let n,t,s=[e[2]],r={};for(let e=0;e<s.length;e+=1)r=A(r,s[e]);return{c(){n=_("div"),t=M("Image loading ..."),D(n,r)},m(e,s){d(e,n,s),C(n,t)},p:R,d(e){e&&a(n)}}}function St(e){let n,t,s=[e[3]],r={};for(let e=0;e<s.length;e+=1)r=A(r,s[e]);return{c(){n=_("div"),t=M("Image loading error!"),D(n,r)},m(e,s){d(e,n,s),C(n,t)},p:R,d(e){e&&a(n)}}}function wt(e){let n,t,s;const r=[{src:e[1]},e[4]];let i={$$slots:{error:[St],loading:[Ct]},$$scope:{ctx:e}};for(let e=0;e<r.length;e+=1)i=A(i,r[e]);t=new yt({props:i});let c=[e[0]],p={};for(let e=0;e<c.length;e+=1)p=A(p,c[e]);return{c(){n=_("div"),h(t.$$.fragment),D(n,p)},m(e,r){d(e,n,r),$(t,n,null),s=!0},p(e,[s]){const a=18&s?E(r,[2&s&&{src:e[1]},16&s&&I(e[4])]):{};256&s&&(a.$$scope={dirty:s,ctx:e}),t.$set(a),D(n,p=E(c,[1&s&&e[0]]))},i(e){s||(l(t.$$.fragment,e),s=!0)},o(e){o(t.$$.fragment,e),s=!1},d(e){e&&a(n),g(t)}}}function Mt(e,n,t){const s=["frameProps","width","height","src"];let r=U(n,s),{frameProps:a={class:"pipen-report-image-frame"}}=n,{width:i=300}=n,{height:o=200}=n,{src:l}=n;a||(a={"style:position":"relative"});let c=`width: ${i}px; height: ${o}px; aspect-ratio: ${i} / ${o}`,p={...r,class:[r.class,"pipen-report-image-loading"].filter(Boolean).join(" "),style:r.style?`${r.style}; ${c}`:c},d={...p,class:[p.class,"error"].join(" ")};return e.$$set=e=>{n=A(A({},n),k(e)),t(4,r=U(n,s)),"frameProps"in e&&t(0,a=e.frameProps),"width"in e&&t(5,i=e.width),"height"in e&&t(6,o=e.height),"src"in e&&t(1,l=e.src)},[a,l,p,d,r,i,o]}let Rt=class extends n{constructor(e){super(),t(this,e,Mt,wt,s,{frameProps:0,width:5,height:6,src:1})}};function Nt(e){let n,t;return{c(){n=_("div"),t=M(e[0]),u(n,"class","pipen-report-descr-title svelte-1xzxuva")},m(e,s){d(e,n,s),C(n,t)},p(e,n){1&n&&w(t,e[0])},d(e){e&&a(n)}}}function Gt(e){let n,t,s,r=e[0]&&Nt(e);const i=e[3].default,c=G(i,e,e[2],null);let p=[e[1]],u={};for(let e=0;e<p.length;e+=1)u=A(u,p[e]);return{c(){n=_("div"),r&&r.c(),t=b(),c&&c.c(),D(n,u),S(n,"svelte-1xzxuva",!0)},m(e,a){d(e,n,a),r&&r.m(n,null),C(n,t),c&&c.m(n,null),s=!0},p(e,[a]){e[0]?r?r.p(e,a):(r=Nt(e),r.c(),r.m(n,t)):r&&(r.d(1),r=null),c&&c.p&&(!s||4&a)&&L(c,i,e,e[2],s?T(i,e[2],a,null):P(e[2]),null),D(n,u=E(p,[2&a&&e[1]])),S(n,"svelte-1xzxuva",!0)},i(e){s||(l(c,e),s=!0)},o(e){o(c,e),s=!1},d(e){e&&a(n),r&&r.d(),c&&c.d(e)}}}function Lt(e,n,t){const s=["title"];let r=U(n,s),{$$slots:a={},$$scope:i}=n,{title:o}=n,{klass:l,...c}=r;return l=l?`pipen-report-descr ${l}`:"pipen-report-descr",c||(c={}),c.class=l,e.$$set=e=>{n=A(A({},n),k(e)),t(5,r=U(n,s)),"title"in e&&t(0,o=e.title),"$$scope"in e&&t(2,i=e.$$scope)},[o,c,i,a]}class Pt extends n{constructor(e){super(),t(this,e,Lt,Gt,s,{title:0})}}function Tt(e){let n;return{c(){n=M("The plot shows the distribution or pattern of the specified features (TCF7, SELL, LEF1, CCR7, IL2, GZMA, GNLY, PRF1, GZMB, GZMK, IFNG, NKG7, TNF, CX3CR1, GZMH, FGFBP2, CD28, TNFRSF14, ICOS, TNFRSF9, LAG3, TIGIT, PDCD1, HAVCR2, CTLA4, EOMES, HOPX, TBX21, ZEB2, HIF1A, ID2, TOX, IL2RA, FOXP3, IKZF2, CD38, ENTPD1, ITGAE, CD4, CD8A) across cells. The plot type is 'violin', showing the distribution of feature values by the grouping variables.")},m(e,t){d(e,n,t)},d(e){e&&a(n)}}}function At(e){let n;return{c(){n=M("The plot shows the distribution or pattern of the specified features (Cytotoxicity, Exhaustion, NaiveScore, Treg) across cells. The plot type is 'dot', showing aggregated feature values by the grouping variables.")},m(e,t){d(e,n,t)},d(e){e&&a(n)}}}function Et(e){let n,t,s,r,i,c,p,m,f,v,x,y,C,S;return r=new Pt({props:{$$slots:{default:[Tt]},$$scope:{ctx:e}}}),c=new Rt({props:{src:"../SeuratClusterStats/samples.scRep.cluster_stats/features/T-cell-function-associated-genes-violin-plots-.png",width:3886,height:1800}}),x=new Pt({props:{$$slots:{default:[At]},$$scope:{ctx:e}}}),C=new Rt({props:{src:"../SeuratClusterStats/samples.scRep.cluster_stats/features/Gene-signatures.png",width:897,height:400}}),{c(){n=_("h1"),n.textContent="T cell function-associated genes (violin plots)",t=_("a"),t.innerHTML="",s=b(),h(r.$$.fragment),i=b(),h(c.$$.fragment),p=b(),m=_("h1"),m.textContent="Gene signatures",f=_("a"),f.innerHTML="",v=b(),h(x.$$.fragment),y=b(),h(C.$$.fragment),u(t,"id","prt-h1-8-t-cell-function-associated-genes-violin-plots"),u(t,"class","pipen-report-toc-anchor"),u(f,"id","prt-h1-9-gene-signatures"),u(f,"class","pipen-report-toc-anchor")},m(e,a){d(e,n,a),d(e,t,a),d(e,s,a),$(r,e,a),d(e,i,a),$(c,e,a),d(e,p,a),d(e,m,a),d(e,f,a),d(e,v,a),$(x,e,a),d(e,y,a),$(C,e,a),S=!0},p(e,[n]){const t={};1&n&&(t.$$scope={dirty:n,ctx:e}),r.$set(t);const s={};1&n&&(s.$$scope={dirty:n,ctx:e}),x.$set(s)},i(e){S||(l(r.$$.fragment,e),l(c.$$.fragment,e),l(x.$$.fragment,e),l(C.$$.fragment,e),S=!0)},o(e){o(r.$$.fragment,e),o(c.$$.fragment,e),o(x.$$.fragment,e),o(C.$$.fragment,e),S=!1},d(e){e&&(a(n),a(t),a(s),a(i),a(p),a(m),a(f),a(v),a(y)),g(r,e),g(c,e),g(x,e),g(C,e)}}}class It extends n{constructor(e){super(),t(this,e,null,Et,s,{})}}var Ut=[{slug:"prt-h1-0-dimensional-reduction-plot",text:"Dimensional reduction plot",children:[],page:0},{slug:"prt-h1-1-vdj-presence",text:"VDJ Presence",children:[],page:0},{slug:"prt-h1-2-umap",text:"UMAP",children:[],page:0},{slug:"prt-h1-3-number-of-cells-in-each-cluster-bar-chart",text:"Number of cells in each cluster (Bar Chart)",children:[],page:0},{slug:"prt-h1-4-number-of-cells-in-each-cluster-by-sample-bar-chart",text:"Number of cells in each cluster by Sample (Bar Chart)",children:[],page:0},{slug:"prt-h1-5-number-of-genes-expressed-in-each-cluster",text:"Number of genes expressed in each cluster",children:[],page:0},{slug:"prt-h1-6-t-cell-function-associated-genes",text:"T cell function-associated genes",children:[],page:0},{slug:"prt-h1-7-t-cell-function-associated-genes-feature-plots",text:"T cell function-associated genes (feature plots)",children:[],page:0},{slug:"prt-h1-8-t-cell-function-associated-genes-violin-plots",text:"T cell function-associated genes (violin plots)",children:[],page:1},{slug:"prt-h1-9-gene-signatures",text:"Gene signatures",children:[],page:1}],kt={pipeline:{name:"Immunopipe"},versions:'\n    <a href="https://github.com/pwwang/pipen" target="_blank">\n        pipen v1.1.8\n    </a> and\n    <a href="https://github.com/pwwang/pipen-report" target="_blank">\n        pipen-report v1.1.0\n    </a>\n',entries:[{name:"SampleInfo",desc:"List sample information and perform statistics",npages:1,report_toc:!0,order:0,runinfo:{session:"# Generated by pipen_runinfo v%(version)s\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 12 (bookworm)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.30.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n[1] plotthis_0.10.0      biopipen.utils_0.3.5 gglogger_0.1.7      \n[4] dplyr_1.1.4          rlang_1.1.6         \n\nloaded via a namespace (and not attached):\n  [1] cubature_2.0.4.6            RcppAnnoy_0.0.22           \n  [3] splines_4.3.3               later_1.4.4                \n  [5] bitops_1.0-9                tibble_3.3.0               \n  [7] polyclip_1.10-7             enrichit_0.0.4             \n  [9] fastDummies_1.7.5           lifecycle_1.0.4            \n [11] globals_0.18.0              lattice_0.22-7             \n [13] MASS_7.3-60.0.1             magrittr_2.0.3             \n [15] plotly_4.11.0               httpuv_1.6.16              \n [17] Seurat_5.3.0                sctransform_0.4.2          \n [19] spam_2.11-1                 sp_2.2-0                   \n [21] spatstat.sparse_3.1-0       reticulate_1.43.0          \n [23] cowplot_1.2.0               pbapply_1.7-4              \n [25] RColorBrewer_1.1-3          abind_1.4-5                \n [27] zlibbioc_1.48.0             Rtsne_0.17                 \n [29] GenomicRanges_1.54.1        purrr_1.1.0                \n [31] ggraph_2.2.1                BiocGenerics_0.48.1        \n [33] RCurl_1.98-1.17             tweenr_2.0.3               \n [35] evmix_2.12                  circlize_0.4.16            \n [37] GenomeInfoDbData_1.2.11     IRanges_2.36.0             \n [39] S4Vectors_0.40.2            ggrepel_0.9.6              \n [41] irlba_2.3.5.1               listenv_0.9.1              \n [43] spatstat.utils_3.1-5        iNEXT_3.0.2                \n [45] MatrixModels_0.5-4          goftest_1.2-3              \n [47] RSpectra_0.16-2             scRepertoire_2.2.1         \n [49] spatstat.random_3.4-1       fitdistrplus_1.2-4         \n [51] parallelly_1.45.1           codetools_0.2-20           \n [53] DelayedArray_0.28.0         xml2_1.4.0                 \n [55] ggforce_0.5.0               tidyselect_1.2.1           \n [57] shape_1.4.6.1               farver_2.1.2               \n [59] viridis_0.6.5               matrixStats_1.5.0          \n [61] stats4_4.3.3                spatstat.explore_3.5-2     \n [63] jsonlite_2.0.0              ellipsis_0.3.2             \n [65] tidygraph_1.3.0             progressr_0.15.1           \n [67] ggridges_0.5.7              ggalluvial_0.12.5          \n [69] survival_3.8-3              tools_4.3.3                \n [71] ggnewscale_0.5.2            stringdist_0.9.15          \n [73] ica_1.0-3                   Rcpp_1.1.0                 \n [75] glue_1.8.0                  gridExtra_2.3              \n [77] SparseArray_1.2.2           qvalue_2.34.0              \n [79] MatrixGenerics_1.14.0       GenomeInfoDb_1.38.1        \n [81] withr_3.0.2                 fastmap_1.2.0              \n [83] fansi_1.0.6                 SparseM_1.84-2             \n [85] ttservice_0.5.3             digest_0.6.37              \n [87] R6_2.6.1                    mime_0.13                  \n [89] colorspace_2.1-1            scattermore_1.2            \n [91] tensor_1.5.1                dichromat_2.0-0.1          \n [93] spatstat.data_3.1-8         tidyr_1.3.1                \n [95] generics_0.1.4              data.table_1.17.8          \n [97] graphlayouts_1.2.2          httr_1.4.7                 \n [99] htmlwidgets_1.6.4           S4Arrays_1.2.0             \n[101] uwot_0.2.3                  pkgconfig_2.0.3            \n[103] gtable_0.3.6                lmtest_0.9-40              \n[105] SingleCellExperiment_1.24.0 XVector_0.42.0             \n[107] htmltools_0.5.8.1           dotCall64_1.2              \n[109] SeuratObject_5.2.0          scales_1.4.0               \n[111] Biobase_2.62.0              png_0.1-8                  \n[113] spatstat.univar_3.1-4       ggdendro_0.2.0             \n[115] rjson_0.2.23                reshape2_1.4.4             \n[117] nlme_3.1-168                cachem_1.1.0               \n[119] zoo_1.8-14                  GlobalOptions_0.1.2        \n[121] stringr_1.5.2               KernSmooth_2.23-26         \n[123] parallel_4.3.3              miniUI_0.1.2               \n[125] pillar_1.11.0               grid_4.3.3                 \n[127] logger_0.4.0                vctrs_0.6.5                \n[129] RANN_2.6.2                  VGAM_1.1-13                \n[131] promises_1.3.3              xtable_1.8-4               \n[133] cluster_2.1.8.1             truncdist_1.0-2            \n[135] cli_3.6.5                   compiler_4.3.3             \n[137] crayon_1.5.3                future.apply_1.20.0        \n[139] plyr_1.8.9                  forcats_1.0.0              \n[141] stringi_1.8.7               viridisLite_0.4.2          \n[143] tidyseurat_0.8.0            deldir_2.0-4               \n[145] scplotter_0.6.5             assertthat_0.2.1           \n[147] gsl_2.1-8                   lazyeval_0.2.2             \n[149] spatstat.geom_3.5-0         quantreg_6.1               \n[151] Matrix_1.6-5                RcppHNSW_0.6.0             \n[153] patchwork_1.3.2             future_1.67.0              \n[155] ggplot2_3.5.2               shiny_1.11.1               \n[157] SummarizedExperiment_1.32.0 evd_2.3-7.1                \n[159] ROCR_1.0-11                 gridtext_0.1.5             \n[161] memoise_2.0.1               igraph_2.1.4               \n[163] bracer_1.2.1               \n",time:"# Generated by pipen-runinfo v1.1.2\n\nCommand: stdbuf -oL Rscript /mnt/disks/.cwd/.pipen/Immunopipe/SampleInfo/0/job.script\nVoluntary context switches: 362\nInvoluntary context switches: 180\nPercentage of CPU this job got: 159%\nMajor page faults: 0\nMinor page faults: 202684\nMaximum resident set size (kB): 813032\nElapsed real time (s): 8.92\nSystem (kernel) time (s): 0.47\nUser time (s): 13.79\nExit status: 0\n",device:"# Generated by pipen-runinfo v1.1.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nf523046eea1a\n\nCPU\n----\nArchitecture:                            x86_64\nCPU op-mode(s):                          32-bit, 64-bit\nAddress sizes:                           46 bits physical, 48 bits virtual\nByte Order:                              Little Endian\nCPU(s):                                  64\nOn-line CPU(s) list:                     0-63\nVendor ID:                               GenuineIntel\nModel name:                              Intel(R) Xeon(R) CPU @ 2.80GHz\nCPU family:                              6\nModel:                                   85\nThread(s) per core:                      2\nCore(s) per socket:                      16\nSocket(s):                               2\nStepping:                                7\nBogoMIPS:                                5599.99\nFlags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch ssbd ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves arat avx512_vnni md_clear arch_capabilities\nHypervisor vendor:                       KVM\nVirtualization type:                     full\nL1d cache:                               1 MiB (32 instances)\nL1i cache:                               1 MiB (32 instances)\nL2 cache:                                32 MiB (32 instances)\nL3 cache:                                66 MiB (2 instances)\nNUMA node(s):                            2\nNUMA node0 CPU(s):                       0-15,32-47\nNUMA node1 CPU(s):                       16-31,48-63\nVulnerability Gather data sampling:      Not affected\nVulnerability Indirect target selection: Mitigation; Aligned branch/return thunks\nVulnerability Itlb multihit:             Not affected\nVulnerability L1tf:                      Not affected\nVulnerability Mds:                       Not affected\nVulnerability Meltdown:                  Not affected\nVulnerability Mmio stale data:           Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown\nVulnerability Reg file data sampling:    Not affected\nVulnerability Retbleed:                  Mitigation; Enhanced IBRS\nVulnerability Spec rstack overflow:      Not affected\nVulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:                Mitigation; Enhanced / Automatic IBRS; IBPB conditional; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                     Not affected\nVulnerability Tsa:                       Not affected\nVulnerability Tsx async abort:           Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown\nVulnerability Vmscape:                   Not affected\n\nMemory\n------\n               total        used        free      shared  buff/cache   available\nMem:           503Gi       4.7Gi       494Gi       704Ki       7.1Gi       499Gi\nSwap:             0B          0B          0B\n\nDisk\n----\nFilesystem                                     Size  Used Avail Use% Mounted on\noverlay                                        248G  8.8G  240G   4% /\ntmpfs                                           64M     0   64M   0% /dev\nshm                                             64M     0   64M   0% /dev/shm\n/dev/sda1                                      248G  8.8G  240G   4% /etc/hosts\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/xqute_workdir\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/editable\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/.cwd\ntmpfs                                          252G     0  252G   0% /proc/acpi\ntmpfs                                          252G     0  252G   0% /proc/scsi\ntmpfs                                          252G     0  252G   0% /sys/firmware\n\nNetwork\n-------\nNeither ifconfig nor ip is available.\n\nGPU\n---\nnvidia-smi is not available.\n\n"}},{name:"SeuratPreparing",desc:"Load, prepare and apply QC to data, using `Seurat`",npages:1,report_toc:!0,order:2,runinfo:{session:"# Generated by pipen_runinfo v%(version)s\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 12 (bookworm)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.30.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n[1] biopipen.utils_0.3.5 glue_1.8.0           dplyr_1.1.4         \n[4] bracer_1.2.1         future_1.67.0        Seurat_5.3.0        \n[7] SeuratObject_5.2.0   sp_2.2-0            \n\nloaded via a namespace (and not attached):\n  [1] cubature_2.0.4.6            RcppAnnoy_0.0.22           \n  [3] splines_4.3.3               later_1.4.4                \n  [5] bitops_1.0-9                R.oo_1.27.1                \n  [7] tibble_3.3.0                polyclip_1.10-7            \n  [9] enrichit_0.0.4              fastDummies_1.7.5          \n [11] lifecycle_1.0.4             globals_0.18.0             \n [13] lattice_0.22-7              MASS_7.3-60.0.1            \n [15] magrittr_2.0.3              plotly_4.11.0              \n [17] httpuv_1.6.16               sctransform_0.4.2          \n [19] spam_2.11-1                 spatstat.sparse_3.1-0      \n [21] reticulate_1.43.0           cowplot_1.2.0              \n [23] pbapply_1.7-4               RColorBrewer_1.1-3         \n [25] abind_1.4-5                 zlibbioc_1.48.0            \n [27] Rtsne_0.17                  GenomicRanges_1.54.1       \n [29] R.utils_2.13.0              purrr_1.1.0                \n [31] ggraph_2.2.1                BiocGenerics_0.48.1        \n [33] RCurl_1.98-1.17             tweenr_2.0.3               \n [35] evmix_2.12                  circlize_0.4.16            \n [37] GenomeInfoDbData_1.2.11     IRanges_2.36.0             \n [39] S4Vectors_0.40.2            ggrepel_0.9.6              \n [41] irlba_2.3.5.1               listenv_0.9.1              \n [43] spatstat.utils_3.1-5        iNEXT_3.0.2                \n [45] MatrixModels_0.5-4          goftest_1.2-3              \n [47] RSpectra_0.16-2             scRepertoire_2.2.1         \n [49] spatstat.random_3.4-1       fitdistrplus_1.2-4         \n [51] parallelly_1.45.1           codetools_0.2-20           \n [53] DelayedArray_0.28.0         xml2_1.4.0                 \n [55] ggforce_0.5.0               tidyselect_1.2.1           \n [57] shape_1.4.6.1               farver_2.1.2               \n [59] viridis_0.6.5               matrixStats_1.5.0          \n [61] stats4_4.3.3                spatstat.explore_3.5-2     \n [63] jsonlite_2.0.0              ellipsis_0.3.2             \n [65] tidygraph_1.3.0             progressr_0.15.1           \n [67] ggridges_0.5.7              ggalluvial_0.12.5          \n [69] survival_3.8-3              systemfonts_1.2.3          \n [71] tools_4.3.3                 ggnewscale_0.5.2           \n [73] ragg_1.5.0                  stringdist_0.9.15          \n [75] ica_1.0-3                   Rcpp_1.1.0                 \n [77] gridExtra_2.3               SparseArray_1.2.2          \n [79] qs2_0.1.5                   qvalue_2.34.0              \n [81] MatrixGenerics_1.14.0       GenomeInfoDb_1.38.1        \n [83] withr_3.0.2                 fastmap_1.2.0              \n [85] fansi_1.0.6                 SparseM_1.84-2             \n [87] ttservice_0.5.3             digest_0.6.37              \n [89] R6_2.6.1                    mime_0.13                  \n [91] textshaping_0.4.0           colorspace_2.1-1           \n [93] scattermore_1.2             tensor_1.5.1               \n [95] dichromat_2.0-0.1           spatstat.data_3.1-8        \n [97] R.methodsS3_1.8.2           tidyr_1.3.1                \n [99] generics_0.1.4              data.table_1.17.8          \n[101] graphlayouts_1.2.2          httr_1.4.7                 \n[103] htmlwidgets_1.6.4           S4Arrays_1.2.0             \n[105] uwot_0.2.3                  pkgconfig_2.0.3            \n[107] gtable_0.3.6                lmtest_0.9-40              \n[109] SingleCellExperiment_1.24.0 XVector_0.42.0             \n[111] htmltools_0.5.8.1           dotCall64_1.2              \n[113] gglogger_0.1.7              scales_1.4.0               \n[115] Biobase_2.62.0              png_0.1-8                  \n[117] spatstat.univar_3.1-4       ggdendro_0.2.0             \n[119] rjson_0.2.23                reshape2_1.4.4             \n[121] nlme_3.1-168                cachem_1.1.0               \n[123] zoo_1.8-14                  GlobalOptions_0.1.2        \n[125] stringr_1.5.2               KernSmooth_2.23-26         \n[127] parallel_4.3.3              miniUI_0.1.2               \n[129] pillar_1.11.0               grid_4.3.3                 \n[131] logger_0.4.0                vctrs_0.6.5                \n[133] RANN_2.6.2                  VGAM_1.1-13                \n[135] promises_1.3.3              stringfish_0.17.0          \n[137] xtable_1.8-4                cluster_2.1.8.1            \n[139] truncdist_1.0-2             cli_3.6.5                  \n[141] compiler_4.3.3              rlang_1.1.6                \n[143] crayon_1.5.3                future.apply_1.20.0        \n[145] labeling_0.4.3              plyr_1.8.9                 \n[147] forcats_1.0.0               stringi_1.8.7              \n[149] viridisLite_0.4.2           tidyseurat_0.8.0           \n[151] deldir_2.0-4                scplotter_0.6.5            \n[153] assertthat_0.2.1            gsl_2.1-8                  \n[155] lazyeval_0.2.2              spatstat.geom_3.5-0        \n[157] quantreg_6.1                Matrix_1.6-5               \n[159] RcppHNSW_0.6.0              patchwork_1.3.2            \n[161] ggplot2_3.5.2               shiny_1.11.1               \n[163] plotthis_0.10.0             SummarizedExperiment_1.32.0\n[165] evd_2.3-7.1                 ROCR_1.0-11                \n[167] gridtext_0.1.5              memoise_2.0.1              \n[169] igraph_2.1.4                RcppParallel_5.1.9         \n",time:"# Generated by pipen-runinfo v1.1.2\n\nCommand: stdbuf -oL Rscript /mnt/disks/.cwd/.pipen/Immunopipe/SeuratPreparing/0/job.script\nVoluntary context switches: 646471\nInvoluntary context switches: 1075267\nPercentage of CPU this job got: 961%\nMajor page faults: 22\nMinor page faults: 130836646\nMaximum resident set size (kB): 33712364\nElapsed real time (s): 1199.30\nSystem (kernel) time (s): 355.83\nUser time (s): 11177.76\nExit status: 0\n",device:"# Generated by pipen-runinfo v1.1.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nf523046eea1a\n\nCPU\n----\nArchitecture:                            x86_64\nCPU op-mode(s):                          32-bit, 64-bit\nAddress sizes:                           46 bits physical, 48 bits virtual\nByte Order:                              Little Endian\nCPU(s):                                  64\nOn-line CPU(s) list:                     0-63\nVendor ID:                               GenuineIntel\nModel name:                              Intel(R) Xeon(R) CPU @ 2.80GHz\nCPU family:                              6\nModel:                                   85\nThread(s) per core:                      2\nCore(s) per socket:                      16\nSocket(s):                               2\nStepping:                                7\nBogoMIPS:                                5599.99\nFlags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch ssbd ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves arat avx512_vnni md_clear arch_capabilities\nHypervisor vendor:                       KVM\nVirtualization type:                     full\nL1d cache:                               1 MiB (32 instances)\nL1i cache:                               1 MiB (32 instances)\nL2 cache:                                32 MiB (32 instances)\nL3 cache:                                66 MiB (2 instances)\nNUMA node(s):                            2\nNUMA node0 CPU(s):                       0-15,32-47\nNUMA node1 CPU(s):                       16-31,48-63\nVulnerability Gather data sampling:      Not affected\nVulnerability Indirect target selection: Mitigation; Aligned branch/return thunks\nVulnerability Itlb multihit:             Not affected\nVulnerability L1tf:                      Not affected\nVulnerability Mds:                       Not affected\nVulnerability Meltdown:                  Not affected\nVulnerability Mmio stale data:           Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown\nVulnerability Reg file data sampling:    Not affected\nVulnerability Retbleed:                  Mitigation; Enhanced IBRS\nVulnerability Spec rstack overflow:      Not affected\nVulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:                Mitigation; Enhanced / Automatic IBRS; IBPB conditional; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                     Not affected\nVulnerability Tsa:                       Not affected\nVulnerability Tsx async abort:           Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown\nVulnerability Vmscape:                   Not affected\n\nMemory\n------\n               total        used        free      shared  buff/cache   available\nMem:           503Gi       4.3Gi       492Gi       704Ki        10Gi       499Gi\nSwap:             0B          0B          0B\n\nDisk\n----\nFilesystem                                     Size  Used Avail Use% Mounted on\noverlay                                        248G   11G  238G   5% /\ntmpfs                                           64M     0   64M   0% /dev\nshm                                             64M     0   64M   0% /dev/shm\n/dev/sda1                                      248G   11G  238G   5% /etc/hosts\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/xqute_workdir\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/editable\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/.cwd\ntmpfs                                          252G     0  252G   0% /proc/acpi\ntmpfs                                          252G     0  252G   0% /proc/scsi\ntmpfs                                          252G     0  252G   0% /sys/firmware\n\nNetwork\n-------\nNeither ifconfig nor ip is available.\n\nGPU\n---\nnvidia-smi is not available.\n\n"}},{name:"ClusterMarkersOfAllCells",desc:"Markers for clusters of all cells.",npages:1,report_toc:!0,order:4,runinfo:{session:"# Generated by pipen_runinfo v%(version)s\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 12 (bookworm)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.30.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] biopipen.utils_0.3.5 plotthis_0.10.0      ggplot2_3.5.2       \n [4] tidyr_1.3.1          tidyseurat_0.8.0     ttservice_0.5.3     \n [7] Seurat_5.3.0         SeuratObject_5.2.0   sp_2.2-0            \n[10] dplyr_1.1.4          rlang_1.1.6         \n\nloaded via a namespace (and not attached):\n  [1] cubature_2.0.4.6            RcppAnnoy_0.0.22           \n  [3] splines_4.3.3               later_1.4.4                \n  [5] bitops_1.0-9                tibble_3.3.0               \n  [7] polyclip_1.10-7             enrichit_0.0.4             \n  [9] fastDummies_1.7.5           lifecycle_1.0.4            \n [11] doParallel_1.0.17           globals_0.18.0             \n [13] lattice_0.22-7              MASS_7.3-60.0.1            \n [15] magrittr_2.0.3              limma_3.58.1               \n [17] plotly_4.11.0               httpuv_1.6.16              \n [19] sctransform_0.4.2           spam_2.11-1                \n [21] spatstat.sparse_3.1-0       reticulate_1.43.0          \n [23] cowplot_1.2.0               pbapply_1.7-4              \n [25] RColorBrewer_1.1-3          abind_1.4-5                \n [27] zlibbioc_1.48.0             Rtsne_0.17                 \n [29] GenomicRanges_1.54.1        presto_1.0.0               \n [31] purrr_1.1.0                 ggraph_2.2.1               \n [33] BiocGenerics_0.48.1         RCurl_1.98-1.17            \n [35] tweenr_2.0.3                evmix_2.12                 \n [37] circlize_0.4.16             GenomeInfoDbData_1.2.11    \n [39] IRanges_2.36.0              S4Vectors_0.40.2           \n [41] ggrepel_0.9.6               irlba_2.3.5.1              \n [43] listenv_0.9.1               spatstat.utils_3.1-5       \n [45] iNEXT_3.0.2                 MatrixModels_0.5-4         \n [47] goftest_1.2-3               RSpectra_0.16-2            \n [49] scRepertoire_2.2.1          spatstat.random_3.4-1      \n [51] fitdistrplus_1.2-4          parallelly_1.45.1          \n [53] codetools_0.2-20            DelayedArray_0.28.0        \n [55] xml2_1.4.0                  ggforce_0.5.0              \n [57] tidyselect_1.2.1            shape_1.4.6.1              \n [59] farver_2.1.2                viridis_0.6.5              \n [61] matrixStats_1.5.0           stats4_4.3.3               \n [63] spatstat.explore_3.5-2      jsonlite_2.0.0             \n [65] GetoptLong_1.0.5            ellipsis_0.3.2             \n [67] tidygraph_1.3.0             progressr_0.15.1           \n [69] iterators_1.0.14            ggridges_0.5.7             \n [71] ggalluvial_0.12.5           survival_3.8-3             \n [73] systemfonts_1.2.3           foreach_1.5.2              \n [75] tools_4.3.3                 ggnewscale_0.5.2           \n [77] ragg_1.5.0                  stringdist_0.9.15          \n [79] ica_1.0-3                   Rcpp_1.1.0                 \n [81] glue_1.8.0                  gridExtra_2.3              \n [83] SparseArray_1.2.2           qs2_0.1.5                  \n [85] qvalue_2.34.0               MatrixGenerics_1.14.0      \n [87] GenomeInfoDb_1.38.1         withr_3.0.2                \n [89] fastmap_1.2.0               fansi_1.0.6                \n [91] SparseM_1.84-2              digest_0.6.37              \n [93] R6_2.6.1                    mime_0.13                  \n [95] textshaping_0.4.0           colorspace_2.1-1           \n [97] Cairo_1.6-2                 scattermore_1.2            \n [99] tensor_1.5.1                dichromat_2.0-0.1          \n[101] spatstat.data_3.1-8         generics_0.1.4             \n[103] data.table_1.17.8           graphlayouts_1.2.2         \n[105] httr_1.4.7                  htmlwidgets_1.6.4          \n[107] S4Arrays_1.2.0              uwot_0.2.3                 \n[109] pkgconfig_2.0.3             gtable_0.3.6               \n[111] ComplexHeatmap_2.18.0       lmtest_0.9-40              \n[113] SingleCellExperiment_1.24.0 XVector_0.42.0             \n[115] htmltools_0.5.8.1           dotCall64_1.2              \n[117] clue_0.3-66                 scales_1.4.0               \n[119] Biobase_2.62.0              png_0.1-8                  \n[121] spatstat.univar_3.1-4       ggdendro_0.2.0             \n[123] rjson_0.2.23                reshape2_1.4.4             \n[125] nlme_3.1-168                cachem_1.1.0               \n[127] zoo_1.8-14                  GlobalOptions_0.1.2        \n[129] stringr_1.5.2               KernSmooth_2.23-26         \n[131] parallel_4.3.3              miniUI_0.1.2               \n[133] pillar_1.11.0               grid_4.3.3                 \n[135] logger_0.4.0                vctrs_0.6.5                \n[137] RANN_2.6.2                  VGAM_1.1-13                \n[139] promises_1.3.3              stringfish_0.17.0          \n[141] xtable_1.8-4                cluster_2.1.8.1            \n[143] truncdist_1.0-2             cli_3.6.5                  \n[145] compiler_4.3.3              crayon_1.5.3               \n[147] future.apply_1.20.0         labeling_0.4.3             \n[149] plyr_1.8.9                  forcats_1.0.0              \n[151] stringi_1.8.7               viridisLite_0.4.2          \n[153] deldir_2.0-4                scplotter_0.6.5            \n[155] assertthat_0.2.1            gsl_2.1-8                  \n[157] lazyeval_0.2.2              spatstat.geom_3.5-0        \n[159] quantreg_6.1                Matrix_1.6-5               \n[161] RcppHNSW_0.6.0              patchwork_1.3.2            \n[163] future_1.67.0               statmod_1.5.0              \n[165] shiny_1.11.1                SummarizedExperiment_1.32.0\n[167] evd_2.3-7.1                 ROCR_1.0-11                \n[169] gridtext_0.1.5              memoise_2.0.1              \n[171] igraph_2.1.4                RcppParallel_5.1.9         \n[173] bracer_1.2.1               \n",time:"# Generated by pipen-runinfo v1.1.2\n\nCommand: stdbuf -oL Rscript /mnt/disks/.cwd/.pipen/Immunopipe/ClusterMarkersOfAllCells/0/job.script\nVoluntary context switches: 25948\nInvoluntary context switches: 3192\nPercentage of CPU this job got: 97%\nMajor page faults: 0\nMinor page faults: 167714326\nMaximum resident set size (kB): 27154380\nElapsed real time (s): 1277.38\nSystem (kernel) time (s): 201.71\nUser time (s): 1047.79\nExit status: 0\n",device:"# Generated by pipen-runinfo v1.1.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nf523046eea1a\n\nCPU\n----\nArchitecture:                            x86_64\nCPU op-mode(s):                          32-bit, 64-bit\nAddress sizes:                           46 bits physical, 48 bits virtual\nByte Order:                              Little Endian\nCPU(s):                                  64\nOn-line CPU(s) list:                     0-63\nVendor ID:                               GenuineIntel\nModel name:                              Intel(R) Xeon(R) CPU @ 2.80GHz\nCPU family:                              6\nModel:                                   85\nThread(s) per core:                      2\nCore(s) per socket:                      16\nSocket(s):                               2\nStepping:                                7\nBogoMIPS:                                5599.99\nFlags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch ssbd ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves arat avx512_vnni md_clear arch_capabilities\nHypervisor vendor:                       KVM\nVirtualization type:                     full\nL1d cache:                               1 MiB (32 instances)\nL1i cache:                               1 MiB (32 instances)\nL2 cache:                                32 MiB (32 instances)\nL3 cache:                                66 MiB (2 instances)\nNUMA node(s):                            2\nNUMA node0 CPU(s):                       0-15,32-47\nNUMA node1 CPU(s):                       16-31,48-63\nVulnerability Gather data sampling:      Not affected\nVulnerability Indirect target selection: Mitigation; Aligned branch/return thunks\nVulnerability Itlb multihit:             Not affected\nVulnerability L1tf:                      Not affected\nVulnerability Mds:                       Not affected\nVulnerability Meltdown:                  Not affected\nVulnerability Mmio stale data:           Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown\nVulnerability Reg file data sampling:    Not affected\nVulnerability Retbleed:                  Mitigation; Enhanced IBRS\nVulnerability Spec rstack overflow:      Not affected\nVulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:                Mitigation; Enhanced / Automatic IBRS; IBPB conditional; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                     Not affected\nVulnerability Tsa:                       Not affected\nVulnerability Tsx async abort:           Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown\nVulnerability Vmscape:                   Not affected\n\nMemory\n------\n               total        used        free      shared  buff/cache   available\nMem:           503Gi       4.4Gi       488Gi       704Ki        13Gi       499Gi\nSwap:             0B          0B          0B\n\nDisk\n----\nFilesystem                                     Size  Used Avail Use% Mounted on\noverlay                                        248G   11G  238G   5% /\ntmpfs                                           64M     0   64M   0% /dev\nshm                                             64M     0   64M   0% /dev/shm\n/dev/sda1                                      248G   11G  238G   5% /etc/hosts\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/xqute_workdir\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/editable\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/.cwd\ntmpfs                                          252G     0  252G   0% /proc/acpi\ntmpfs                                          252G     0  252G   0% /proc/scsi\ntmpfs                                          252G     0  252G   0% /sys/firmware\n\nNetwork\n-------\nNeither ifconfig nor ip is available.\n\nGPU\n---\nnvidia-smi is not available.\n\n"}},{name:"TOrBCellSelection",desc:"Separate T and non-T cells and select T cells; or separate B and non-B cells and select B cells.",npages:1,report_toc:!0,order:5,runinfo:{session:"# Generated by pipen_runinfo v%(version)s\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 12 (bookworm)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.30.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] biopipen.utils_0.3.5 scplotter_0.6.5      plotthis_0.10.0     \n [4] Seurat_5.3.0         SeuratObject_5.2.0   sp_2.2-0            \n [7] tibble_3.3.0         tidyr_1.3.1          dplyr_1.1.4         \n[10] rlang_1.1.6         \n\nloaded via a namespace (and not attached):\n  [1] cubature_2.0.4.6            RcppAnnoy_0.0.22           \n  [3] splines_4.3.3               later_1.4.4                \n  [5] bitops_1.0-9                polyclip_1.10-7            \n  [7] enrichit_0.0.4              fastDummies_1.7.5          \n  [9] lifecycle_1.0.4             globals_0.18.0             \n [11] lattice_0.22-7              MASS_7.3-60.0.1            \n [13] magrittr_2.0.3              plotly_4.11.0              \n [15] httpuv_1.6.16               sctransform_0.4.2          \n [17] spam_2.11-1                 spatstat.sparse_3.1-0      \n [19] reticulate_1.43.0           cowplot_1.2.0              \n [21] pbapply_1.7-4               RColorBrewer_1.1-3         \n [23] abind_1.4-5                 zlibbioc_1.48.0            \n [25] Rtsne_0.17                  GenomicRanges_1.54.1       \n [27] purrr_1.1.0                 ggraph_2.2.1               \n [29] BiocGenerics_0.48.1         RCurl_1.98-1.17            \n [31] tweenr_2.0.3                evmix_2.12                 \n [33] circlize_0.4.16             GenomeInfoDbData_1.2.11    \n [35] IRanges_2.36.0              S4Vectors_0.40.2           \n [37] ggrepel_0.9.6               irlba_2.3.5.1              \n [39] listenv_0.9.1               spatstat.utils_3.1-5       \n [41] iNEXT_3.0.2                 MatrixModels_0.5-4         \n [43] goftest_1.2-3               RSpectra_0.16-2            \n [45] scRepertoire_2.2.1          spatstat.random_3.4-1      \n [47] fitdistrplus_1.2-4          parallelly_1.45.1          \n [49] codetools_0.2-20            DelayedArray_0.28.0        \n [51] xml2_1.4.0                  ggforce_0.5.0              \n [53] tidyselect_1.2.1            shape_1.4.6.1              \n [55] farver_2.1.2                viridis_0.6.5              \n [57] matrixStats_1.5.0           stats4_4.3.3               \n [59] spatstat.explore_3.5-2      jsonlite_2.0.0             \n [61] ellipsis_0.3.2              tidygraph_1.3.0            \n [63] progressr_0.15.1            ggridges_0.5.7             \n [65] ggalluvial_0.12.5           survival_3.8-3             \n [67] tools_4.3.3                 ggnewscale_0.5.2           \n [69] stringdist_0.9.15           ica_1.0-3                  \n [71] Rcpp_1.1.0                  glue_1.8.0                 \n [73] gridExtra_2.3               SparseArray_1.2.2          \n [75] qs2_0.1.5                   qvalue_2.34.0              \n [77] MatrixGenerics_1.14.0       GenomeInfoDb_1.38.1        \n [79] withr_3.0.2                 fastmap_1.2.0              \n [81] fansi_1.0.6                 ttservice_0.5.3            \n [83] SparseM_1.84-2              digest_0.6.37              \n [85] R6_2.6.1                    mime_0.13                  \n [87] colorspace_2.1-1            scattermore_1.2            \n [89] tensor_1.5.1                dichromat_2.0-0.1          \n [91] spatstat.data_3.1-8         generics_0.1.4             \n [93] data.table_1.17.8           graphlayouts_1.2.2         \n [95] httr_1.4.7                  htmlwidgets_1.6.4          \n [97] S4Arrays_1.2.0              uwot_0.2.3                 \n [99] pkgconfig_2.0.3             gtable_0.3.6               \n[101] lmtest_0.9-40               SingleCellExperiment_1.24.0\n[103] XVector_0.42.0              htmltools_0.5.8.1          \n[105] dotCall64_1.2               scales_1.4.0               \n[107] Biobase_2.62.0              png_0.1-8                  \n[109] spatstat.univar_3.1-4       ggdendro_0.2.0             \n[111] reshape2_1.4.4              rjson_0.2.23               \n[113] nlme_3.1-168                zoo_1.8-14                 \n[115] cachem_1.1.0                GlobalOptions_0.1.2        \n[117] stringr_1.5.2               KernSmooth_2.23-26         \n[119] parallel_4.3.3              miniUI_0.1.2               \n[121] logger_0.4.0                pillar_1.11.0              \n[123] grid_4.3.3                  vctrs_0.6.5                \n[125] RANN_2.6.2                  VGAM_1.1-13                \n[127] promises_1.3.3              stringfish_0.17.0          \n[129] xtable_1.8-4                cluster_2.1.8.1            \n[131] truncdist_1.0-2             cli_3.6.5                  \n[133] compiler_4.3.3              crayon_1.5.3               \n[135] future.apply_1.20.0         labeling_0.4.3             \n[137] plyr_1.8.9                  forcats_1.0.0              \n[139] stringi_1.8.7               tidyseurat_0.8.0           \n[141] viridisLite_0.4.2           deldir_2.0-4               \n[143] assertthat_0.2.1            gsl_2.1-8                  \n[145] lazyeval_0.2.2              spatstat.geom_3.5-0        \n[147] quantreg_6.1                Matrix_1.6-5               \n[149] RcppHNSW_0.6.0              patchwork_1.3.2            \n[151] future_1.67.0               ggplot2_3.5.2              \n[153] shiny_1.11.1                SummarizedExperiment_1.32.0\n[155] evd_2.3-7.1                 ROCR_1.0-11                \n[157] gridtext_0.1.5              igraph_2.1.4               \n[159] memoise_2.0.1               RcppParallel_5.1.9         \n[161] bracer_1.2.1               \n",time:"# Generated by pipen-runinfo v1.1.2\n\nCommand: stdbuf -oL Rscript /mnt/disks/.cwd/.pipen/Immunopipe/TOrBCellSelection/0/job.script\nVoluntary context switches: 5423\nInvoluntary context switches: 446\nPercentage of CPU this job got: 98%\nMajor page faults: 0\nMinor page faults: 4944127\nMaximum resident set size (kB): 12627788\nElapsed real time (s): 48.87\nSystem (kernel) time (s): 6.59\nUser time (s): 41.38\nExit status: 0\n",device:"# Generated by pipen-runinfo v1.1.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nf523046eea1a\n\nCPU\n----\nArchitecture:                            x86_64\nCPU op-mode(s):                          32-bit, 64-bit\nAddress sizes:                           46 bits physical, 48 bits virtual\nByte Order:                              Little Endian\nCPU(s):                                  64\nOn-line CPU(s) list:                     0-63\nVendor ID:                               GenuineIntel\nModel name:                              Intel(R) Xeon(R) CPU @ 2.80GHz\nCPU family:                              6\nModel:                                   85\nThread(s) per core:                      2\nCore(s) per socket:                      16\nSocket(s):                               2\nStepping:                                7\nBogoMIPS:                                5599.99\nFlags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch ssbd ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves arat avx512_vnni md_clear arch_capabilities\nHypervisor vendor:                       KVM\nVirtualization type:                     full\nL1d cache:                               1 MiB (32 instances)\nL1i cache:                               1 MiB (32 instances)\nL2 cache:                                32 MiB (32 instances)\nL3 cache:                                66 MiB (2 instances)\nNUMA node(s):                            2\nNUMA node0 CPU(s):                       0-15,32-47\nNUMA node1 CPU(s):                       16-31,48-63\nVulnerability Gather data sampling:      Not affected\nVulnerability Indirect target selection: Mitigation; Aligned branch/return thunks\nVulnerability Itlb multihit:             Not affected\nVulnerability L1tf:                      Not affected\nVulnerability Mds:                       Not affected\nVulnerability Meltdown:                  Not affected\nVulnerability Mmio stale data:           Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown\nVulnerability Reg file data sampling:    Not affected\nVulnerability Retbleed:                  Mitigation; Enhanced IBRS\nVulnerability Spec rstack overflow:      Not affected\nVulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:                Mitigation; Enhanced / Automatic IBRS; IBPB conditional; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                     Not affected\nVulnerability Tsa:                       Not affected\nVulnerability Tsx async abort:           Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown\nVulnerability Vmscape:                   Not affected\n\nMemory\n------\n               total        used        free      shared  buff/cache   available\nMem:           503Gi       4.5Gi       487Gi       704Ki        14Gi       499Gi\nSwap:             0B          0B          0B\n\nDisk\n----\nFilesystem                                     Size  Used Avail Use% Mounted on\noverlay                                        248G   11G  238G   5% /\ntmpfs                                           64M     0   64M   0% /dev\nshm                                             64M     0   64M   0% /dev/shm\n/dev/sda1                                      248G   11G  238G   5% /etc/hosts\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/xqute_workdir\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/editable\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/.cwd\ntmpfs                                          252G     0  252G   0% /proc/acpi\ntmpfs                                          252G     0  252G   0% /proc/scsi\ntmpfs                                          252G     0  252G   0% /sys/firmware\n\nNetwork\n-------\nNeither ifconfig nor ip is available.\n\nGPU\n---\nnvidia-smi is not available.\n\n"}},{name:"SeuratClusterStats",desc:"Statistics of the clustering.",npages:2,report_toc:!0,order:9,runinfo:{session:"# Generated by pipen_runinfo v%(version)s\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 12 (bookworm)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.30.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] biopipen.utils_0.3.5 scplotter_0.6.5      gglogger_0.1.7      \n [4] ggplot2_3.5.2        tidyseurat_0.8.0     ttservice_0.5.3     \n [7] forcats_1.0.0        glue_1.8.0           tibble_3.3.0        \n[10] tidyr_1.3.1          dplyr_1.1.4          rlang_1.1.6         \n[13] Seurat_5.3.0         SeuratObject_5.2.0   sp_2.2-0            \n\nloaded via a namespace (and not attached):\n  [1] cubature_2.0.4.6            RcppAnnoy_0.0.22           \n  [3] splines_4.3.3               later_1.4.4                \n  [5] bitops_1.0-9                polyclip_1.10-7            \n  [7] enrichit_0.0.4              fastDummies_1.7.5          \n  [9] lifecycle_1.0.4             doParallel_1.0.17          \n [11] globals_0.18.0              lattice_0.22-7             \n [13] MASS_7.3-60.0.1             magrittr_2.0.3             \n [15] plotly_4.11.0               httpuv_1.6.16              \n [17] sctransform_0.4.2           spam_2.11-1                \n [19] spatstat.sparse_3.1-0       reticulate_1.43.0          \n [21] cowplot_1.2.0               pbapply_1.7-4              \n [23] RColorBrewer_1.1-3          abind_1.4-5                \n [25] zlibbioc_1.48.0             Rtsne_0.17                 \n [27] GenomicRanges_1.54.1        purrr_1.1.0                \n [29] ggraph_2.2.1                BiocGenerics_0.48.1        \n [31] RCurl_1.98-1.17             tweenr_2.0.3               \n [33] evmix_2.12                  circlize_0.4.16            \n [35] GenomeInfoDbData_1.2.11     IRanges_2.36.0             \n [37] S4Vectors_0.40.2            ggrepel_0.9.6              \n [39] irlba_2.3.5.1               listenv_0.9.1              \n [41] spatstat.utils_3.1-5        iNEXT_3.0.2                \n [43] MatrixModels_0.5-4          goftest_1.2-3              \n [45] RSpectra_0.16-2             scRepertoire_2.2.1         \n [47] spatstat.random_3.4-1       fitdistrplus_1.2-4         \n [49] parallelly_1.45.1           codetools_0.2-20           \n [51] DelayedArray_0.28.0         xml2_1.4.0                 \n [53] ggforce_0.5.0               tidyselect_1.2.1           \n [55] shape_1.4.6.1               farver_2.1.2               \n [57] viridis_0.6.5               matrixStats_1.5.0          \n [59] stats4_4.3.3                spatstat.explore_3.5-2     \n [61] jsonlite_2.0.0              GetoptLong_1.0.5           \n [63] ellipsis_0.3.2              tidygraph_1.3.0            \n [65] progressr_0.15.1            iterators_1.0.14           \n [67] ggridges_0.5.7              ggalluvial_0.12.5          \n [69] survival_3.8-3              systemfonts_1.2.3          \n [71] foreach_1.5.2               tools_4.3.3                \n [73] ggnewscale_0.5.2            ragg_1.5.0                 \n [75] stringdist_0.9.15           ica_1.0-3                  \n [77] Rcpp_1.1.0                  gridExtra_2.3              \n [79] SparseArray_1.2.2           qs2_0.1.5                  \n [81] qvalue_2.34.0               MatrixGenerics_1.14.0      \n [83] GenomeInfoDb_1.38.1         withr_3.0.2                \n [85] fastmap_1.2.0               fansi_1.0.6                \n [87] SparseM_1.84-2              digest_0.6.37              \n [89] R6_2.6.1                    mime_0.13                  \n [91] textshaping_0.4.0           colorspace_2.1-1           \n [93] Cairo_1.6-2                 scattermore_1.2            \n [95] tensor_1.5.1                dichromat_2.0-0.1          \n [97] spatstat.data_3.1-8         generics_0.1.4             \n [99] data.table_1.17.8           graphlayouts_1.2.2         \n[101] httr_1.4.7                  htmlwidgets_1.6.4          \n[103] S4Arrays_1.2.0              uwot_0.2.3                 \n[105] pkgconfig_2.0.3             gtable_0.3.6               \n[107] ComplexHeatmap_2.18.0       lmtest_0.9-40              \n[109] SingleCellExperiment_1.24.0 XVector_0.42.0             \n[111] htmltools_0.5.8.1           dotCall64_1.2              \n[113] clue_0.3-66                 scales_1.4.0               \n[115] Biobase_2.62.0              png_0.1-8                  \n[117] spatstat.univar_3.1-4       ggdendro_0.2.0             \n[119] rjson_0.2.23                reshape2_1.4.4             \n[121] nlme_3.1-168                zoo_1.8-14                 \n[123] cachem_1.1.0                GlobalOptions_0.1.2        \n[125] stringr_1.5.2               KernSmooth_2.23-26         \n[127] parallel_4.3.3              miniUI_0.1.2               \n[129] logger_0.4.0                pillar_1.11.0              \n[131] grid_4.3.3                  vctrs_0.6.5                \n[133] RANN_2.6.2                  VGAM_1.1-13                \n[135] promises_1.3.3              stringfish_0.17.0          \n[137] xtable_1.8-4                cluster_2.1.8.1            \n[139] truncdist_1.0-2             cli_3.6.5                  \n[141] compiler_4.3.3              crayon_1.5.3               \n[143] future.apply_1.20.0         labeling_0.4.3             \n[145] plyr_1.8.9                  stringi_1.8.7              \n[147] viridisLite_0.4.2           deldir_2.0-4               \n[149] assertthat_0.2.1            gsl_2.1-8                  \n[151] lazyeval_0.2.2              spatstat.geom_3.5-0        \n[153] quantreg_6.1                Matrix_1.6-5               \n[155] RcppHNSW_0.6.0              patchwork_1.3.2            \n[157] future_1.67.0               shiny_1.11.1               \n[159] plotthis_0.10.0             SummarizedExperiment_1.32.0\n[161] evd_2.3-7.1                 ROCR_1.0-11                \n[163] gridtext_0.1.5              igraph_2.1.4               \n[165] memoise_2.0.1               RcppParallel_5.1.9         \n[167] bracer_1.2.1               \n",time:"# Generated by pipen-runinfo v1.1.2\n\nCommand: stdbuf -oL Rscript /mnt/disks/.cwd/.pipen/Immunopipe/SeuratClusterStats/0/job.script\nVoluntary context switches: 5045\nInvoluntary context switches: 356\nPercentage of CPU this job got: 118%\nMajor page faults: 0\nMinor page faults: 2868055\nMaximum resident set size (kB): 6910156\nElapsed real time (s): 71.87\nSystem (kernel) time (s): 4.54\nUser time (s): 80.86\nExit status: 0\n",device:"# Generated by pipen-runinfo v1.1.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\n203b1956eb36\n\n"}},{name:"ClusterMarkers",desc:"Markers for clusters of all or selected T/B cells.",npages:1,report_toc:!0,order:10},{name:"ClonalStats",desc:"Visualize the clonal information.",npages:1,report_toc:!0,order:11}]};function Vt(e){let n,t;return n=new Un({props:{dir:"up"}}),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function Bt(e){let n,t;return n=new Un({props:{dir:"down"}}),{c(){h(n.$$.fragment)},m(e,s){$(n,e,s),t=!0},i(e){t||(l(n.$$.fragment,e),t=!0)},o(e){o(n.$$.fragment,e),t=!1},d(e){g(n,e)}}}function Dt(e){let n,t,s,r,i,_=e[1]>0&&Vt();t=new It({});let u=e[1]<e[3].npages-1&&Bt();return{c(){_&&_.c(),n=b(),h(t.$$.fragment),s=b(),u&&u.c(),r=f()},m(e,a){_&&_.m(e,a),d(e,n,a),$(t,e,a),d(e,s,a),u&&u.m(e,a),d(e,r,a),i=!0},p(e,t){e[1]>0?_?2&t&&l(_,1):(_=Vt(),_.c(),l(_,1),_.m(n.parentNode,n)):_&&(c(),o(_,1,1,()=>{_=null}),p()),e[1]<e[3].npages-1?u?10&t&&l(u,1):(u=Bt(),u.c(),l(u,1),u.m(r.parentNode,r)):u&&(c(),o(u,1,1,()=>{u=null}),p())},i(e){i||(l(_),l(t.$$.fragment,e),l(u),i=!0)},o(e){o(_),o(t.$$.fragment,e),o(u),i=!1},d(e){e&&(a(n),a(s),a(r)),_&&_.d(e),g(t,e),u&&u.d(e)}}}function Ot(e){let n,t,s;function r(n){e[4](n)}let a={logo:e[0],logotext:e[3].desc,versions:kt.versions,entries:kt.entries,pipeline_name:kt.pipeline.name,report_toc:e[3].report_toc,runinfo:e[3].runinfo,toc:Ut,page:e[1],$$slots:{default:[Dt]},$$scope:{ctx:e}};return void 0!==e[2]&&(a.isSideNavOpen=e[2]),n=new Tn({props:a}),v.push(()=>x(n,"isSideNavOpen",r)),{c(){h(n.$$.fragment)},m(e,t){$(n,e,t),s=!0},p(e,[s]){const r={};1&s&&(r.logo=e[0]),8&s&&(r.logotext=e[3].desc),8&s&&(r.report_toc=e[3].report_toc),8&s&&(r.runinfo=e[3].runinfo),2&s&&(r.page=e[1]),42&s&&(r.$$scope={dirty:s,ctx:e}),!t&&4&s&&(t=!0,r.isSideNavOpen=e[2],y(()=>t=!1)),n.$set(r)},i(e){s||(l(n.$$.fragment,e),s=!0)},o(e){o(n.$$.fragment,e),s=!1},d(e){g(n,e)}}}function zt(e,n,t){let s,{name:r}=n,{page:a=0}=n,i=!1;for(const e of kt.entries)if(e.procs){for(const n of e.procs)if(n.name==r){s=n;break}}else if(e.name==r){s=e;break}return m(()=>{const e=new URL(window.location.href);let n;e.hash.length>1&&(n=document.getElementById(`${e.hash.slice(1)}`)),n&&n.scrollIntoView({block:"start",behavior:"smooth"})}),e.$$set=e=>{"name"in e&&t(0,r=e.name),"page"in e&&t(1,a=e.page)},[r,a,i,s,function(e){i=e,t(2,i)}]}e("default",new class extends n{constructor(e){super(),t(this,e,zt,Ot,s,{name:0,page:1})}}({target:document.body,props:{name:"SeuratClusterStats",page:1}}))}}});
