System.register(["./vendor/svelte-CkBJVEsY.js","./vendor/carbon-components-svelte-DfRG9uuU.js","./vendor/carbon-icons-svelte-Boxs1M4A.js","./vendor/temml-BTciUaue.js"],function(e){"use strict";var t,n,s,r,a,i,o,l,c,p,d,_,u,m,f,g,$,h,b,v,x,y,C,S,w,M,R,N,G,L,T,P,A,I,E,U,k,B,V,D,O,F,z,H,j,q,K,X,W,Z,Y,J,Q,ee,te,ne,se,re,ae,ie,oe,le,ce,pe,de,_e,ue,me,fe,ge,$e,he,be,ve,xe,ye,Ce,Se,we,Me,Re,Ne,Ge,Le,Te,Pe;return{setters:[function(e){t=e.S,n=e.i,s=e.s,r=e.a3,a=e.d,i=e.a8,o=e.t,l=e.a,c=e.g,p=e.c,d=e.b,_=e.p,u=e.Q,m=e.B,f=e.e,g=e.K,$=e.M,h=e.L,b=e.z,v=e.I,x=e.T,y=e.U,C=e.D,S=e.q,w=e.F,M=e.G,R=e.N,N=e.y,G=e.k,L=e.u,T=e.l,P=e.m,A=e.h,I=e.o,E=e.a2,U=e.f,k=e.j,B=e.A,V=e.ac,D=e.n,O=e.W,F=e.ad,z=e.x,H=e.ae,j=e.R},function(e){q=e.S,K=e.a,X=e.b,W=e.H,Z=e.T,Y=e.c,J=e.d,Q=e.e,ee=e.f,te=e.g,ne=e.M,se=e.I,re=e.h,ae=e.i,ie=e.j,oe=e.C,le=e.k,ce=e.l,pe=e.G,de=e.R,_e=e.m,ue=e.B,me=e.n,fe=e.o,ge=e.A},function(e){$e=e.C,he=e.a,be=e.D,ve=e.A,xe=e.S,ye=e.I,Ce=e.b,Se=e.c,we=e.V,Me=e.d,Re=e.T,Ne=e.G,Ge=e.J,Le=e.P,Te=e.e,Pe=e.f},null],execute:function(){function Ae(e,t,n){const s=e.slice();return s[4]=t[n],s}function Ie(e,t,n){const s=e.slice();return s[7]=t[n],s}function Ee(e){let t,n;return t=new K({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text,expanded:!0,$$slots:{default:[Be]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},p(e,n){const s={};1&n&&(s.href=`${e[1](e[4].page)}#${e[4].slug}`),1&n&&(s.class=e[2](e[4].page)),1&n&&(s.title=e[4].text),1&n&&(s.text=e[4].text),1025&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function Ue(e){let t,n;return t=new q({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text}}),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},p(e,n){const s={};1&n&&(s.href=`${e[1](e[4].page)}#${e[4].slug}`),1&n&&(s.class=e[2](e[4].page)),1&n&&(s.title=e[4].text),1&n&&(s.text=e[4].text),t.$set(s)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function ke(e){let t,n;return t=new X({props:{href:`${e[1](e[7].page)}#${e[7].slug}`,class:e[2](e[7].page),title:e[7].text,text:e[7].text}}),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},p(e,n){const s={};1&n&&(s.href=`${e[1](e[7].page)}#${e[7].slug}`),1&n&&(s.class=e[2](e[7].page)),1&n&&(s.title=e[7].text),1&n&&(s.text=e[7].text),t.$set(s)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function Be(e){let t,n,s=r(e[4].children),_=[];for(let t=0;t<s.length;t+=1)_[t]=ke(Ie(e,s,t));const u=e=>o(_[e],1,1,()=>{_[e]=null});return{c(){for(let e=0;e<_.length;e+=1)_[e].c();t=b()},m(e,s){for(let t=0;t<_.length;t+=1)_[t]&&_[t].m(e,s);d(e,t,s),n=!0},p(e,n){if(7&n){let a;for(s=r(e[4].children),a=0;a<s.length;a+=1){const r=Ie(e,s,a);_[a]?(_[a].p(r,n),l(_[a],1)):(_[a]=ke(r),_[a].c(),l(_[a],1),_[a].m(t.parentNode,t))}for(c(),a=s.length;a<_.length;a+=1)u(a);p()}},i(e){if(!n){for(let e=0;e<s.length;e+=1)l(_[e]);n=!0}},o(e){_=_.filter(Boolean);for(let e=0;e<_.length;e+=1)o(_[e]);n=!1},d(e){e&&a(t),i(_,e)}}}function Ve(e){let t,n,s,r;const i=[Ue,Ee],_=[];function u(e,t){return 0==e[4].children.length?0:1}return t=u(e),n=_[t]=i[t](e),{c(){n.c(),s=f()},m(e,n){_[t].m(e,n),d(e,s,n),r=!0},p(e,r){let a=t;t=u(e),t===a?_[t].p(e,r):(c(),o(_[a],1,1,()=>{_[a]=null}),p(),n=_[t],n?n.p(e,r):(n=_[t]=i[t](e),n.c()),l(n,1),n.m(s.parentNode,s))},i(e){r||(l(n),r=!0)},o(e){o(n),r=!1},d(e){e&&a(s),_[t].d(e)}}}function De(e){let t,n,s=r(e[0]),m=[];for(let t=0;t<s.length;t+=1)m[t]=Ve(Ae(e,s,t));const f=e=>o(m[e],1,1,()=>{m[e]=null});return{c(){t=_("ul");for(let e=0;e<m.length;e+=1)m[e].c();u(t,"class","bx--side-nav__items pipen-report-sidenav svelte-12gpq3")},m(e,s){d(e,t,s);for(let e=0;e<m.length;e+=1)m[e]&&m[e].m(t,null);n=!0},p(e,[n]){if(7&n){let a;for(s=r(e[0]),a=0;a<s.length;a+=1){const r=Ae(e,s,a);m[a]?(m[a].p(r,n),l(m[a],1)):(m[a]=Ve(r),m[a].c(),l(m[a],1),m[a].m(t,null))}for(c(),a=s.length;a<m.length;a+=1)f(a);p()}},i(e){if(!n){for(let e=0;e<s.length;e+=1)l(m[e]);n=!0}},o(e){m=m.filter(Boolean);for(let e=0;e<m.length;e+=1)o(m[e]);n=!1},d(e){e&&a(t),i(m,e)}}}function Oe(e,t,n){let{toc:s}=t,{page:r}=t;return m(()=>{let e,t=new URL(window.location.href);e=t.hash.length>0?document.querySelector(`[href="${t.hash}"]`):document.querySelector(".toc-current-page"),e.scrollIntoView({behavior:"smooth"})}),e.$$set=e=>{"toc"in e&&n(0,s=e.toc),"page"in e&&n(3,r=e.page)},[s,e=>{if(r===e)return"";let t=new URL(window.location.href);t.hash="";const n=(t.searchParams.get("proc")||"_index").replace(/-\d+$/,"");return t.searchParams.set("proc",0===e?n:`${n}-${e}`),t.toString()},e=>e!=r?"toc-noncurrent-page":"toc-current-page",r]}class Fe extends t{constructor(e){super(),n(this,e,Oe,De,s,{toc:0,page:3})}}function ze(e,t,n){const s=e.slice();return s[7]=t[n],s}function He(e,t,n){const s=e.slice();return s[10]=t[n],s}function je(e){let t,n;return t=new J({props:{isOpen:ct,transition:e[5],$$slots:{default:[st]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},p(e,n){const s={};8197&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function qe(e){let t;return{c(){t=M("Go to home:")},m(e,n){d(e,t,n)},d(e){e&&a(t)}}}function Ke(e){let t;return{c(){t=M(e[0])},m(e,n){d(e,t,n)},p(e,n){1&n&&w(t,e[0])},d(e){e&&a(t)}}}function Xe(e){let t;return{c(){t=M("Switch to process:")},m(e,n){d(e,t,n)},d(e){e&&a(t)}}}function We(e){let t,n;return t=new te({props:{title:e[7].name,href:`?proc=${e[7].name}`,$$slots:{default:[Ye]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},p(e,n){const s={};4&n&&(s.title=e[7].name),4&n&&(s.href=`?proc=${e[7].name}`),8196&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function Ze(e){let t,n,s,_;t=new te({props:{class:"procgroup",$$slots:{default:[Je]},$$scope:{ctx:e}}});let u=r(e[7].procs),m=[];for(let t=0;t<u.length;t+=1)m[t]=et(He(e,u,t));const v=e=>o(m[e],1,1,()=>{m[e]=null});return{c(){h(t.$$.fragment),n=b();for(let e=0;e<m.length;e+=1)m[e].c();s=f()},m(e,r){$(t,e,r),d(e,n,r);for(let t=0;t<m.length;t+=1)m[t]&&m[t].m(e,r);d(e,s,r),_=!0},p(e,n){const a={};if(8196&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a),4&n){let t;for(u=r(e[7].procs),t=0;t<u.length;t+=1){const r=He(e,u,t);m[t]?(m[t].p(r,n),l(m[t],1)):(m[t]=et(r),m[t].c(),l(m[t],1),m[t].m(s.parentNode,s))}for(c(),t=u.length;t<m.length;t+=1)v(t);p()}},i(e){if(!_){l(t.$$.fragment,e);for(let e=0;e<u.length;e+=1)l(m[e]);_=!0}},o(e){o(t.$$.fragment,e),m=m.filter(Boolean);for(let e=0;e<m.length;e+=1)o(m[e]);_=!1},d(e){e&&(a(n),a(s)),g(t,e),i(m,e)}}}function Ye(e){let t,n,s=e[7].name+"";return{c(){t=M(s),n=b()},m(e,s){d(e,t,s),d(e,n,s)},p(e,n){4&n&&s!==(s=e[7].name+"")&&w(t,s)},d(e){e&&(a(t),a(n))}}}function Je(e){let t,n=e[7].name+"";return{c(){t=M(n)},m(e,n){d(e,t,n)},p(e,s){4&s&&n!==(n=e[7].name+"")&&w(t,n)},d(e){e&&a(t)}}}function Qe(e){let t,n,s,r=e[10].name+"";return{c(){t=M("- "),n=M(r),s=b()},m(e,r){d(e,t,r),d(e,n,r),d(e,s,r)},p(e,t){4&t&&r!==(r=e[10].name+"")&&w(n,r)},d(e){e&&(a(t),a(n),a(s))}}}function et(e){let t,n;return t=new te({props:{title:e[10].name,class:"procgroup-proc",href:`?proc=${e[10].name}`,$$slots:{default:[Qe]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},p(e,n){const s={};4&n&&(s.title=e[10].name),4&n&&(s.href=`?proc=${e[10].name}`),8196&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function tt(e){let t,n,s,r;const i=[Ze,We],_=[];function u(e,t){return e[7].procs?0:1}return t=u(e),n=_[t]=i[t](e),{c(){n.c(),s=f()},m(e,n){_[t].m(e,n),d(e,s,n),r=!0},p(e,r){let a=t;t=u(e),t===a?_[t].p(e,r):(c(),o(_[a],1,1,()=>{_[a]=null}),p(),n=_[t],n?n.p(e,r):(n=_[t]=i[t](e),n.c()),l(n,1),n.m(s.parentNode,s))},i(e){r||(l(n),r=!0)},o(e){o(n),r=!1},d(e){e&&a(s),_[t].d(e)}}}function nt(e){let t,n,s,_,u,m,v,x;t=new ee({props:{$$slots:{default:[qe]},$$scope:{ctx:e}}}),s=new te({props:{href:"index.html",$$slots:{default:[Ke]},$$scope:{ctx:e}}}),u=new ee({props:{$$slots:{default:[Xe]},$$scope:{ctx:e}}});let y=r(e[2]),C=[];for(let t=0;t<y.length;t+=1)C[t]=tt(ze(e,y,t));const S=e=>o(C[e],1,1,()=>{C[e]=null});return{c(){h(t.$$.fragment),n=b(),h(s.$$.fragment),_=b(),h(u.$$.fragment),m=b();for(let e=0;e<C.length;e+=1)C[e].c();v=f()},m(e,r){$(t,e,r),d(e,n,r),$(s,e,r),d(e,_,r),$(u,e,r),d(e,m,r);for(let t=0;t<C.length;t+=1)C[t]&&C[t].m(e,r);d(e,v,r),x=!0},p(e,n){const a={};8192&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a);const i={};8193&n&&(i.$$scope={dirty:n,ctx:e}),s.$set(i);const o={};if(8192&n&&(o.$$scope={dirty:n,ctx:e}),u.$set(o),4&n){let t;for(y=r(e[2]),t=0;t<y.length;t+=1){const s=ze(e,y,t);C[t]?(C[t].p(s,n),l(C[t],1)):(C[t]=tt(s),C[t].c(),l(C[t],1),C[t].m(v.parentNode,v))}for(c(),t=y.length;t<C.length;t+=1)S(t);p()}},i(e){if(!x){l(t.$$.fragment,e),l(s.$$.fragment,e),l(u.$$.fragment,e);for(let e=0;e<y.length;e+=1)l(C[e]);x=!0}},o(e){o(t.$$.fragment,e),o(s.$$.fragment,e),o(u.$$.fragment,e),C=C.filter(Boolean);for(let e=0;e<C.length;e+=1)o(C[e]);x=!1},d(e){e&&(a(n),a(_),a(m),a(v)),g(t,e),g(s,e),g(u,e),i(C,e)}}}function st(e){let t,n;return t=new Q({props:{$$slots:{default:[nt]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},p(e,n){const s={};8197&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function rt(e){let t,n,s,r;n=new Z({props:{render:"select",theme:"g10",persist:!0,persistKey:"__pipen-report-theme",select:{themes:["g10","white","g80","g90","g100"],labelText:"Theme",inline:!0}}});let i=e[2].length>0&&je(e);return{c(){t=_("div"),h(n.$$.fragment),s=b(),i&&i.c(),u(t,"class","header-util"),S(t,"bx--header__global",!0)},m(e,a){d(e,t,a),$(n,t,null),C(t,s),i&&i.m(t,null),r=!0},p(e,n){e[2].length>0?i?(i.p(e,n),4&n&&l(i,1)):(i=je(e),i.c(),l(i,1),i.m(t,null)):i&&(c(),o(i,1,1,()=>{i=null}),p())},i(e){r||(l(n.$$.fragment,e),l(i),r=!0)},o(e){o(n.$$.fragment,e),o(i),r=!1},d(e){e&&a(t),g(n),i&&i.d()}}}function at(e){let t;return{c(){t=M(e[4])},m(e,n){d(e,t,n)},p(e,n){16&n&&w(t,e[4])},d(e){e&&a(t)}}}function it(e){let t,n,s,r,i=null!==e[4]&&at(e);return{c(){t=_("h1"),n=M(e[3]),s=b(),i&&i.c(),r=f()},m(e,a){d(e,t,a),C(t,n),d(e,s,a),i&&i.m(e,a),d(e,r,a)},p(e,t){8&t&&w(n,e[3]),null!==e[4]?i?i.p(e,t):(i=at(e),i.c(),i.m(r.parentNode,r)):i&&(i.d(1),i=null)},d(e){e&&(a(t),a(s),a(r)),i&&i.d(e)}}}function ot(e){let t,n;return t=new Y({props:{class:"header-tile",slot:"platform",$$slots:{default:[it]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},p(e,n){const s={};8216&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function lt(e){let t,n,s;function r(t){e[6](t)}let a={$$slots:{platform:[ot],default:[rt]},$$scope:{ctx:e}};return void 0!==e[1]&&(a.isSideNavOpen=e[1]),t=new W({props:a}),v.push(()=>x(t,"isSideNavOpen",r)),{c(){h(t.$$.fragment)},m(e,n){$(t,e,n),s=!0},p(e,[s]){const r={};8221&s&&(r.$$scope={dirty:s,ctx:e}),!n&&2&s&&(n=!0,r.isSideNavOpen=e[1],y(()=>n=!1)),t.$set(r)},i(e){s||(l(t.$$.fragment,e),s=!0)},o(e){o(t.$$.fragment,e),s=!1},d(e){g(t,e)}}}let ct=!1;function pt(e,t,n){let{entries:s}=t,{logo:r}=t,{logotext:a=null}=t,{pipeline_name:i}=t,{isSideNavOpen:o}=t;return i=i||r,e.$$set=e=>{"entries"in e&&n(2,s=e.entries),"logo"in e&&n(3,r=e.logo),"logotext"in e&&n(4,a=e.logotext),"pipeline_name"in e&&n(0,i=e.pipeline_name),"isSideNavOpen"in e&&n(1,o=e.isSideNavOpen)},[i,o,s,r,a,{duration:200},function(e){o=e,n(1,o)}]}class dt extends t{constructor(e){super(),n(this,e,pt,lt,s,{entries:2,logo:3,logotext:4,pipeline_name:0,isSideNavOpen:1})}}function _t(e){let t,n,s;function r(t){e[3](t)}let a={modalHeading:"Running Information",passiveModal:!0,hasScrollingContent:!0,class:"runinfo",$$slots:{default:[ht]},$$scope:{ctx:e}};return void 0!==e[2]&&(a.open=e[2]),t=new ne({props:a}),v.push(()=>x(t,"open",r)),{c(){h(t.$$.fragment)},m(e,n){$(t,e,n),s=!0},p(e,s){const r={};34&s&&(r.$$scope={dirty:s,ctx:e}),!n&&4&s&&(n=!0,r.open=e[2],y(()=>n=!1)),t.$set(r)},i(e){s||(l(t.$$.fragment,e),s=!0)},o(e){o(t.$$.fragment,e),s=!1},d(e){g(t,e)}}}function ut(e){let t,n,s,r,i,c;return t=new ae({props:{label:"Session Information"}}),s=new ae({props:{label:"Time Spent"}}),i=new ae({props:{label:"Device Information"}}),{c(){h(t.$$.fragment),n=b(),h(s.$$.fragment),r=b(),h(i.$$.fragment)},m(e,a){$(t,e,a),d(e,n,a),$(s,e,a),d(e,r,a),$(i,e,a),c=!0},p:R,i(e){c||(l(t.$$.fragment,e),l(s.$$.fragment,e),l(i.$$.fragment,e),c=!0)},o(e){o(t.$$.fragment,e),o(s.$$.fragment,e),o(i.$$.fragment,e),c=!1},d(e){e&&(a(n),a(r)),g(t,e),g(s,e),g(i,e)}}}function mt(e){let t,n;return t=new oe({props:{type:"multi",code:e[1].session}}),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},p(e,n){const s={};2&n&&(s.code=e[1].session),t.$set(s)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function ft(e){let t,n;return t=new oe({props:{type:"multi",code:e[1].time}}),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},p(e,n){const s={};2&n&&(s.code=e[1].time),t.$set(s)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function gt(e){let t,n;return t=new oe({props:{type:"multi",code:e[1].device}}),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},p(e,n){const s={};2&n&&(s.code=e[1].device),t.$set(s)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function $t(e){let t,n,s,r,i,c;return t=new ie({props:{$$slots:{default:[mt]},$$scope:{ctx:e}}}),s=new ie({props:{$$slots:{default:[ft]},$$scope:{ctx:e}}}),i=new ie({props:{$$slots:{default:[gt]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment),n=b(),h(s.$$.fragment),r=b(),h(i.$$.fragment)},m(e,a){$(t,e,a),d(e,n,a),$(s,e,a),d(e,r,a),$(i,e,a),c=!0},p(e,n){const r={};34&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r);const a={};34&n&&(a.$$scope={dirty:n,ctx:e}),s.$set(a);const o={};34&n&&(o.$$scope={dirty:n,ctx:e}),i.$set(o)},i(e){c||(l(t.$$.fragment,e),l(s.$$.fragment,e),l(i.$$.fragment,e),c=!0)},o(e){o(t.$$.fragment,e),o(s.$$.fragment,e),o(i.$$.fragment,e),c=!1},d(e){e&&(a(n),a(r)),g(t,e),g(s,e),g(i,e)}}}function ht(e){let t,n,s,r;return t=new se({props:{kind:"warning",lowContrast:!0,hideCloseButton:!0,title:"Showing the first job only. Check the workdir for information of other jobs if any."}}),s=new re({props:{$$slots:{content:[$t],default:[ut]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment),n=b(),h(s.$$.fragment)},m(e,a){$(t,e,a),d(e,n,a),$(s,e,a),r=!0},p(e,t){const n={};34&t&&(n.$$scope={dirty:t,ctx:e}),s.$set(n)},i(e){r||(l(t.$$.fragment,e),l(s.$$.fragment,e),r=!0)},o(e){o(t.$$.fragment,e),o(s.$$.fragment,e),r=!1},d(e){e&&a(n),g(t,e),g(s,e)}}}function bt(e){let t,n,s;return{c(){t=_("a"),t.textContent="> Running Information",u(t,"href","javascript:void(0);")},m(r,a){d(r,t,a),n||(s=N(t,"click",e[4]),n=!0)},p:R,d(e){e&&a(t),n=!1,s()}}}function vt(e){let t,n,s,r,i,m,f,g,$=e[1]&&_t(e),h=e[1]&&bt(e);return{c(){$&&$.c(),t=b(),n=_("div"),s=_("div"),h&&h.c(),r=b(),i=_("div"),m=M("Powered by "),f=_("span"),u(s,"class","runinfo"),u(f,"class","pipen-report-footer-versions"),u(i,"class","versions"),u(n,"class","pipen-report-footer bx--grid")},m(a,o){$&&$.m(a,o),d(a,t,o),d(a,n,o),C(n,s),h&&h.m(s,null),C(n,r),C(n,i),C(i,m),C(i,f),f.innerHTML=e[0],g=!0},p(e,[n]){e[1]?$?($.p(e,n),2&n&&l($,1)):($=_t(e),$.c(),l($,1),$.m(t.parentNode,t)):$&&(c(),o($,1,1,()=>{$=null}),p()),e[1]?h?h.p(e,n):(h=bt(e),h.c(),h.m(s,null)):h&&(h.d(1),h=null),(!g||1&n)&&(f.innerHTML=e[0])},i(e){g||(l($),g=!0)},o(e){o($),g=!1},d(e){e&&(a(t),a(n)),$&&$.d(e),h&&h.d()}}}function xt(e,t,n){let{versions:s}=t,{runinfo:r}=t,a=!1;return e.$$set=e=>{"versions"in e&&n(0,s=e.versions),"runinfo"in e&&n(1,r=e.runinfo)},[s,r,a,function(e){a=e,n(2,a)},()=>{n(2,a=!0)}]}class yt extends t{constructor(e){super(),n(this,e,xt,vt,s,{versions:0,runinfo:1})}}function Ct(e){let t,n,s;function r(t){e[12](t)}let a={$$slots:{default:[St]},$$scope:{ctx:e}};return void 0!==e[0]&&(a.isOpen=e[0]),t=new ce({props:a}),v.push(()=>x(t,"isOpen",r)),{c(){h(t.$$.fragment)},m(e,n){$(t,e,n),s=!0},p(e,s){const r={};8576&s&&(r.$$scope={dirty:s,ctx:e}),!n&&1&s&&(n=!0,r.isOpen=e[0],y(()=>n=!1)),t.$set(r)},i(e){s||(l(t.$$.fragment,e),s=!0)},o(e){o(t.$$.fragment,e),s=!1},d(e){g(t,e)}}}function St(e){let t,n;return t=new Fe({props:{toc:e[7],page:e[8]}}),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},p(e,n){const s={};128&n&&(s.toc=e[7]),256&n&&(s.page=e[8]),t.$set(s)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function wt(e){let t;const n=e[10].default,s=G(n,e,e[13],null);return{c(){s&&s.c()},m(e,n){s&&s.m(e,n),t=!0},p(e,r){s&&s.p&&(!t||8192&r)&&L(s,n,e,e[13],t?P(n,e[13],r,null):T(e[13]),null)},i(e){t||(l(s,e),t=!0)},o(e){o(s,e),t=!1},d(e){s&&s.d(e)}}}function Mt(e){let t,n;return t=new _e({props:{$$slots:{default:[wt]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},p(e,n){const s={};8192&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function Rt(e){let t,n;return t=new de({props:{$$slots:{default:[Mt]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},p(e,n){const s={};8192&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function Nt(e){let t,n,s,r;return t=new pe({props:{$$slots:{default:[Rt]},$$scope:{ctx:e}}}),s=new yt({props:{versions:e[3],runinfo:e[9]}}),{c(){h(t.$$.fragment),n=b(),h(s.$$.fragment)},m(e,a){$(t,e,a),d(e,n,a),$(s,e,a),r=!0},p(e,n){const r={};8192&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r);const a={};8&n&&(a.versions=e[3]),512&n&&(a.runinfo=e[9]),s.$set(a)},i(e){r||(l(t.$$.fragment,e),l(s.$$.fragment,e),r=!0)},o(e){o(t.$$.fragment,e),o(s.$$.fragment,e),r=!1},d(e){e&&a(n),g(t,e),g(s,e)}}}function Gt(e){let t,n,s,r,i,_;function u(t){e[11](t)}let m={logo:e[1],logotext:e[2],entries:e[4],pipeline_name:e[5]};void 0!==e[0]&&(m.isSideNavOpen=e[0]),t=new dt({props:m}),v.push(()=>x(t,"isSideNavOpen",u));let f=e[6]&&e[7].length>0&&Ct(e);return i=new le({props:{$$slots:{default:[Nt]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment),s=b(),f&&f.c(),r=b(),h(i.$$.fragment)},m(e,n){$(t,e,n),d(e,s,n),f&&f.m(e,n),d(e,r,n),$(i,e,n),_=!0},p(e,[s]){const a={};2&s&&(a.logo=e[1]),4&s&&(a.logotext=e[2]),16&s&&(a.entries=e[4]),32&s&&(a.pipeline_name=e[5]),!n&&1&s&&(n=!0,a.isSideNavOpen=e[0],y(()=>n=!1)),t.$set(a),e[6]&&e[7].length>0?f?(f.p(e,s),192&s&&l(f,1)):(f=Ct(e),f.c(),l(f,1),f.m(r.parentNode,r)):f&&(c(),o(f,1,1,()=>{f=null}),p());const d={};8712&s&&(d.$$scope={dirty:s,ctx:e}),i.$set(d)},i(e){_||(l(t.$$.fragment,e),l(f),l(i.$$.fragment,e),_=!0)},o(e){o(t.$$.fragment,e),o(f),o(i.$$.fragment,e),_=!1},d(e){e&&(a(s),a(r)),g(t,e),f&&f.d(e),g(i,e)}}}function Lt(e,t,n){let s,{$$slots:r={},$$scope:a}=t,{logo:i}=t,{logotext:o=null}=t,{versions:l}=t,{entries:c}=t,{pipeline_name:p}=t,{report_toc:d}=t,{toc:_}=t,{page:u}=t,{isSideNavOpen:f}=t,{runinfo:g}=t;return m(()=>{const e=new URL(window.location.href);let t;t=e.hash.length>0?document.querySelector(`li:has(>[href="${e.hash}"])`):document.querySelector("li:has(>.toc-current-page)"),t&&(t.classList.add("toc-active"),t.scrollIntoView());const n=document.querySelectorAll(".pipen-report-toc-anchor");s=new IntersectionObserver(e=>{e.forEach(e=>{const t=e.target.getAttribute("id"),n=document.querySelector(`li:has(>[href="#${t}"])`);n&&(e.isIntersecting?n.classList.add("toc-active"):n.classList.remove("toc-active"))})},{root:null,rootMargin:"0px",threshold:.5}),n.forEach(e=>{s.observe(e)})}),e.$$set=e=>{"logo"in e&&n(1,i=e.logo),"logotext"in e&&n(2,o=e.logotext),"versions"in e&&n(3,l=e.versions),"entries"in e&&n(4,c=e.entries),"pipeline_name"in e&&n(5,p=e.pipeline_name),"report_toc"in e&&n(6,d=e.report_toc),"toc"in e&&n(7,_=e.toc),"page"in e&&n(8,u=e.page),"isSideNavOpen"in e&&n(0,f=e.isSideNavOpen),"runinfo"in e&&n(9,g=e.runinfo),"$$scope"in e&&n(13,a=e.$$scope)},[f,i,o,l,c,p,d,_,u,g,r,function(e){f=e,n(0,f)},function(e){f=e,n(0,f)},a]}class Tt extends t{constructor(e){super(),n(this,e,Lt,Gt,s,{logo:1,logotext:2,versions:3,entries:4,pipeline_name:5,report_toc:6,toc:7,page:8,isSideNavOpen:0,runinfo:9})}}function Pt(e){let t;return{c(){t=M(e[2])},m(e,n){d(e,t,n)},p:R,d(e){e&&a(t)}}}function At(e){let t,n,s,r;return n=new ue({props:{icon:e[1],kind:"tertiary",href:e[3](),$$slots:{default:[Pt]},$$scope:{ctx:e}}}),{c(){t=_("div"),h(n.$$.fragment),u(t,"class",s=`pagenav ${e[0]}`)},m(e,s){d(e,t,s),$(n,t,null),r=!0},p(e,[a]){const i={};16&a&&(i.$$scope={dirty:a,ctx:e}),n.$set(i),(!r||1&a&&s!==(s=`pagenav ${e[0]}`))&&u(t,"class",s)},i(e){r||(l(n.$$.fragment,e),r=!0)},o(e){o(n.$$.fragment,e),r=!1},d(e){e&&a(t),g(n)}}}function It(e,t,n){let{dir:s}=t,r="up"===s?$e:he,a="up"===s?"Go to previous page":"Go to next page";return e.$$set=e=>{"dir"in e&&n(0,s=e.dir)},[s,r,a,()=>{let e=new URL(window.location.href);e.hash="";const t=e.searchParams.get("proc")||"_index",n=parseInt((t.match(/-(\d+)$/)||["-0","0"])[1]),r="up"===s?n-1:n+1,a=t.replace(/-\d+$/,""),i=0===r?a:`${a}-${r}`;return e.searchParams.set("proc",i),e.toString()}]}class Et extends t{constructor(e){super(),n(this,e,It,At,s,{dir:0})}}function Ut(e){let t,n;const s=[{tooltipText:e[0]},{icon:e[1]},{direction:"bottom"},{align:"start"},e[3]];let r={};for(let e=0;e<s.length;e+=1)r=A(r,s[e]);return t=new me({props:r}),t.$on("click",e[2]),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},p(e,[n]){const r=11&n?I(s,[1&n&&{tooltipText:e[0]},2&n&&{icon:e[1]},s[2],s[3],8&n&&E(e[3])]):{};t.$set(r)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function kt(e,t,n){const s=["src","tip","icon"];let r,a=U(t,s),{src:i=""}=t,{tip:o=""}=t,{icon:l="auto"}=t,c=i.split(".").at(-1).toLocaleLowerCase();if("string"==typeof l)switch("auto"===l&&(l=c),l){case"pdf":case"DocumentPdf":r=Pe;break;case"html":case"htm":case"ChartLineData":r=Te;break;case"png":case"Png":r=Le;break;case"jpg":case"jpeg":case"Jpg":r=Ge;break;case"gif":case"Gif":r=Ne;break;case"tif":case"tiff":case"Tif":r=Re;break;case"eps":case"DatabaseElastic":r=Me;break;case"mp4":case"avi":case"mov":case"mkv":case"webm":case"VideoPlayer":r=we;break;case"Image":r=Se;break;case"Data":case"DataVolume":r=Ce;break;case"Code":case"IbmWatsonxCodeAssistantForZRefactor":r=ye;break;case"svg":case"Svg":r=xe;break;case"zip":case"tar":case"gz":case"bz2":case"7z":case"rar":case"xz":case"Archive":r=ve;break;default:r=be}const p={...a,class:[a.class,"pipen-report-image-download"].filter(Boolean).join(" ")};return e.$$set=e=>{t=A(A({},t),k(e)),n(7,a=U(t,s)),"src"in e&&n(5,i=e.src),"tip"in e&&n(0,o=e.tip),"icon"in e&&n(4,l=e.icon)},[o,r,()=>{const e=document.createElement("a");e.href=i,e.download=i.split("/").pop(),e.click()},p,l,i]}class Bt extends t{constructor(e){super(),n(this,e,kt,Ut,s,{src:5,tip:0,icon:4})}}function Vt(e){let t,n,s;return{c(){t=M(e[1]),n=M("x"),s=M(e[2])},m(e,r){d(e,t,r),d(e,n,r),d(e,s,r)},p(e,n){2&n&&w(t,e[1]),4&n&&w(s,e[2])},d(e){e&&(a(t),a(n),a(s))}}}function Dt(e){let t,n;return t=new fe({props:{tooltipText:e[0],class:"pipen-report-image-dimension",align:"start",direction:"bottom",$$slots:{default:[Vt]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},p(e,[n]){const s={};1&n&&(s.tooltipText=e[0]),38&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function Ot(e,t,n){let{src:s=""}=t,{tip:r=""}=t,a=0,i=0,o=new Image;return o.src=s,o.onload=()=>{n(1,a=o.width),n(2,i=o.height)},e.$$set=e=>{"src"in e&&n(3,s=e.src),"tip"in e&&n(0,r=e.tip)},[r,a,i,s]}class Ft extends t{constructor(e){super(),n(this,e,Ot,Dt,s,{src:3,tip:0})}}const{Boolean:zt}=V;function Ht(e,t,n){const s=e.slice();return s[6]=t[n].src,s[7]=t[n].tip,s[8]=t[n].icon,s[10]=n,s}function jt(e){let t,n,s,u,m=r(e[0]),f=[];for(let t=0;t<m.length;t+=1)f[t]=Xt(Ht(e,m,t));const g=e=>o(f[e],1,1,()=>{f[e]=null});let $=[e[2],{role:"button"},{tabindex:"0"}],h={};for(let e=0;e<$.length;e+=1)h=A(h,$[e]);return{c(){t=_("div");for(let e=0;e<f.length;e+=1)f[e].c();D(t,h),S(t,"svelte-14lcnln",!0)},m(r,a){d(r,t,a);for(let e=0;e<f.length;e+=1)f[e]&&f[e].m(t,null);n=!0,s||(u=N(t,"mouseenter",e[4]),s=!0)},p(e,n){if(3&n){let s;for(m=r(e[0]),s=0;s<m.length;s+=1){const r=Ht(e,m,s);f[s]?(f[s].p(r,n),l(f[s],1)):(f[s]=Xt(r),f[s].c(),l(f[s],1),f[s].m(t,null))}for(c(),s=m.length;s<f.length;s+=1)g(s);p()}S(t,"svelte-14lcnln",!0)},i(e){if(!n){for(let e=0;e<m.length;e+=1)l(f[e]);n=!0}},o(e){f=f.filter(zt);for(let e=0;e<f.length;e+=1)o(f[e]);n=!1},d(e){e&&a(t),i(f,e),s=!1,u()}}}function qt(e){let t,n;return t=new Bt({props:{src:e[6],tip:e[7],icon:e[8]}}),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},p(e,n){const s={};1&n&&(s.src=e[6]),1&n&&(s.tip=e[7]),1&n&&(s.icon=e[8]),t.$set(s)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function Kt(e){let t,n;return t=new Ft({props:{src:e[1],tip:e[7]}}),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},p(e,n){const s={};2&n&&(s.src=e[1]),1&n&&(s.tip=e[7]),t.$set(s)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function Xt(e){let t,n,s,r;const i=[Kt,qt],_=[];function u(e,t){return"x"===e[6]?0:1}return t=u(e),n=_[t]=i[t](e),{c(){n.c(),s=f()},m(e,n){_[t].m(e,n),d(e,s,n),r=!0},p(e,r){let a=t;t=u(e),t===a?_[t].p(e,r):(c(),o(_[a],1,1,()=>{_[a]=null}),p(),n=_[t],n?n.p(e,r):(n=_[t]=i[t](e),n.c()),l(n,1),n.m(s.parentNode,s))},i(e){r||(l(n),r=!0)},o(e){o(n),r=!1},d(e){e&&a(s),_[t].d(e)}}}function Wt(e){let t,n,s=e[0].length>0&&jt(e);return{c(){s&&s.c(),t=f()},m(e,r){s&&s.m(e,r),d(e,t,r),n=!0},p(e,[n]){e[0].length>0?s?(s.p(e,n),1&n&&l(s,1)):(s=jt(e),s.c(),l(s,1),s.m(t.parentNode,t)):s&&(c(),o(s,1,1,()=>{s=null}),p())},i(e){n||(l(s),n=!0)},o(e){o(s),n=!1},d(e){e&&a(t),s&&s.d(e)}}}function Zt(e,t,n){const s=["data","dsrc"];let r=U(t,s),{data:a=[]}=t,{dsrc:i=""}=t;Array.isArray(a)||(a=[a]),a=[{src:"x",tip:"widthxheight"},...a],a=a.map(e=>"string"==typeof e?{src:e,tip:`Download the ${e.split(".").at(-1)} format.`,icon:"auto"}:(e.tip=e.tip||`Download the ${e.src.split(".").at(-1)} format.`,e.icon=e.icon||"auto",e));const o={...r,class:[r.class,"pipen-report-image-download-list"].filter(Boolean).join(" ")},l=B();return e.$$set=e=>{t=A(A({},t),k(e)),n(5,r=U(t,s)),"data"in e&&n(0,a=e.data),"dsrc"in e&&n(1,i=e.dsrc)},[a,i,o,l,()=>l("mouseenter")]}class Yt extends t{constructor(e){super(),n(this,e,Zt,Wt,s,{data:0,dsrc:1})}}const Jt=e=>({}),Qt=e=>({}),en=e=>({}),tn=e=>({}),nn=e=>({}),sn=e=>({}),rn=e=>({}),an=e=>({});function on(e){let t,n;return t=new ge({props:{ratio:e[6],$$slots:{default:[un]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},p(e,n){const s={};64&n&&(s.ratio=e[6]),33556159&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function ln(e){let t,n,s,r,i=e[2]&&mn(e),_=e[0]&&fn(e),u=e[1]&&$n(e);return{c(){i&&i.c(),t=b(),_&&_.c(),n=b(),u&&u.c(),s=f()},m(e,a){i&&i.m(e,a),d(e,t,a),_&&_.m(e,a),d(e,n,a),u&&u.m(e,a),d(e,s,a),r=!0},p(e,r){e[2]?i?(i.p(e,r),4&r&&l(i,1)):(i=mn(e),i.c(),l(i,1),i.m(t.parentNode,t)):i&&(c(),o(i,1,1,()=>{i=null}),p()),e[0]?_?(_.p(e,r),1&r&&l(_,1)):(_=fn(e),_.c(),l(_,1),_.m(n.parentNode,n)):_&&(c(),o(_,1,1,()=>{_=null}),p()),e[1]?u?(u.p(e,r),2&r&&l(u,1)):(u=$n(e),u.c(),l(u,1),u.m(s.parentNode,s)):u&&(c(),o(u,1,1,()=>{u=null}),p())},i(e){r||(l(i),l(_),l(u),r=!0)},o(e){o(i),o(_),o(u),r=!1},d(e){e&&(a(t),a(n),a(s)),i&&i.d(e),_&&_.d(e),u&&u.d(e)}}}function cn(e){let t;const n=e[15].loading,s=G(n,e,e[25],tn);return{c(){s&&s.c()},m(e,n){s&&s.m(e,n),t=!0},p(e,r){s&&s.p&&(!t||33554432&r)&&L(s,n,e,e[25],t?P(n,e[25],r,en):T(e[25]),tn)},i(e){t||(l(s,e),t=!0)},o(e){o(s,e),t=!1},d(e){s&&s.d(e)}}}function pn(e){let t,n,s,r,i,u,m,g,$=[e[12],{src:n=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],h={};for(let e=0;e<$.length;e+=1)h=A(h,$[e]);let v=e[9]&&dn(e);return{c(){t=_("img"),r=b(),v&&v.c(),i=f(),D(t,h)},m(n,s){d(n,t,s),d(n,r,s),v&&v.m(n,s),d(n,i,s),u=!0,m||(g=[N(t,"click",e[11]),N(t,"mouseenter",e[19]),N(t,"mouseleave",e[20])],m=!0)},p(s,r){D(t,h=I($,[(e=s)[12],(!u||8&r&&!H(t.src,n=e[3]))&&{src:n},(!u||16&r)&&{alt:e[4]},{title:"Click to zoom in ..."}])),e[9]?v?(v.p(e,r),512&r&&l(v,1)):(v=dn(e),v.c(),l(v,1),v.m(i.parentNode,i)):v&&(c(),o(v,1,1,()=>{v=null}),p())},i(n){u||(n&&z(()=>{u&&(s||(s=O(t,F,{duration:e[7]?vn:0},!0)),s.run(1))}),l(v),u=!0)},o(n){n&&(s||(s=O(t,F,{duration:e[7]?vn:0},!1)),s.run(0)),o(v),u=!1},d(e){e&&(a(t),a(r),a(i)),e&&s&&s.end(),v&&v.d(e),m=!1,j(g)}}}function dn(e){let t,n;return t=new Yt({props:{data:e[5]}}),t.$on("mouseenter",e[21]),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},p(e,n){const s={};32&n&&(s.data=e[5]),t.$set(s)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function _n(e){let t;const n=e[15].error,s=G(n,e,e[25],Qt);return{c(){s&&s.c()},m(e,n){s&&s.m(e,n),t=!0},p(e,r){s&&s.p&&(!t||33554432&r)&&L(s,n,e,e[25],t?P(n,e[25],r,Jt):T(e[25]),Qt)},i(e){t||(l(s,e),t=!0)},o(e){o(s,e),t=!1},d(e){s&&s.d(e)}}}function un(e){let t,n,s,r,i=e[2]&&cn(e),_=e[0]&&pn(e),u=e[1]&&_n(e);return{c(){i&&i.c(),t=b(),_&&_.c(),n=b(),u&&u.c(),s=f()},m(e,a){i&&i.m(e,a),d(e,t,a),_&&_.m(e,a),d(e,n,a),u&&u.m(e,a),d(e,s,a),r=!0},p(e,r){e[2]?i?(i.p(e,r),4&r&&l(i,1)):(i=cn(e),i.c(),l(i,1),i.m(t.parentNode,t)):i&&(c(),o(i,1,1,()=>{i=null}),p()),e[0]?_?(_.p(e,r),1&r&&l(_,1)):(_=pn(e),_.c(),l(_,1),_.m(n.parentNode,n)):_&&(c(),o(_,1,1,()=>{_=null}),p()),e[1]?u?(u.p(e,r),2&r&&l(u,1)):(u=_n(e),u.c(),l(u,1),u.m(s.parentNode,s)):u&&(c(),o(u,1,1,()=>{u=null}),p())},i(e){r||(l(i),l(_),l(u),r=!0)},o(e){o(i),o(_),o(u),r=!1},d(e){e&&(a(t),a(n),a(s)),i&&i.d(e),_&&_.d(e),u&&u.d(e)}}}function mn(e){let t;const n=e[15].loading,s=G(n,e,e[25],an);return{c(){s&&s.c()},m(e,n){s&&s.m(e,n),t=!0},p(e,r){s&&s.p&&(!t||33554432&r)&&L(s,n,e,e[25],t?P(n,e[25],r,rn):T(e[25]),an)},i(e){t||(l(s,e),t=!0)},o(e){o(s,e),t=!1},d(e){s&&s.d(e)}}}function fn(e){let t,n,s,r,i,u,m,g,$=[e[12],{src:n=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],h={};for(let e=0;e<$.length;e+=1)h=A(h,$[e]);let v=e[9]&&gn(e);return{c(){t=_("img"),r=b(),v&&v.c(),i=f(),D(t,h)},m(n,s){d(n,t,s),d(n,r,s),v&&v.m(n,s),d(n,i,s),u=!0,m||(g=[N(t,"click",e[11]),N(t,"mouseenter",e[16]),N(t,"mouseleave",e[17])],m=!0)},p(s,r){D(t,h=I($,[(e=s)[12],(!u||8&r&&!H(t.src,n=e[3]))&&{src:n},(!u||16&r)&&{alt:e[4]},{title:"Click to zoom in ..."}])),e[9]?v?(v.p(e,r),512&r&&l(v,1)):(v=gn(e),v.c(),l(v,1),v.m(i.parentNode,i)):v&&(c(),o(v,1,1,()=>{v=null}),p())},i(n){u||(n&&z(()=>{u&&(s||(s=O(t,F,{duration:e[7]?vn:0},!0)),s.run(1))}),l(v),u=!0)},o(n){n&&(s||(s=O(t,F,{duration:e[7]?vn:0},!1)),s.run(0)),o(v),u=!1},d(e){e&&(a(t),a(r),a(i)),e&&s&&s.end(),v&&v.d(e),m=!1,j(g)}}}function gn(e){let t,n;return t=new Yt({props:{dsrc:e[3],data:e[5]}}),t.$on("mouseenter",e[18]),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},p(e,n){const s={};8&n&&(s.dsrc=e[3]),32&n&&(s.data=e[5]),t.$set(s)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function $n(e){let t;const n=e[15].error,s=G(n,e,e[25],sn);return{c(){s&&s.c()},m(e,n){s&&s.m(e,n),t=!0},p(e,r){s&&s.p&&(!t||33554432&r)&&L(s,n,e,e[25],t?P(n,e[25],r,nn):T(e[25]),sn)},i(e){t||(l(s,e),t=!0)},o(e){o(s,e),t=!1},d(e){s&&s.d(e)}}}function hn(e){let t,n,s,r;return{c(){t=_("img"),u(t,"class","pipen-report-image-lg"),H(t.src,n=e[3])||u(t,"src",n),u(t,"alt",e[4])},m(e,n){d(e,t,n),r=!0},p(s,a){e=s,(!r||8&a&&!H(t.src,n=e[3]))&&u(t,"src",n),(!r||16&a)&&u(t,"alt",e[4])},i(n){r||(n&&z(()=>{r&&(s||(s=O(t,F,{duration:e[7]?vn:0},!0)),s.run(1))}),r=!0)},o(n){n&&(s||(s=O(t,F,{duration:e[7]?vn:0},!1)),s.run(0)),r=!1},d(e){e&&a(t),e&&s&&s.end()}}}function bn(e){let t,n,s,r,i,_;const u=[ln,on],m=[];function f(e,t){return void 0===e[6]?0:1}function C(t){e[22](t)}t=f(e),n=m[t]=u[t](e);let S={modalHeading:e[4],primaryButtonText:"Open in new window ...",secondaryButtonText:"Close",$$slots:{default:[hn]},$$scope:{ctx:e}};return void 0!==e[8]&&(S.open=e[8]),r=new ne({props:S}),v.push(()=>x(r,"open",C)),r.$on("click:button--secondary",e[23]),r.$on("click:button--primary",e[24]),{c(){n.c(),s=b(),h(r.$$.fragment)},m(e,n){m[t].m(e,n),d(e,s,n),$(r,e,n),_=!0},p(e,[a]){let d=t;t=f(e),t===d?m[t].p(e,a):(c(),o(m[d],1,1,()=>{m[d]=null}),p(),n=m[t],n?n.p(e,a):(n=m[t]=u[t](e),n.c()),l(n,1),n.m(s.parentNode,s));const _={};16&a&&(_.modalHeading=e[4]),33554584&a&&(_.$$scope={dirty:a,ctx:e}),!i&&256&a&&(i=!0,_.open=e[8],y(()=>i=!1)),r.$set(_)},i(e){_||(l(n),l(r.$$.fragment,e),_=!0)},o(e){o(n),o(r.$$.fragment,e),_=!1},d(e){e&&a(s),m[t].d(e),g(r,e)}}}const vn=110;function xn(e,t,n){const s=["src","alt","download","ratio","loading","loaded","error","fadeIn","border","loadImage"];let r=U(t,s),{$$slots:a={},$$scope:i}=t,{src:o=""}=t,{alt:l=""}=t,{download:c=[]}=t,{ratio:p}=t,{loading:d=!1}=t,{loaded:_=!1}=t,{error:u=!1}=t,{fadeIn:f=!1}=t,{border:g=!0}=t,$=!1,h=!1;const b=e=>{null!=y&&(y=null),n(0,_=!1),n(1,u=!1),y=new Image,y.src=e||o,y.onload=()=>n(0,_=!0),y.onerror=()=>n(1,u=!0)},v=B();let x,y=null;m(()=>()=>y=null);const C={...r,class:[r.class,"pipen-report-image",g?"":"pipen-report-image-noborder"].filter(Boolean).join(" ")};return e.$$set=e=>{t=A(A({},t),k(e)),n(28,r=U(t,s)),"src"in e&&n(3,o=e.src),"alt"in e&&n(4,l=e.alt),"download"in e&&n(5,c=e.download),"ratio"in e&&n(6,p=e.ratio),"loading"in e&&n(2,d=e.loading),"loaded"in e&&n(0,_=e.loaded),"error"in e&&n(1,u=e.error),"fadeIn"in e&&n(7,f=e.fadeIn),"border"in e&&n(13,g=e.border),"$$scope"in e&&n(25,i=e.$$scope)},e.$$.update=()=>{3&e.$$.dirty&&n(2,d=!_&&!u),8&e.$$.dirty&&o&&"undefined"!=typeof window&&b(),1&e.$$.dirty&&_&&v("load"),2&e.$$.dirty&&u&&v("error")},[_,u,d,o,l,c,p,f,$,h,x,()=>{n(8,$=!0)},C,g,b,a,()=>{n(9,h=!0)},()=>{n(10,x=setTimeout(()=>{n(9,h=!1)},100))},()=>{clearTimeout(x),n(9,h=!0)},()=>{n(9,h=!0)},()=>{n(10,x=setTimeout(()=>{n(9,h=!1)},100))},()=>{clearTimeout(x),n(9,h=!0)},function(e){$=e,n(8,$)},()=>{n(8,$=!1)},()=>{n(8,$=!1),window.open(o)},i]}class yn extends t{constructor(e){super(),n(this,e,xn,bn,s,{src:3,alt:4,download:5,ratio:6,loading:2,loaded:0,error:1,fadeIn:7,border:13,loadImage:14})}get loadImage(){return this.$$.ctx[14]}}function Cn(e){let t,n,s=[e[2]],r={};for(let e=0;e<s.length;e+=1)r=A(r,s[e]);return{c(){t=_("div"),n=M("Image loading ..."),D(t,r)},m(e,s){d(e,t,s),C(t,n)},p:R,d(e){e&&a(t)}}}function Sn(e){let t,n,s=[e[3]],r={};for(let e=0;e<s.length;e+=1)r=A(r,s[e]);return{c(){t=_("div"),n=M("Image loading error!"),D(t,r)},m(e,s){d(e,t,s),C(t,n)},p:R,d(e){e&&a(t)}}}function wn(e){let t,n,s;const r=[{src:e[1]},e[4]];let i={$$slots:{error:[Sn],loading:[Cn]},$$scope:{ctx:e}};for(let e=0;e<r.length;e+=1)i=A(i,r[e]);n=new yn({props:i});let c=[e[0]],p={};for(let e=0;e<c.length;e+=1)p=A(p,c[e]);return{c(){t=_("div"),h(n.$$.fragment),D(t,p)},m(e,r){d(e,t,r),$(n,t,null),s=!0},p(e,[s]){const a=18&s?I(r,[2&s&&{src:e[1]},16&s&&E(e[4])]):{};256&s&&(a.$$scope={dirty:s,ctx:e}),n.$set(a),D(t,p=I(c,[1&s&&e[0]]))},i(e){s||(l(n.$$.fragment,e),s=!0)},o(e){o(n.$$.fragment,e),s=!1},d(e){e&&a(t),g(n)}}}function Mn(e,t,n){const s=["frameProps","width","height","src"];let r=U(t,s),{frameProps:a={class:"pipen-report-image-frame"}}=t,{width:i=300}=t,{height:o=200}=t,{src:l}=t;a||(a={"style:position":"relative"});let c=`width: ${i}px; height: ${o}px; aspect-ratio: ${i} / ${o}`,p={...r,class:[r.class,"pipen-report-image-loading"].filter(Boolean).join(" "),style:r.style?`${r.style}; ${c}`:c},d={...p,class:[p.class,"error"].join(" ")};return e.$$set=e=>{t=A(A({},t),k(e)),n(4,r=U(t,s)),"frameProps"in e&&n(0,a=e.frameProps),"width"in e&&n(5,i=e.width),"height"in e&&n(6,o=e.height),"src"in e&&n(1,l=e.src)},[a,l,p,d,r,i,o]}let Rn=class extends t{constructor(e){super(),n(this,e,Mn,wn,s,{frameProps:0,width:5,height:6,src:1})}};function Nn(e){let t,n;return{c(){t=_("div"),n=M(e[0]),u(t,"class","pipen-report-descr-title svelte-1xzxuva")},m(e,s){d(e,t,s),C(t,n)},p(e,t){1&t&&w(n,e[0])},d(e){e&&a(t)}}}function Gn(e){let t,n,s,r=e[0]&&Nn(e);const i=e[3].default,c=G(i,e,e[2],null);let p=[e[1]],u={};for(let e=0;e<p.length;e+=1)u=A(u,p[e]);return{c(){t=_("div"),r&&r.c(),n=b(),c&&c.c(),D(t,u),S(t,"svelte-1xzxuva",!0)},m(e,a){d(e,t,a),r&&r.m(t,null),C(t,n),c&&c.m(t,null),s=!0},p(e,[a]){e[0]?r?r.p(e,a):(r=Nn(e),r.c(),r.m(t,n)):r&&(r.d(1),r=null),c&&c.p&&(!s||4&a)&&L(c,i,e,e[2],s?P(i,e[2],a,null):T(e[2]),null),D(t,u=I(p,[2&a&&e[1]])),S(t,"svelte-1xzxuva",!0)},i(e){s||(l(c,e),s=!0)},o(e){o(c,e),s=!1},d(e){e&&a(t),r&&r.d(),c&&c.d(e)}}}function Ln(e,t,n){const s=["title"];let r=U(t,s),{$$slots:a={},$$scope:i}=t,{title:o}=t,{klass:l,...c}=r;return l=l?`pipen-report-descr ${l}`:"pipen-report-descr",c||(c={}),c.class=l,e.$$set=e=>{t=A(A({},t),k(e)),n(5,r=U(t,s)),"title"in e&&n(0,o=e.title),"$$scope"in e&&n(2,i=e.$$scope)},[o,c,i,a]}class Tn extends t{constructor(e){super(),n(this,e,Ln,Gn,s,{title:0})}}function Pn(e){let t;return{c(){t=M("Dimensionality reduction plot for seurat_clusters")},m(e,n){d(e,t,n)},d(e){e&&a(t)}}}function An(e){let t;return{c(){t=M("Dimensionality reduction plot for VDJ_Presence")},m(e,n){d(e,t,n)},d(e){e&&a(t)}}}function In(e){let t;return{c(){t=M("Dimensionality reduction plot for seurat_clusters")},m(e,n){d(e,t,n)},d(e){e&&a(t)}}}function En(e){let t;return{c(){t=M("The bar plot shows the distribution of cells across categories defined by 'seurat_clusters'. The values represent the number of cells.")},m(e,n){d(e,t,n)},d(e){e&&a(t)}}}function Un(e){let t;return{c(){t=M("The bar plot shows the distribution of cells across categories defined by 'seurat_clusters', grouped by Sample. The values represent the number of cells.")},m(e,n){d(e,t,n)},d(e){e&&a(t)}}}function kn(e){let t;return{c(){t=M("Number of genes expressed in each cluster")},m(e,n){d(e,t,n)},d(e){e&&a(t)}}}function Bn(e){let t;return{c(){t=M("The plot shows the distribution or pattern of the specified features (TCF7, SELL, LEF1, CCR7, IL2, GZMA, GNLY, PRF1, GZMB, GZMK, IFNG, NKG7, TNF, CX3CR1, GZMH, FGFBP2, CD28, TNFRSF14, ICOS, TNFRSF9, LAG3, TIGIT, PDCD1, HAVCR2, CTLA4, EOMES, HOPX, TBX21, ZEB2, HIF1A, ID2, TOX, IL2RA, FOXP3, IKZF2, CD38, ENTPD1, ITGAE, CD4, CD8A) across cells. The plot type is 'heatmap', arranged as a heatmap by rows_name and other grouping variables.")},m(e,n){d(e,t,n)},d(e){e&&a(t)}}}function Vn(e){let t;return{c(){t=M("The plot shows the distribution or pattern of the specified features (TCF7, SELL, LEF1, CCR7, IL2, GZMA, GNLY, PRF1, GZMB, GZMK, IFNG, NKG7, TNF, CX3CR1, GZMH, FGFBP2, CD28, TNFRSF14, ICOS, TNFRSF9, LAG3, TIGIT, PDCD1, HAVCR2, CTLA4, EOMES, HOPX, TBX21, ZEB2, HIF1A, ID2, TOX, IL2RA, FOXP3, IKZF2, CD38, ENTPD1, ITGAE, CD4, CD8A) across cells. The plot type is 'dim', displaying the features on a dimensional reduction embedding. The reduction used is \"umap\", using dimensions 1,2")},m(e,n){d(e,t,n)},d(e){e&&a(t)}}}function Dn(e){let t,n,s,r,i,c,p,m,f,v,x,y,C,S,w,M,R,N,G,L,T,P,A,I,E,U,k,B,V,D,O,F,z,H,j,q,K,X,W,Z,Y,J,Q,ee,te,ne,se,re,ae,ie,oe,le,ce,pe,de,_e;return r=new Tn({props:{$$slots:{default:[Pn]},$$scope:{ctx:e}}}),c=new Rn({props:{src:"../SeuratClusterStats/samples.scRep.cluster_stats/dimplots/Dimensional-reduction-plot.dim.png",download:[{src:"../SeuratClusterStats/samples.scRep.cluster_stats/dimplots/Dimensional-reduction-plot.dim.pdf"}],width:650,height:550}}),x=new Tn({props:{$$slots:{default:[An]},$$scope:{ctx:e}}}),C=new Rn({props:{src:"../SeuratClusterStats/samples.scRep.cluster_stats/dimplots/VDJ-Presence.dim.png",download:[{src:"../SeuratClusterStats/samples.scRep.cluster_stats/dimplots/VDJ-Presence.dim.pdf"}],width:650,height:550}}),N=new Tn({props:{$$slots:{default:[In]},$$scope:{ctx:e}}}),L=new Rn({props:{src:"../SeuratClusterStats/samples.scRep.cluster_stats/dimplots/UMAP.dim.png",download:[{src:"../SeuratClusterStats/samples.scRep.cluster_stats/dimplots/UMAP.dim.pdf"}],width:650,height:550}}),E=new Tn({props:{$$slots:{default:[En]},$$scope:{ctx:e}}}),k=new Rn({props:{src:"../SeuratClusterStats/samples.scRep.cluster_stats/stats/Number-of-cells-in-each-cluster-Bar-Chart-.png",width:600,height:450}}),F=new Tn({props:{$$slots:{default:[Un]},$$scope:{ctx:e}}}),H=new Rn({props:{src:"../SeuratClusterStats/samples.scRep.cluster_stats/stats/Number-of-cells-in-each-cluster-by-Sample-Bar-Chart-.png",width:640,height:450}}),W=new Tn({props:{$$slots:{default:[kn]},$$scope:{ctx:e}}}),Y=new Rn({props:{src:"../SeuratClusterStats/samples.scRep.cluster_stats/ngenes/Number-of-genes-expressed-in-each-cluster.png",width:1e3,height:800}}),ne=new Tn({props:{$$slots:{default:[Bn]},$$scope:{ctx:e}}}),re=new Rn({props:{src:"../SeuratClusterStats/samples.scRep.cluster_stats/features/T-cell-function-associated-genes.png",width:1097,height:1450}}),ce=new Tn({props:{$$slots:{default:[Vn]},$$scope:{ctx:e}}}),de=new Rn({props:{src:"../SeuratClusterStats/samples.scRep.cluster_stats/features/T-cell-function-associated-genes-feature-plots-.png",width:3770,height:3300}}),{c(){t=_("h1"),t.textContent="Dimensional reduction plot",n=_("a"),n.innerHTML="",s=b(),h(r.$$.fragment),i=b(),h(c.$$.fragment),p=b(),m=_("h1"),m.textContent="VDJ Presence",f=_("a"),f.innerHTML="",v=b(),h(x.$$.fragment),y=b(),h(C.$$.fragment),S=b(),w=_("h1"),w.textContent="UMAP",M=_("a"),M.innerHTML="",R=b(),h(N.$$.fragment),G=b(),h(L.$$.fragment),T=b(),P=_("h1"),P.textContent="Number of cells in each cluster (Bar Chart)",A=_("a"),A.innerHTML="",I=b(),h(E.$$.fragment),U=b(),h(k.$$.fragment),B=b(),V=_("h1"),V.textContent="Number of cells in each cluster by Sample (Bar Chart)",D=_("a"),D.innerHTML="",O=b(),h(F.$$.fragment),z=b(),h(H.$$.fragment),j=b(),q=_("h1"),q.textContent="Number of genes expressed in each cluster",K=_("a"),K.innerHTML="",X=b(),h(W.$$.fragment),Z=b(),h(Y.$$.fragment),J=b(),Q=_("h1"),Q.textContent="T cell function-associated genes",ee=_("a"),ee.innerHTML="",te=b(),h(ne.$$.fragment),se=b(),h(re.$$.fragment),ae=b(),ie=_("h1"),ie.textContent="T cell function-associated genes (feature plots)",oe=_("a"),oe.innerHTML="",le=b(),h(ce.$$.fragment),pe=b(),h(de.$$.fragment),u(n,"id","prt-h1-0-dimensional-reduction-plot"),u(n,"class","pipen-report-toc-anchor"),u(f,"id","prt-h1-1-vdj-presence"),u(f,"class","pipen-report-toc-anchor"),u(M,"id","prt-h1-2-umap"),u(M,"class","pipen-report-toc-anchor"),u(A,"id","prt-h1-3-number-of-cells-in-each-cluster-bar-chart"),u(A,"class","pipen-report-toc-anchor"),u(D,"id","prt-h1-4-number-of-cells-in-each-cluster-by-sample-bar-chart"),u(D,"class","pipen-report-toc-anchor"),u(K,"id","prt-h1-5-number-of-genes-expressed-in-each-cluster"),u(K,"class","pipen-report-toc-anchor"),u(ee,"id","prt-h1-6-t-cell-function-associated-genes"),u(ee,"class","pipen-report-toc-anchor"),u(oe,"id","prt-h1-7-t-cell-function-associated-genes-feature-plots"),u(oe,"class","pipen-report-toc-anchor")},m(e,a){d(e,t,a),d(e,n,a),d(e,s,a),$(r,e,a),d(e,i,a),$(c,e,a),d(e,p,a),d(e,m,a),d(e,f,a),d(e,v,a),$(x,e,a),d(e,y,a),$(C,e,a),d(e,S,a),d(e,w,a),d(e,M,a),d(e,R,a),$(N,e,a),d(e,G,a),$(L,e,a),d(e,T,a),d(e,P,a),d(e,A,a),d(e,I,a),$(E,e,a),d(e,U,a),$(k,e,a),d(e,B,a),d(e,V,a),d(e,D,a),d(e,O,a),$(F,e,a),d(e,z,a),$(H,e,a),d(e,j,a),d(e,q,a),d(e,K,a),d(e,X,a),$(W,e,a),d(e,Z,a),$(Y,e,a),d(e,J,a),d(e,Q,a),d(e,ee,a),d(e,te,a),$(ne,e,a),d(e,se,a),$(re,e,a),d(e,ae,a),d(e,ie,a),d(e,oe,a),d(e,le,a),$(ce,e,a),d(e,pe,a),$(de,e,a),_e=!0},p(e,[t]){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),r.$set(n);const s={};1&t&&(s.$$scope={dirty:t,ctx:e}),x.$set(s);const a={};1&t&&(a.$$scope={dirty:t,ctx:e}),N.$set(a);const i={};1&t&&(i.$$scope={dirty:t,ctx:e}),E.$set(i);const o={};1&t&&(o.$$scope={dirty:t,ctx:e}),F.$set(o);const l={};1&t&&(l.$$scope={dirty:t,ctx:e}),W.$set(l);const c={};1&t&&(c.$$scope={dirty:t,ctx:e}),ne.$set(c);const p={};1&t&&(p.$$scope={dirty:t,ctx:e}),ce.$set(p)},i(e){_e||(l(r.$$.fragment,e),l(c.$$.fragment,e),l(x.$$.fragment,e),l(C.$$.fragment,e),l(N.$$.fragment,e),l(L.$$.fragment,e),l(E.$$.fragment,e),l(k.$$.fragment,e),l(F.$$.fragment,e),l(H.$$.fragment,e),l(W.$$.fragment,e),l(Y.$$.fragment,e),l(ne.$$.fragment,e),l(re.$$.fragment,e),l(ce.$$.fragment,e),l(de.$$.fragment,e),_e=!0)},o(e){o(r.$$.fragment,e),o(c.$$.fragment,e),o(x.$$.fragment,e),o(C.$$.fragment,e),o(N.$$.fragment,e),o(L.$$.fragment,e),o(E.$$.fragment,e),o(k.$$.fragment,e),o(F.$$.fragment,e),o(H.$$.fragment,e),o(W.$$.fragment,e),o(Y.$$.fragment,e),o(ne.$$.fragment,e),o(re.$$.fragment,e),o(ce.$$.fragment,e),o(de.$$.fragment,e),_e=!1},d(e){e&&(a(t),a(n),a(s),a(i),a(p),a(m),a(f),a(v),a(y),a(S),a(w),a(M),a(R),a(G),a(T),a(P),a(A),a(I),a(U),a(B),a(V),a(D),a(O),a(z),a(j),a(q),a(K),a(X),a(Z),a(J),a(Q),a(ee),a(te),a(se),a(ae),a(ie),a(oe),a(le),a(pe)),g(r,e),g(c,e),g(x,e),g(C,e),g(N,e),g(L,e),g(E,e),g(k,e),g(F,e),g(H,e),g(W,e),g(Y,e),g(ne,e),g(re,e),g(ce,e),g(de,e)}}}class On extends t{constructor(e){super(),n(this,e,null,Dn,s,{})}}var Fn=[{slug:"prt-h1-0-dimensional-reduction-plot",text:"Dimensional reduction plot",children:[],page:0},{slug:"prt-h1-1-vdj-presence",text:"VDJ Presence",children:[],page:0},{slug:"prt-h1-2-umap",text:"UMAP",children:[],page:0},{slug:"prt-h1-3-number-of-cells-in-each-cluster-bar-chart",text:"Number of cells in each cluster (Bar Chart)",children:[],page:0},{slug:"prt-h1-4-number-of-cells-in-each-cluster-by-sample-bar-chart",text:"Number of cells in each cluster by Sample (Bar Chart)",children:[],page:0},{slug:"prt-h1-5-number-of-genes-expressed-in-each-cluster",text:"Number of genes expressed in each cluster",children:[],page:0},{slug:"prt-h1-6-t-cell-function-associated-genes",text:"T cell function-associated genes",children:[],page:0},{slug:"prt-h1-7-t-cell-function-associated-genes-feature-plots",text:"T cell function-associated genes (feature plots)",children:[],page:0},{slug:"prt-h1-8-t-cell-function-associated-genes-violin-plots",text:"T cell function-associated genes (violin plots)",children:[],page:1},{slug:"prt-h1-9-gene-signatures",text:"Gene signatures",children:[],page:1}],zn={pipeline:{name:"Immunopipe"},versions:'\n    <a href="https://github.com/pwwang/pipen" target="_blank">\n        pipen v1.1.8\n    </a> and\n    <a href="https://github.com/pwwang/pipen-report" target="_blank">\n        pipen-report v1.1.0\n    </a>\n',entries:[{name:"SampleInfo",desc:"List sample information and perform statistics",npages:1,report_toc:!0,order:0,runinfo:{session:"# Generated by pipen_runinfo v%(version)s\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 12 (bookworm)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.30.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n[1] plotthis_0.10.0      biopipen.utils_0.3.5 gglogger_0.1.7      \n[4] dplyr_1.1.4          rlang_1.1.6         \n\nloaded via a namespace (and not attached):\n  [1] cubature_2.0.4.6            RcppAnnoy_0.0.22           \n  [3] splines_4.3.3               later_1.4.4                \n  [5] bitops_1.0-9                tibble_3.3.0               \n  [7] polyclip_1.10-7             enrichit_0.0.4             \n  [9] fastDummies_1.7.5           lifecycle_1.0.4            \n [11] globals_0.18.0              lattice_0.22-7             \n [13] MASS_7.3-60.0.1             magrittr_2.0.3             \n [15] plotly_4.11.0               httpuv_1.6.16              \n [17] Seurat_5.3.0                sctransform_0.4.2          \n [19] spam_2.11-1                 sp_2.2-0                   \n [21] spatstat.sparse_3.1-0       reticulate_1.43.0          \n [23] cowplot_1.2.0               pbapply_1.7-4              \n [25] RColorBrewer_1.1-3          abind_1.4-5                \n [27] zlibbioc_1.48.0             Rtsne_0.17                 \n [29] GenomicRanges_1.54.1        purrr_1.1.0                \n [31] ggraph_2.2.1                BiocGenerics_0.48.1        \n [33] RCurl_1.98-1.17             tweenr_2.0.3               \n [35] evmix_2.12                  circlize_0.4.16            \n [37] GenomeInfoDbData_1.2.11     IRanges_2.36.0             \n [39] S4Vectors_0.40.2            ggrepel_0.9.6              \n [41] irlba_2.3.5.1               listenv_0.9.1              \n [43] spatstat.utils_3.1-5        iNEXT_3.0.2                \n [45] MatrixModels_0.5-4          goftest_1.2-3              \n [47] RSpectra_0.16-2             scRepertoire_2.2.1         \n [49] spatstat.random_3.4-1       fitdistrplus_1.2-4         \n [51] parallelly_1.45.1           codetools_0.2-20           \n [53] DelayedArray_0.28.0         xml2_1.4.0                 \n [55] ggforce_0.5.0               tidyselect_1.2.1           \n [57] shape_1.4.6.1               farver_2.1.2               \n [59] viridis_0.6.5               matrixStats_1.5.0          \n [61] stats4_4.3.3                spatstat.explore_3.5-2     \n [63] jsonlite_2.0.0              ellipsis_0.3.2             \n [65] tidygraph_1.3.0             progressr_0.15.1           \n [67] ggridges_0.5.7              ggalluvial_0.12.5          \n [69] survival_3.8-3              tools_4.3.3                \n [71] ggnewscale_0.5.2            stringdist_0.9.15          \n [73] ica_1.0-3                   Rcpp_1.1.0                 \n [75] glue_1.8.0                  gridExtra_2.3              \n [77] SparseArray_1.2.2           qvalue_2.34.0              \n [79] MatrixGenerics_1.14.0       GenomeInfoDb_1.38.1        \n [81] withr_3.0.2                 fastmap_1.2.0              \n [83] fansi_1.0.6                 SparseM_1.84-2             \n [85] ttservice_0.5.3             digest_0.6.37              \n [87] R6_2.6.1                    mime_0.13                  \n [89] colorspace_2.1-1            scattermore_1.2            \n [91] tensor_1.5.1                dichromat_2.0-0.1          \n [93] spatstat.data_3.1-8         tidyr_1.3.1                \n [95] generics_0.1.4              data.table_1.17.8          \n [97] graphlayouts_1.2.2          httr_1.4.7                 \n [99] htmlwidgets_1.6.4           S4Arrays_1.2.0             \n[101] uwot_0.2.3                  pkgconfig_2.0.3            \n[103] gtable_0.3.6                lmtest_0.9-40              \n[105] SingleCellExperiment_1.24.0 XVector_0.42.0             \n[107] htmltools_0.5.8.1           dotCall64_1.2              \n[109] SeuratObject_5.2.0          scales_1.4.0               \n[111] Biobase_2.62.0              png_0.1-8                  \n[113] spatstat.univar_3.1-4       ggdendro_0.2.0             \n[115] rjson_0.2.23                reshape2_1.4.4             \n[117] nlme_3.1-168                cachem_1.1.0               \n[119] zoo_1.8-14                  GlobalOptions_0.1.2        \n[121] stringr_1.5.2               KernSmooth_2.23-26         \n[123] parallel_4.3.3              miniUI_0.1.2               \n[125] pillar_1.11.0               grid_4.3.3                 \n[127] logger_0.4.0                vctrs_0.6.5                \n[129] RANN_2.6.2                  VGAM_1.1-13                \n[131] promises_1.3.3              xtable_1.8-4               \n[133] cluster_2.1.8.1             truncdist_1.0-2            \n[135] cli_3.6.5                   compiler_4.3.3             \n[137] crayon_1.5.3                future.apply_1.20.0        \n[139] plyr_1.8.9                  forcats_1.0.0              \n[141] stringi_1.8.7               viridisLite_0.4.2          \n[143] tidyseurat_0.8.0            deldir_2.0-4               \n[145] scplotter_0.6.5             assertthat_0.2.1           \n[147] gsl_2.1-8                   lazyeval_0.2.2             \n[149] spatstat.geom_3.5-0         quantreg_6.1               \n[151] Matrix_1.6-5                RcppHNSW_0.6.0             \n[153] patchwork_1.3.2             future_1.67.0              \n[155] ggplot2_3.5.2               shiny_1.11.1               \n[157] SummarizedExperiment_1.32.0 evd_2.3-7.1                \n[159] ROCR_1.0-11                 gridtext_0.1.5             \n[161] memoise_2.0.1               igraph_2.1.4               \n[163] bracer_1.2.1               \n",time:"# Generated by pipen-runinfo v1.1.2\n\nCommand: stdbuf -oL Rscript /mnt/disks/.cwd/.pipen/Immunopipe/SampleInfo/0/job.script\nVoluntary context switches: 362\nInvoluntary context switches: 180\nPercentage of CPU this job got: 159%\nMajor page faults: 0\nMinor page faults: 202684\nMaximum resident set size (kB): 813032\nElapsed real time (s): 8.92\nSystem (kernel) time (s): 0.47\nUser time (s): 13.79\nExit status: 0\n",device:"# Generated by pipen-runinfo v1.1.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nf523046eea1a\n\nCPU\n----\nArchitecture:                            x86_64\nCPU op-mode(s):                          32-bit, 64-bit\nAddress sizes:                           46 bits physical, 48 bits virtual\nByte Order:                              Little Endian\nCPU(s):                                  64\nOn-line CPU(s) list:                     0-63\nVendor ID:                               GenuineIntel\nModel name:                              Intel(R) Xeon(R) CPU @ 2.80GHz\nCPU family:                              6\nModel:                                   85\nThread(s) per core:                      2\nCore(s) per socket:                      16\nSocket(s):                               2\nStepping:                                7\nBogoMIPS:                                5599.99\nFlags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch ssbd ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves arat avx512_vnni md_clear arch_capabilities\nHypervisor vendor:                       KVM\nVirtualization type:                     full\nL1d cache:                               1 MiB (32 instances)\nL1i cache:                               1 MiB (32 instances)\nL2 cache:                                32 MiB (32 instances)\nL3 cache:                                66 MiB (2 instances)\nNUMA node(s):                            2\nNUMA node0 CPU(s):                       0-15,32-47\nNUMA node1 CPU(s):                       16-31,48-63\nVulnerability Gather data sampling:      Not affected\nVulnerability Indirect target selection: Mitigation; Aligned branch/return thunks\nVulnerability Itlb multihit:             Not affected\nVulnerability L1tf:                      Not affected\nVulnerability Mds:                       Not affected\nVulnerability Meltdown:                  Not affected\nVulnerability Mmio stale data:           Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown\nVulnerability Reg file data sampling:    Not affected\nVulnerability Retbleed:                  Mitigation; Enhanced IBRS\nVulnerability Spec rstack overflow:      Not affected\nVulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:                Mitigation; Enhanced / Automatic IBRS; IBPB conditional; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                     Not affected\nVulnerability Tsa:                       Not affected\nVulnerability Tsx async abort:           Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown\nVulnerability Vmscape:                   Not affected\n\nMemory\n------\n               total        used        free      shared  buff/cache   available\nMem:           503Gi       4.7Gi       494Gi       704Ki       7.1Gi       499Gi\nSwap:             0B          0B          0B\n\nDisk\n----\nFilesystem                                     Size  Used Avail Use% Mounted on\noverlay                                        248G  8.8G  240G   4% /\ntmpfs                                           64M     0   64M   0% /dev\nshm                                             64M     0   64M   0% /dev/shm\n/dev/sda1                                      248G  8.8G  240G   4% /etc/hosts\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/xqute_workdir\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/editable\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/.cwd\ntmpfs                                          252G     0  252G   0% /proc/acpi\ntmpfs                                          252G     0  252G   0% /proc/scsi\ntmpfs                                          252G     0  252G   0% /sys/firmware\n\nNetwork\n-------\nNeither ifconfig nor ip is available.\n\nGPU\n---\nnvidia-smi is not available.\n\n"}},{name:"SeuratPreparing",desc:"Load, prepare and apply QC to data, using `Seurat`",npages:1,report_toc:!0,order:2,runinfo:{session:"# Generated by pipen_runinfo v%(version)s\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 12 (bookworm)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.30.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n[1] biopipen.utils_0.3.5 glue_1.8.0           dplyr_1.1.4         \n[4] bracer_1.2.1         future_1.67.0        Seurat_5.3.0        \n[7] SeuratObject_5.2.0   sp_2.2-0            \n\nloaded via a namespace (and not attached):\n  [1] cubature_2.0.4.6            RcppAnnoy_0.0.22           \n  [3] splines_4.3.3               later_1.4.4                \n  [5] bitops_1.0-9                R.oo_1.27.1                \n  [7] tibble_3.3.0                polyclip_1.10-7            \n  [9] enrichit_0.0.4              fastDummies_1.7.5          \n [11] lifecycle_1.0.4             globals_0.18.0             \n [13] lattice_0.22-7              MASS_7.3-60.0.1            \n [15] magrittr_2.0.3              plotly_4.11.0              \n [17] httpuv_1.6.16               sctransform_0.4.2          \n [19] spam_2.11-1                 spatstat.sparse_3.1-0      \n [21] reticulate_1.43.0           cowplot_1.2.0              \n [23] pbapply_1.7-4               RColorBrewer_1.1-3         \n [25] abind_1.4-5                 zlibbioc_1.48.0            \n [27] Rtsne_0.17                  GenomicRanges_1.54.1       \n [29] R.utils_2.13.0              purrr_1.1.0                \n [31] ggraph_2.2.1                BiocGenerics_0.48.1        \n [33] RCurl_1.98-1.17             tweenr_2.0.3               \n [35] evmix_2.12                  circlize_0.4.16            \n [37] GenomeInfoDbData_1.2.11     IRanges_2.36.0             \n [39] S4Vectors_0.40.2            ggrepel_0.9.6              \n [41] irlba_2.3.5.1               listenv_0.9.1              \n [43] spatstat.utils_3.1-5        iNEXT_3.0.2                \n [45] MatrixModels_0.5-4          goftest_1.2-3              \n [47] RSpectra_0.16-2             scRepertoire_2.2.1         \n [49] spatstat.random_3.4-1       fitdistrplus_1.2-4         \n [51] parallelly_1.45.1           codetools_0.2-20           \n [53] DelayedArray_0.28.0         xml2_1.4.0                 \n [55] ggforce_0.5.0               tidyselect_1.2.1           \n [57] shape_1.4.6.1               farver_2.1.2               \n [59] viridis_0.6.5               matrixStats_1.5.0          \n [61] stats4_4.3.3                spatstat.explore_3.5-2     \n [63] jsonlite_2.0.0              ellipsis_0.3.2             \n [65] tidygraph_1.3.0             progressr_0.15.1           \n [67] ggridges_0.5.7              ggalluvial_0.12.5          \n [69] survival_3.8-3              systemfonts_1.2.3          \n [71] tools_4.3.3                 ggnewscale_0.5.2           \n [73] ragg_1.5.0                  stringdist_0.9.15          \n [75] ica_1.0-3                   Rcpp_1.1.0                 \n [77] gridExtra_2.3               SparseArray_1.2.2          \n [79] qs2_0.1.5                   qvalue_2.34.0              \n [81] MatrixGenerics_1.14.0       GenomeInfoDb_1.38.1        \n [83] withr_3.0.2                 fastmap_1.2.0              \n [85] fansi_1.0.6                 SparseM_1.84-2             \n [87] ttservice_0.5.3             digest_0.6.37              \n [89] R6_2.6.1                    mime_0.13                  \n [91] textshaping_0.4.0           colorspace_2.1-1           \n [93] scattermore_1.2             tensor_1.5.1               \n [95] dichromat_2.0-0.1           spatstat.data_3.1-8        \n [97] R.methodsS3_1.8.2           tidyr_1.3.1                \n [99] generics_0.1.4              data.table_1.17.8          \n[101] graphlayouts_1.2.2          httr_1.4.7                 \n[103] htmlwidgets_1.6.4           S4Arrays_1.2.0             \n[105] uwot_0.2.3                  pkgconfig_2.0.3            \n[107] gtable_0.3.6                lmtest_0.9-40              \n[109] SingleCellExperiment_1.24.0 XVector_0.42.0             \n[111] htmltools_0.5.8.1           dotCall64_1.2              \n[113] gglogger_0.1.7              scales_1.4.0               \n[115] Biobase_2.62.0              png_0.1-8                  \n[117] spatstat.univar_3.1-4       ggdendro_0.2.0             \n[119] rjson_0.2.23                reshape2_1.4.4             \n[121] nlme_3.1-168                cachem_1.1.0               \n[123] zoo_1.8-14                  GlobalOptions_0.1.2        \n[125] stringr_1.5.2               KernSmooth_2.23-26         \n[127] parallel_4.3.3              miniUI_0.1.2               \n[129] pillar_1.11.0               grid_4.3.3                 \n[131] logger_0.4.0                vctrs_0.6.5                \n[133] RANN_2.6.2                  VGAM_1.1-13                \n[135] promises_1.3.3              stringfish_0.17.0          \n[137] xtable_1.8-4                cluster_2.1.8.1            \n[139] truncdist_1.0-2             cli_3.6.5                  \n[141] compiler_4.3.3              rlang_1.1.6                \n[143] crayon_1.5.3                future.apply_1.20.0        \n[145] labeling_0.4.3              plyr_1.8.9                 \n[147] forcats_1.0.0               stringi_1.8.7              \n[149] viridisLite_0.4.2           tidyseurat_0.8.0           \n[151] deldir_2.0-4                scplotter_0.6.5            \n[153] assertthat_0.2.1            gsl_2.1-8                  \n[155] lazyeval_0.2.2              spatstat.geom_3.5-0        \n[157] quantreg_6.1                Matrix_1.6-5               \n[159] RcppHNSW_0.6.0              patchwork_1.3.2            \n[161] ggplot2_3.5.2               shiny_1.11.1               \n[163] plotthis_0.10.0             SummarizedExperiment_1.32.0\n[165] evd_2.3-7.1                 ROCR_1.0-11                \n[167] gridtext_0.1.5              memoise_2.0.1              \n[169] igraph_2.1.4                RcppParallel_5.1.9         \n",time:"# Generated by pipen-runinfo v1.1.2\n\nCommand: stdbuf -oL Rscript /mnt/disks/.cwd/.pipen/Immunopipe/SeuratPreparing/0/job.script\nVoluntary context switches: 646471\nInvoluntary context switches: 1075267\nPercentage of CPU this job got: 961%\nMajor page faults: 22\nMinor page faults: 130836646\nMaximum resident set size (kB): 33712364\nElapsed real time (s): 1199.30\nSystem (kernel) time (s): 355.83\nUser time (s): 11177.76\nExit status: 0\n",device:"# Generated by pipen-runinfo v1.1.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nf523046eea1a\n\nCPU\n----\nArchitecture:                            x86_64\nCPU op-mode(s):                          32-bit, 64-bit\nAddress sizes:                           46 bits physical, 48 bits virtual\nByte Order:                              Little Endian\nCPU(s):                                  64\nOn-line CPU(s) list:                     0-63\nVendor ID:                               GenuineIntel\nModel name:                              Intel(R) Xeon(R) CPU @ 2.80GHz\nCPU family:                              6\nModel:                                   85\nThread(s) per core:                      2\nCore(s) per socket:                      16\nSocket(s):                               2\nStepping:                                7\nBogoMIPS:                                5599.99\nFlags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch ssbd ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves arat avx512_vnni md_clear arch_capabilities\nHypervisor vendor:                       KVM\nVirtualization type:                     full\nL1d cache:                               1 MiB (32 instances)\nL1i cache:                               1 MiB (32 instances)\nL2 cache:                                32 MiB (32 instances)\nL3 cache:                                66 MiB (2 instances)\nNUMA node(s):                            2\nNUMA node0 CPU(s):                       0-15,32-47\nNUMA node1 CPU(s):                       16-31,48-63\nVulnerability Gather data sampling:      Not affected\nVulnerability Indirect target selection: Mitigation; Aligned branch/return thunks\nVulnerability Itlb multihit:             Not affected\nVulnerability L1tf:                      Not affected\nVulnerability Mds:                       Not affected\nVulnerability Meltdown:                  Not affected\nVulnerability Mmio stale data:           Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown\nVulnerability Reg file data sampling:    Not affected\nVulnerability Retbleed:                  Mitigation; Enhanced IBRS\nVulnerability Spec rstack overflow:      Not affected\nVulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:                Mitigation; Enhanced / Automatic IBRS; IBPB conditional; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                     Not affected\nVulnerability Tsa:                       Not affected\nVulnerability Tsx async abort:           Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown\nVulnerability Vmscape:                   Not affected\n\nMemory\n------\n               total        used        free      shared  buff/cache   available\nMem:           503Gi       4.3Gi       492Gi       704Ki        10Gi       499Gi\nSwap:             0B          0B          0B\n\nDisk\n----\nFilesystem                                     Size  Used Avail Use% Mounted on\noverlay                                        248G   11G  238G   5% /\ntmpfs                                           64M     0   64M   0% /dev\nshm                                             64M     0   64M   0% /dev/shm\n/dev/sda1                                      248G   11G  238G   5% /etc/hosts\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/xqute_workdir\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/editable\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/.cwd\ntmpfs                                          252G     0  252G   0% /proc/acpi\ntmpfs                                          252G     0  252G   0% /proc/scsi\ntmpfs                                          252G     0  252G   0% /sys/firmware\n\nNetwork\n-------\nNeither ifconfig nor ip is available.\n\nGPU\n---\nnvidia-smi is not available.\n\n"}},{name:"ClusterMarkersOfAllCells",desc:"Markers for clusters of all cells.",npages:1,report_toc:!0,order:4,runinfo:{session:"# Generated by pipen_runinfo v%(version)s\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 12 (bookworm)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.30.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] biopipen.utils_0.3.5 plotthis_0.10.0      ggplot2_3.5.2       \n [4] tidyr_1.3.1          tidyseurat_0.8.0     ttservice_0.5.3     \n [7] Seurat_5.3.0         SeuratObject_5.2.0   sp_2.2-0            \n[10] dplyr_1.1.4          rlang_1.1.6         \n\nloaded via a namespace (and not attached):\n  [1] cubature_2.0.4.6            RcppAnnoy_0.0.22           \n  [3] splines_4.3.3               later_1.4.4                \n  [5] bitops_1.0-9                tibble_3.3.0               \n  [7] polyclip_1.10-7             enrichit_0.0.4             \n  [9] fastDummies_1.7.5           lifecycle_1.0.4            \n [11] doParallel_1.0.17           globals_0.18.0             \n [13] lattice_0.22-7              MASS_7.3-60.0.1            \n [15] magrittr_2.0.3              limma_3.58.1               \n [17] plotly_4.11.0               httpuv_1.6.16              \n [19] sctransform_0.4.2           spam_2.11-1                \n [21] spatstat.sparse_3.1-0       reticulate_1.43.0          \n [23] cowplot_1.2.0               pbapply_1.7-4              \n [25] RColorBrewer_1.1-3          abind_1.4-5                \n [27] zlibbioc_1.48.0             Rtsne_0.17                 \n [29] GenomicRanges_1.54.1        presto_1.0.0               \n [31] purrr_1.1.0                 ggraph_2.2.1               \n [33] BiocGenerics_0.48.1         RCurl_1.98-1.17            \n [35] tweenr_2.0.3                evmix_2.12                 \n [37] circlize_0.4.16             GenomeInfoDbData_1.2.11    \n [39] IRanges_2.36.0              S4Vectors_0.40.2           \n [41] ggrepel_0.9.6               irlba_2.3.5.1              \n [43] listenv_0.9.1               spatstat.utils_3.1-5       \n [45] iNEXT_3.0.2                 MatrixModels_0.5-4         \n [47] goftest_1.2-3               RSpectra_0.16-2            \n [49] scRepertoire_2.2.1          spatstat.random_3.4-1      \n [51] fitdistrplus_1.2-4          parallelly_1.45.1          \n [53] codetools_0.2-20            DelayedArray_0.28.0        \n [55] xml2_1.4.0                  ggforce_0.5.0              \n [57] tidyselect_1.2.1            shape_1.4.6.1              \n [59] farver_2.1.2                viridis_0.6.5              \n [61] matrixStats_1.5.0           stats4_4.3.3               \n [63] spatstat.explore_3.5-2      jsonlite_2.0.0             \n [65] GetoptLong_1.0.5            ellipsis_0.3.2             \n [67] tidygraph_1.3.0             progressr_0.15.1           \n [69] iterators_1.0.14            ggridges_0.5.7             \n [71] ggalluvial_0.12.5           survival_3.8-3             \n [73] systemfonts_1.2.3           foreach_1.5.2              \n [75] tools_4.3.3                 ggnewscale_0.5.2           \n [77] ragg_1.5.0                  stringdist_0.9.15          \n [79] ica_1.0-3                   Rcpp_1.1.0                 \n [81] glue_1.8.0                  gridExtra_2.3              \n [83] SparseArray_1.2.2           qs2_0.1.5                  \n [85] qvalue_2.34.0               MatrixGenerics_1.14.0      \n [87] GenomeInfoDb_1.38.1         withr_3.0.2                \n [89] fastmap_1.2.0               fansi_1.0.6                \n [91] SparseM_1.84-2              digest_0.6.37              \n [93] R6_2.6.1                    mime_0.13                  \n [95] textshaping_0.4.0           colorspace_2.1-1           \n [97] Cairo_1.6-2                 scattermore_1.2            \n [99] tensor_1.5.1                dichromat_2.0-0.1          \n[101] spatstat.data_3.1-8         generics_0.1.4             \n[103] data.table_1.17.8           graphlayouts_1.2.2         \n[105] httr_1.4.7                  htmlwidgets_1.6.4          \n[107] S4Arrays_1.2.0              uwot_0.2.3                 \n[109] pkgconfig_2.0.3             gtable_0.3.6               \n[111] ComplexHeatmap_2.18.0       lmtest_0.9-40              \n[113] SingleCellExperiment_1.24.0 XVector_0.42.0             \n[115] htmltools_0.5.8.1           dotCall64_1.2              \n[117] clue_0.3-66                 scales_1.4.0               \n[119] Biobase_2.62.0              png_0.1-8                  \n[121] spatstat.univar_3.1-4       ggdendro_0.2.0             \n[123] rjson_0.2.23                reshape2_1.4.4             \n[125] nlme_3.1-168                cachem_1.1.0               \n[127] zoo_1.8-14                  GlobalOptions_0.1.2        \n[129] stringr_1.5.2               KernSmooth_2.23-26         \n[131] parallel_4.3.3              miniUI_0.1.2               \n[133] pillar_1.11.0               grid_4.3.3                 \n[135] logger_0.4.0                vctrs_0.6.5                \n[137] RANN_2.6.2                  VGAM_1.1-13                \n[139] promises_1.3.3              stringfish_0.17.0          \n[141] xtable_1.8-4                cluster_2.1.8.1            \n[143] truncdist_1.0-2             cli_3.6.5                  \n[145] compiler_4.3.3              crayon_1.5.3               \n[147] future.apply_1.20.0         labeling_0.4.3             \n[149] plyr_1.8.9                  forcats_1.0.0              \n[151] stringi_1.8.7               viridisLite_0.4.2          \n[153] deldir_2.0-4                scplotter_0.6.5            \n[155] assertthat_0.2.1            gsl_2.1-8                  \n[157] lazyeval_0.2.2              spatstat.geom_3.5-0        \n[159] quantreg_6.1                Matrix_1.6-5               \n[161] RcppHNSW_0.6.0              patchwork_1.3.2            \n[163] future_1.67.0               statmod_1.5.0              \n[165] shiny_1.11.1                SummarizedExperiment_1.32.0\n[167] evd_2.3-7.1                 ROCR_1.0-11                \n[169] gridtext_0.1.5              memoise_2.0.1              \n[171] igraph_2.1.4                RcppParallel_5.1.9         \n[173] bracer_1.2.1               \n",time:"# Generated by pipen-runinfo v1.1.2\n\nCommand: stdbuf -oL Rscript /mnt/disks/.cwd/.pipen/Immunopipe/ClusterMarkersOfAllCells/0/job.script\nVoluntary context switches: 25948\nInvoluntary context switches: 3192\nPercentage of CPU this job got: 97%\nMajor page faults: 0\nMinor page faults: 167714326\nMaximum resident set size (kB): 27154380\nElapsed real time (s): 1277.38\nSystem (kernel) time (s): 201.71\nUser time (s): 1047.79\nExit status: 0\n",device:"# Generated by pipen-runinfo v1.1.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nf523046eea1a\n\nCPU\n----\nArchitecture:                            x86_64\nCPU op-mode(s):                          32-bit, 64-bit\nAddress sizes:                           46 bits physical, 48 bits virtual\nByte Order:                              Little Endian\nCPU(s):                                  64\nOn-line CPU(s) list:                     0-63\nVendor ID:                               GenuineIntel\nModel name:                              Intel(R) Xeon(R) CPU @ 2.80GHz\nCPU family:                              6\nModel:                                   85\nThread(s) per core:                      2\nCore(s) per socket:                      16\nSocket(s):                               2\nStepping:                                7\nBogoMIPS:                                5599.99\nFlags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch ssbd ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves arat avx512_vnni md_clear arch_capabilities\nHypervisor vendor:                       KVM\nVirtualization type:                     full\nL1d cache:                               1 MiB (32 instances)\nL1i cache:                               1 MiB (32 instances)\nL2 cache:                                32 MiB (32 instances)\nL3 cache:                                66 MiB (2 instances)\nNUMA node(s):                            2\nNUMA node0 CPU(s):                       0-15,32-47\nNUMA node1 CPU(s):                       16-31,48-63\nVulnerability Gather data sampling:      Not affected\nVulnerability Indirect target selection: Mitigation; Aligned branch/return thunks\nVulnerability Itlb multihit:             Not affected\nVulnerability L1tf:                      Not affected\nVulnerability Mds:                       Not affected\nVulnerability Meltdown:                  Not affected\nVulnerability Mmio stale data:           Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown\nVulnerability Reg file data sampling:    Not affected\nVulnerability Retbleed:                  Mitigation; Enhanced IBRS\nVulnerability Spec rstack overflow:      Not affected\nVulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:                Mitigation; Enhanced / Automatic IBRS; IBPB conditional; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                     Not affected\nVulnerability Tsa:                       Not affected\nVulnerability Tsx async abort:           Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown\nVulnerability Vmscape:                   Not affected\n\nMemory\n------\n               total        used        free      shared  buff/cache   available\nMem:           503Gi       4.4Gi       488Gi       704Ki        13Gi       499Gi\nSwap:             0B          0B          0B\n\nDisk\n----\nFilesystem                                     Size  Used Avail Use% Mounted on\noverlay                                        248G   11G  238G   5% /\ntmpfs                                           64M     0   64M   0% /dev\nshm                                             64M     0   64M   0% /dev/shm\n/dev/sda1                                      248G   11G  238G   5% /etc/hosts\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/xqute_workdir\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/editable\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/.cwd\ntmpfs                                          252G     0  252G   0% /proc/acpi\ntmpfs                                          252G     0  252G   0% /proc/scsi\ntmpfs                                          252G     0  252G   0% /sys/firmware\n\nNetwork\n-------\nNeither ifconfig nor ip is available.\n\nGPU\n---\nnvidia-smi is not available.\n\n"}},{name:"TOrBCellSelection",desc:"Separate T and non-T cells and select T cells; or separate B and non-B cells and select B cells.",npages:1,report_toc:!0,order:5,runinfo:{session:"# Generated by pipen_runinfo v%(version)s\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 12 (bookworm)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.30.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] biopipen.utils_0.3.5 scplotter_0.6.5      plotthis_0.10.0     \n [4] Seurat_5.3.0         SeuratObject_5.2.0   sp_2.2-0            \n [7] tibble_3.3.0         tidyr_1.3.1          dplyr_1.1.4         \n[10] rlang_1.1.6         \n\nloaded via a namespace (and not attached):\n  [1] cubature_2.0.4.6            RcppAnnoy_0.0.22           \n  [3] splines_4.3.3               later_1.4.4                \n  [5] bitops_1.0-9                polyclip_1.10-7            \n  [7] enrichit_0.0.4              fastDummies_1.7.5          \n  [9] lifecycle_1.0.4             globals_0.18.0             \n [11] lattice_0.22-7              MASS_7.3-60.0.1            \n [13] magrittr_2.0.3              plotly_4.11.0              \n [15] httpuv_1.6.16               sctransform_0.4.2          \n [17] spam_2.11-1                 spatstat.sparse_3.1-0      \n [19] reticulate_1.43.0           cowplot_1.2.0              \n [21] pbapply_1.7-4               RColorBrewer_1.1-3         \n [23] abind_1.4-5                 zlibbioc_1.48.0            \n [25] Rtsne_0.17                  GenomicRanges_1.54.1       \n [27] purrr_1.1.0                 ggraph_2.2.1               \n [29] BiocGenerics_0.48.1         RCurl_1.98-1.17            \n [31] tweenr_2.0.3                evmix_2.12                 \n [33] circlize_0.4.16             GenomeInfoDbData_1.2.11    \n [35] IRanges_2.36.0              S4Vectors_0.40.2           \n [37] ggrepel_0.9.6               irlba_2.3.5.1              \n [39] listenv_0.9.1               spatstat.utils_3.1-5       \n [41] iNEXT_3.0.2                 MatrixModels_0.5-4         \n [43] goftest_1.2-3               RSpectra_0.16-2            \n [45] scRepertoire_2.2.1          spatstat.random_3.4-1      \n [47] fitdistrplus_1.2-4          parallelly_1.45.1          \n [49] codetools_0.2-20            DelayedArray_0.28.0        \n [51] xml2_1.4.0                  ggforce_0.5.0              \n [53] tidyselect_1.2.1            shape_1.4.6.1              \n [55] farver_2.1.2                viridis_0.6.5              \n [57] matrixStats_1.5.0           stats4_4.3.3               \n [59] spatstat.explore_3.5-2      jsonlite_2.0.0             \n [61] ellipsis_0.3.2              tidygraph_1.3.0            \n [63] progressr_0.15.1            ggridges_0.5.7             \n [65] ggalluvial_0.12.5           survival_3.8-3             \n [67] tools_4.3.3                 ggnewscale_0.5.2           \n [69] stringdist_0.9.15           ica_1.0-3                  \n [71] Rcpp_1.1.0                  glue_1.8.0                 \n [73] gridExtra_2.3               SparseArray_1.2.2          \n [75] qs2_0.1.5                   qvalue_2.34.0              \n [77] MatrixGenerics_1.14.0       GenomeInfoDb_1.38.1        \n [79] withr_3.0.2                 fastmap_1.2.0              \n [81] fansi_1.0.6                 ttservice_0.5.3            \n [83] SparseM_1.84-2              digest_0.6.37              \n [85] R6_2.6.1                    mime_0.13                  \n [87] colorspace_2.1-1            scattermore_1.2            \n [89] tensor_1.5.1                dichromat_2.0-0.1          \n [91] spatstat.data_3.1-8         generics_0.1.4             \n [93] data.table_1.17.8           graphlayouts_1.2.2         \n [95] httr_1.4.7                  htmlwidgets_1.6.4          \n [97] S4Arrays_1.2.0              uwot_0.2.3                 \n [99] pkgconfig_2.0.3             gtable_0.3.6               \n[101] lmtest_0.9-40               SingleCellExperiment_1.24.0\n[103] XVector_0.42.0              htmltools_0.5.8.1          \n[105] dotCall64_1.2               scales_1.4.0               \n[107] Biobase_2.62.0              png_0.1-8                  \n[109] spatstat.univar_3.1-4       ggdendro_0.2.0             \n[111] reshape2_1.4.4              rjson_0.2.23               \n[113] nlme_3.1-168                zoo_1.8-14                 \n[115] cachem_1.1.0                GlobalOptions_0.1.2        \n[117] stringr_1.5.2               KernSmooth_2.23-26         \n[119] parallel_4.3.3              miniUI_0.1.2               \n[121] logger_0.4.0                pillar_1.11.0              \n[123] grid_4.3.3                  vctrs_0.6.5                \n[125] RANN_2.6.2                  VGAM_1.1-13                \n[127] promises_1.3.3              stringfish_0.17.0          \n[129] xtable_1.8-4                cluster_2.1.8.1            \n[131] truncdist_1.0-2             cli_3.6.5                  \n[133] compiler_4.3.3              crayon_1.5.3               \n[135] future.apply_1.20.0         labeling_0.4.3             \n[137] plyr_1.8.9                  forcats_1.0.0              \n[139] stringi_1.8.7               tidyseurat_0.8.0           \n[141] viridisLite_0.4.2           deldir_2.0-4               \n[143] assertthat_0.2.1            gsl_2.1-8                  \n[145] lazyeval_0.2.2              spatstat.geom_3.5-0        \n[147] quantreg_6.1                Matrix_1.6-5               \n[149] RcppHNSW_0.6.0              patchwork_1.3.2            \n[151] future_1.67.0               ggplot2_3.5.2              \n[153] shiny_1.11.1                SummarizedExperiment_1.32.0\n[155] evd_2.3-7.1                 ROCR_1.0-11                \n[157] gridtext_0.1.5              igraph_2.1.4               \n[159] memoise_2.0.1               RcppParallel_5.1.9         \n[161] bracer_1.2.1               \n",time:"# Generated by pipen-runinfo v1.1.2\n\nCommand: stdbuf -oL Rscript /mnt/disks/.cwd/.pipen/Immunopipe/TOrBCellSelection/0/job.script\nVoluntary context switches: 5423\nInvoluntary context switches: 446\nPercentage of CPU this job got: 98%\nMajor page faults: 0\nMinor page faults: 4944127\nMaximum resident set size (kB): 12627788\nElapsed real time (s): 48.87\nSystem (kernel) time (s): 6.59\nUser time (s): 41.38\nExit status: 0\n",device:"# Generated by pipen-runinfo v1.1.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\nf523046eea1a\n\nCPU\n----\nArchitecture:                            x86_64\nCPU op-mode(s):                          32-bit, 64-bit\nAddress sizes:                           46 bits physical, 48 bits virtual\nByte Order:                              Little Endian\nCPU(s):                                  64\nOn-line CPU(s) list:                     0-63\nVendor ID:                               GenuineIntel\nModel name:                              Intel(R) Xeon(R) CPU @ 2.80GHz\nCPU family:                              6\nModel:                                   85\nThread(s) per core:                      2\nCore(s) per socket:                      16\nSocket(s):                               2\nStepping:                                7\nBogoMIPS:                                5599.99\nFlags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch ssbd ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves arat avx512_vnni md_clear arch_capabilities\nHypervisor vendor:                       KVM\nVirtualization type:                     full\nL1d cache:                               1 MiB (32 instances)\nL1i cache:                               1 MiB (32 instances)\nL2 cache:                                32 MiB (32 instances)\nL3 cache:                                66 MiB (2 instances)\nNUMA node(s):                            2\nNUMA node0 CPU(s):                       0-15,32-47\nNUMA node1 CPU(s):                       16-31,48-63\nVulnerability Gather data sampling:      Not affected\nVulnerability Indirect target selection: Mitigation; Aligned branch/return thunks\nVulnerability Itlb multihit:             Not affected\nVulnerability L1tf:                      Not affected\nVulnerability Mds:                       Not affected\nVulnerability Meltdown:                  Not affected\nVulnerability Mmio stale data:           Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown\nVulnerability Reg file data sampling:    Not affected\nVulnerability Retbleed:                  Mitigation; Enhanced IBRS\nVulnerability Spec rstack overflow:      Not affected\nVulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:                Mitigation; Enhanced / Automatic IBRS; IBPB conditional; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                     Not affected\nVulnerability Tsa:                       Not affected\nVulnerability Tsx async abort:           Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown\nVulnerability Vmscape:                   Not affected\n\nMemory\n------\n               total        used        free      shared  buff/cache   available\nMem:           503Gi       4.5Gi       487Gi       704Ki        14Gi       499Gi\nSwap:             0B          0B          0B\n\nDisk\n----\nFilesystem                                     Size  Used Avail Use% Mounted on\noverlay                                        248G   11G  238G   5% /\ntmpfs                                           64M     0   64M   0% /dev\nshm                                             64M     0   64M   0% /dev/shm\n/dev/sda1                                      248G   11G  238G   5% /etc/hosts\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/xqute_workdir\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/editable\nml-phi-staff-m161047-p-rsa-us-central1-p-84f8  1.0P     0  1.0P   0% /mnt/disks/.cwd\ntmpfs                                          252G     0  252G   0% /proc/acpi\ntmpfs                                          252G     0  252G   0% /proc/scsi\ntmpfs                                          252G     0  252G   0% /sys/firmware\n\nNetwork\n-------\nNeither ifconfig nor ip is available.\n\nGPU\n---\nnvidia-smi is not available.\n\n"}},{name:"SeuratClusterStats",desc:"Statistics of the clustering.",npages:2,report_toc:!0,order:9,runinfo:{session:"# Generated by pipen_runinfo v%(version)s\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 12 (bookworm)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.30.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] biopipen.utils_0.3.5 scplotter_0.6.5      gglogger_0.1.7      \n [4] ggplot2_3.5.2        tidyseurat_0.8.0     ttservice_0.5.3     \n [7] forcats_1.0.0        glue_1.8.0           tibble_3.3.0        \n[10] tidyr_1.3.1          dplyr_1.1.4          rlang_1.1.6         \n[13] Seurat_5.3.0         SeuratObject_5.2.0   sp_2.2-0            \n\nloaded via a namespace (and not attached):\n  [1] cubature_2.0.4.6            RcppAnnoy_0.0.22           \n  [3] splines_4.3.3               later_1.4.4                \n  [5] bitops_1.0-9                polyclip_1.10-7            \n  [7] enrichit_0.0.4              fastDummies_1.7.5          \n  [9] lifecycle_1.0.4             doParallel_1.0.17          \n [11] globals_0.18.0              lattice_0.22-7             \n [13] MASS_7.3-60.0.1             magrittr_2.0.3             \n [15] plotly_4.11.0               httpuv_1.6.16              \n [17] sctransform_0.4.2           spam_2.11-1                \n [19] spatstat.sparse_3.1-0       reticulate_1.43.0          \n [21] cowplot_1.2.0               pbapply_1.7-4              \n [23] RColorBrewer_1.1-3          abind_1.4-5                \n [25] zlibbioc_1.48.0             Rtsne_0.17                 \n [27] GenomicRanges_1.54.1        purrr_1.1.0                \n [29] ggraph_2.2.1                BiocGenerics_0.48.1        \n [31] RCurl_1.98-1.17             tweenr_2.0.3               \n [33] evmix_2.12                  circlize_0.4.16            \n [35] GenomeInfoDbData_1.2.11     IRanges_2.36.0             \n [37] S4Vectors_0.40.2            ggrepel_0.9.6              \n [39] irlba_2.3.5.1               listenv_0.9.1              \n [41] spatstat.utils_3.1-5        iNEXT_3.0.2                \n [43] MatrixModels_0.5-4          goftest_1.2-3              \n [45] RSpectra_0.16-2             scRepertoire_2.2.1         \n [47] spatstat.random_3.4-1       fitdistrplus_1.2-4         \n [49] parallelly_1.45.1           codetools_0.2-20           \n [51] DelayedArray_0.28.0         xml2_1.4.0                 \n [53] ggforce_0.5.0               tidyselect_1.2.1           \n [55] shape_1.4.6.1               farver_2.1.2               \n [57] viridis_0.6.5               matrixStats_1.5.0          \n [59] stats4_4.3.3                spatstat.explore_3.5-2     \n [61] jsonlite_2.0.0              GetoptLong_1.0.5           \n [63] ellipsis_0.3.2              tidygraph_1.3.0            \n [65] progressr_0.15.1            iterators_1.0.14           \n [67] ggridges_0.5.7              ggalluvial_0.12.5          \n [69] survival_3.8-3              systemfonts_1.2.3          \n [71] foreach_1.5.2               tools_4.3.3                \n [73] ggnewscale_0.5.2            ragg_1.5.0                 \n [75] stringdist_0.9.15           ica_1.0-3                  \n [77] Rcpp_1.1.0                  gridExtra_2.3              \n [79] SparseArray_1.2.2           qs2_0.1.5                  \n [81] qvalue_2.34.0               MatrixGenerics_1.14.0      \n [83] GenomeInfoDb_1.38.1         withr_3.0.2                \n [85] fastmap_1.2.0               fansi_1.0.6                \n [87] SparseM_1.84-2              digest_0.6.37              \n [89] R6_2.6.1                    mime_0.13                  \n [91] textshaping_0.4.0           colorspace_2.1-1           \n [93] Cairo_1.6-2                 scattermore_1.2            \n [95] tensor_1.5.1                dichromat_2.0-0.1          \n [97] spatstat.data_3.1-8         generics_0.1.4             \n [99] data.table_1.17.8           graphlayouts_1.2.2         \n[101] httr_1.4.7                  htmlwidgets_1.6.4          \n[103] S4Arrays_1.2.0              uwot_0.2.3                 \n[105] pkgconfig_2.0.3             gtable_0.3.6               \n[107] ComplexHeatmap_2.18.0       lmtest_0.9-40              \n[109] SingleCellExperiment_1.24.0 XVector_0.42.0             \n[111] htmltools_0.5.8.1           dotCall64_1.2              \n[113] clue_0.3-66                 scales_1.4.0               \n[115] Biobase_2.62.0              png_0.1-8                  \n[117] spatstat.univar_3.1-4       ggdendro_0.2.0             \n[119] rjson_0.2.23                reshape2_1.4.4             \n[121] nlme_3.1-168                zoo_1.8-14                 \n[123] cachem_1.1.0                GlobalOptions_0.1.2        \n[125] stringr_1.5.2               KernSmooth_2.23-26         \n[127] parallel_4.3.3              miniUI_0.1.2               \n[129] logger_0.4.0                pillar_1.11.0              \n[131] grid_4.3.3                  vctrs_0.6.5                \n[133] RANN_2.6.2                  VGAM_1.1-13                \n[135] promises_1.3.3              stringfish_0.17.0          \n[137] xtable_1.8-4                cluster_2.1.8.1            \n[139] truncdist_1.0-2             cli_3.6.5                  \n[141] compiler_4.3.3              crayon_1.5.3               \n[143] future.apply_1.20.0         labeling_0.4.3             \n[145] plyr_1.8.9                  stringi_1.8.7              \n[147] viridisLite_0.4.2           deldir_2.0-4               \n[149] assertthat_0.2.1            gsl_2.1-8                  \n[151] lazyeval_0.2.2              spatstat.geom_3.5-0        \n[153] quantreg_6.1                Matrix_1.6-5               \n[155] RcppHNSW_0.6.0              patchwork_1.3.2            \n[157] future_1.67.0               shiny_1.11.1               \n[159] plotthis_0.10.0             SummarizedExperiment_1.32.0\n[161] evd_2.3-7.1                 ROCR_1.0-11                \n[163] gridtext_0.1.5              igraph_2.1.4               \n[165] memoise_2.0.1               RcppParallel_5.1.9         \n[167] bracer_1.2.1               \n",time:"# Generated by pipen-runinfo v1.1.2\n\nCommand: stdbuf -oL Rscript /mnt/disks/.cwd/.pipen/Immunopipe/SeuratClusterStats/0/job.script\nVoluntary context switches: 5045\nInvoluntary context switches: 356\nPercentage of CPU this job got: 118%\nMajor page faults: 0\nMinor page faults: 2868055\nMaximum resident set size (kB): 6910156\nElapsed real time (s): 71.87\nSystem (kernel) time (s): 4.54\nUser time (s): 80.86\nExit status: 0\n",device:"# Generated by pipen-runinfo v1.1.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\n203b1956eb36\n\n"}},{name:"ClusterMarkers",desc:"Markers for clusters of all or selected T/B cells.",npages:1,report_toc:!0,order:10},{name:"ClonalStats",desc:"Visualize the clonal information.",npages:1,report_toc:!0,order:11}]};function Hn(e){let t,n;return t=new Et({props:{dir:"up"}}),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function jn(e){let t,n;return t=new Et({props:{dir:"down"}}),{c(){h(t.$$.fragment)},m(e,s){$(t,e,s),n=!0},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){o(t.$$.fragment,e),n=!1},d(e){g(t,e)}}}function qn(e){let t,n,s,r,i,_=e[1]>0&&Hn();n=new On({});let u=e[1]<e[3].npages-1&&jn();return{c(){_&&_.c(),t=b(),h(n.$$.fragment),s=b(),u&&u.c(),r=f()},m(e,a){_&&_.m(e,a),d(e,t,a),$(n,e,a),d(e,s,a),u&&u.m(e,a),d(e,r,a),i=!0},p(e,n){e[1]>0?_?2&n&&l(_,1):(_=Hn(),_.c(),l(_,1),_.m(t.parentNode,t)):_&&(c(),o(_,1,1,()=>{_=null}),p()),e[1]<e[3].npages-1?u?10&n&&l(u,1):(u=jn(),u.c(),l(u,1),u.m(r.parentNode,r)):u&&(c(),o(u,1,1,()=>{u=null}),p())},i(e){i||(l(_),l(n.$$.fragment,e),l(u),i=!0)},o(e){o(_),o(n.$$.fragment,e),o(u),i=!1},d(e){e&&(a(t),a(s),a(r)),_&&_.d(e),g(n,e),u&&u.d(e)}}}function Kn(e){let t,n,s;function r(t){e[4](t)}let a={logo:e[0],logotext:e[3].desc,versions:zn.versions,entries:zn.entries,pipeline_name:zn.pipeline.name,report_toc:e[3].report_toc,runinfo:e[3].runinfo,toc:Fn,page:e[1],$$slots:{default:[qn]},$$scope:{ctx:e}};return void 0!==e[2]&&(a.isSideNavOpen=e[2]),t=new Tt({props:a}),v.push(()=>x(t,"isSideNavOpen",r)),{c(){h(t.$$.fragment)},m(e,n){$(t,e,n),s=!0},p(e,[s]){const r={};1&s&&(r.logo=e[0]),8&s&&(r.logotext=e[3].desc),8&s&&(r.report_toc=e[3].report_toc),8&s&&(r.runinfo=e[3].runinfo),2&s&&(r.page=e[1]),42&s&&(r.$$scope={dirty:s,ctx:e}),!n&&4&s&&(n=!0,r.isSideNavOpen=e[2],y(()=>n=!1)),t.$set(r)},i(e){s||(l(t.$$.fragment,e),s=!0)},o(e){o(t.$$.fragment,e),s=!1},d(e){g(t,e)}}}function Xn(e,t,n){let s,{name:r}=t,{page:a=0}=t,i=!1;for(const e of zn.entries)if(e.procs){for(const t of e.procs)if(t.name==r){s=t;break}}else if(e.name==r){s=e;break}return m(()=>{const e=new URL(window.location.href);let t;e.hash.length>1&&(t=document.getElementById(`${e.hash.slice(1)}`)),t&&t.scrollIntoView({block:"start",behavior:"smooth"})}),e.$$set=e=>{"name"in e&&n(0,r=e.name),"page"in e&&n(1,a=e.page)},[r,a,i,s,function(e){i=e,n(2,i)}]}e("default",new class extends t{constructor(e){super(),n(this,e,Xn,Kn,s,{name:0,page:1})}}({target:document.body,props:{name:"SeuratClusterStats",page:0}}))}}});
